diff --git a/node_modules/@ariakit/react-core/.DS_Store b/node_modules/@ariakit/react-core/.DS_Store
new file mode 100644
index 0000000..42e34f5
Binary files /dev/null and b/node_modules/@ariakit/react-core/.DS_Store differ
diff --git a/node_modules/@ariakit/react-core/cjs/__chunks/CJDHQUBR.cjs b/node_modules/@ariakit/react-core/cjs/__chunks/CJDHQUBR.cjs
index 69c628e..ebad00d 100644
--- a/node_modules/@ariakit/react-core/cjs/__chunks/CJDHQUBR.cjs
+++ b/node_modules/@ariakit/react-core/cjs/__chunks/CJDHQUBR.cjs
@@ -25,7 +25,7 @@ function isValidElementWithRef(element) {
 function getRefProperty(element) {
   if (!isValidElementWithRef(element))
     return null;
-  return element.ref;
+  return element.props.ref || element.ref;
 }
 function mergeProps(base, overrides) {
   const props = _AV6KTKLEcjs.__spreadValues.call(void 0, {}, base);
diff --git a/node_modules/@ariakit/react-core/esm/.DS_Store b/node_modules/@ariakit/react-core/esm/.DS_Store
new file mode 100644
index 0000000..04abd12
Binary files /dev/null and b/node_modules/@ariakit/react-core/esm/.DS_Store differ
diff --git a/node_modules/@ariakit/react-core/esm/__chunks/XM66DUTO.js b/node_modules/@ariakit/react-core/esm/__chunks/XM66DUTO.js
index 7f904a3..c57b3fb 100644
--- a/node_modules/@ariakit/react-core/esm/__chunks/XM66DUTO.js
+++ b/node_modules/@ariakit/react-core/esm/__chunks/XM66DUTO.js
@@ -25,7 +25,7 @@ function isValidElementWithRef(element) {
 function getRefProperty(element) {
   if (!isValidElementWithRef(element))
     return null;
-  return element.ref;
+  return element.props.ref || element.ref;
 }
 function mergeProps(base, overrides) {
   const props = __spreadValues({}, base);
