diff --git a/node_modules/lighthouse/core/index.d.cts b/node_modules/lighthouse/core/index.d.cts
index 1c399e1..23c3d1f 100644
--- a/node_modules/lighthouse/core/index.d.cts
+++ b/node_modules/lighthouse/core/index.d.cts
@@ -1,4 +1,6 @@
 export = lighthouse;
 /** @type {import('./index.js')['default']} */
+// Otherwise TS is confused when using ES types in CJS.
+// @ts-ignore
 declare const lighthouse: typeof import("./index.js")['default'];
 //# sourceMappingURL=index.d.cts.map
