diff --git a/node_modules/react-native/Libraries/Components/ActivityIndicator/ActivityIndicator.js b/node_modules/react-native/Libraries/Components/ActivityIndicator/ActivityIndicator.js
index 52cbda4..3e3b71e 100644
--- a/node_modules/react-native/Libraries/Components/ActivityIndicator/ActivityIndicator.js
+++ b/node_modules/react-native/Libraries/Components/ActivityIndicator/ActivityIndicator.js
@@ -14,6 +14,7 @@
 const Platform = require('../../Utilities/Platform');
 const React = require('react');
 const StyleSheet = require('../../StyleSheet/StyleSheet');
+const {PlatformColor} = require('../../StyleSheet/PlatformColorValueTypes');
 const View = require('../View/View');
 import type {HostComponent} from '../../Renderer/shims/ReactNativeTypes';
 import type {ViewProps} from '../View/ViewPropTypes';
@@ -183,7 +184,11 @@ ActivityIndicatorWithRef.displayName = 'ActivityIndicator';
  * and run Flow. */
 ActivityIndicatorWithRef.defaultProps = {
   animating: true,
-  color: Platform.OS === 'ios' ? GRAY : null,
+	color: Platform.select({
+     ios: GRAY,
+     android: PlatformColor('?attr/colorControlActivated'),
+     default: null,
+   }),
   hidesWhenStopped: true,
   size: 'small',
 };
diff --git a/node_modules/react-native/Libraries/Image/AssetSourceResolver.js b/node_modules/react-native/Libraries/Image/AssetSourceResolver.js
index 5ebd97a..3f511b9 100644
--- a/node_modules/react-native/Libraries/Image/AssetSourceResolver.js
+++ b/node_modules/react-native/Libraries/Image/AssetSourceResolver.js
@@ -92,9 +92,18 @@ class AssetSourceResolver {
    */
   assetServerURL(): ResolvedAssetSource {
     invariant(!!this.serverUrl, 'need server to load from');
+		/**
+		 * The monorepo configuration we have set up is suffering from local static
+		 * images not displaying. It would appear the server URL is incorrect. This
+		 * is not a proper fix, but a stopgap.
+		 * - https://git.io/JBV4e
+		 * - https://git.io/JBV4k
+		 */
+		const brokenMonorepoAsset = /(^assets\/\.\.\/)(?!\.\.\/|node_modules)(.*)/g;
     return this.fromSource(
       this.serverUrl +
-        getScaledAssetPath(this.asset) +
+        getScaledAssetPath(this.asset)
+					.replace(brokenMonorepoAsset, '$1packages/$2') +
         '?platform=' +
         Platform.OS +
         '&hash=' +
