diff --git a/node_modules/react-native-gesture-handler/android/build.gradle b/node_modules/react-native-gesture-handler/android/build.gradle
index 9ddfa57..7c7218d 100644
--- a/node_modules/react-native-gesture-handler/android/build.gradle
+++ b/node_modules/react-native-gesture-handler/android/build.gradle
@@ -5,6 +5,7 @@ buildscript {
 
     repositories {
         mavenCentral()
+        google()
     }
 
     dependencies {
@@ -35,6 +36,7 @@ def shouldUseCommonInterfaceFromReanimated() {
 
 repositories {
     mavenCentral()
+    google()
 }
 
 android {
