diff --git a/packages/components/src/popover/index.js b/packages/components/src/popover/index.js
index c2844c56a51f9a..420ac78b76bdbf 100644
--- a/packages/components/src/popover/index.js
+++ b/packages/components/src/popover/index.js
@@ -53,11 +53,11 @@ function useThrottledWindowScrollOrResize( handler, ignoredScrollableRef ) {
 		};
 
 		window.addEventListener( 'resize', throttledRefresh );
-		window.addEventListener( 'scroll', throttledRefresh );
+		window.addEventListener( 'scroll', throttledRefresh, true );
 
 		return () => {
 			window.removeEventListener( 'resize', throttledRefresh );
-			window.removeEventListener( 'scroll', throttledRefresh );
+			window.removeEventListener( 'scroll', throttledRefresh, true );
 		};
 	}, [] );
 }
