diff --git a/packages/components/src/color-picker/saturation.js b/packages/components/src/color-picker/saturation.js
index 3b9666d734588f..7ce598aab35c5c 100644
--- a/packages/components/src/color-picker/saturation.js
+++ b/packages/components/src/color-picker/saturation.js
@@ -62,6 +62,17 @@ export class Saturation extends Component {
 		this.handleMouseUp = this.handleMouseUp.bind( this );
 	}
 
+	componentDidMount() {
+		// Ensure container is scrolled to the left.
+		// When pointer starts at right edge, scroll position is offset by a few pixels (issue #19610).
+		const findContainer = setInterval( () => {
+			if ( this.container.current ) {
+				this.container.current.scroll( 0, 0 );
+				clearInterval( findContainer );
+			}
+		}, 10 );
+	}
+
 	componentWillUnmount() {
 		this.throttle.cancel();
 		this.unbindEventListeners();
diff --git a/packages/components/src/color-picker/style.scss b/packages/components/src/color-picker/style.scss
index cecfed966e6d8f..e99d337edced7e 100644
--- a/packages/components/src/color-picker/style.scss
+++ b/packages/components/src/color-picker/style.scss
@@ -126,7 +126,7 @@
 		0 0 1px 2px rgba(0, 0, 0, 0.4);
 	border-radius: 50%;
 	background-color: transparent;
-	transform: translate(-4px, -4px);
+	transform: translate(-50%, -50%);
 }
 
 /* HUE & ALPHA BARS */
