diff --git a/packages/components/src/range-control/index.js b/packages/components/src/range-control/index.js
index cee2659ed04e85..d718dec1a229a0 100644
--- a/packages/components/src/range-control/index.js
+++ b/packages/components/src/range-control/index.js
@@ -64,7 +64,7 @@ const BaseRangeControl = forwardRef(
 			renderTooltipContent = ( v ) => v,
 			showTooltip: showTooltipProp,
 			step = 1,
-			value: valueProp = 0,
+			value: valueProp,
 			withInputField = true,
 			...props
 		},
@@ -72,7 +72,9 @@ const BaseRangeControl = forwardRef(
 	) => {
 		const isRTL = useRtl();
 
-		const sliderValue = valueProp || initialPosition;
+		const sliderValue =
+			valueProp !== undefined ? valueProp : initialPosition;
+
 		const [ value, setValue ] = useControlledRangeValue( {
 			min,
 			max,
@@ -95,8 +97,9 @@ const BaseRangeControl = forwardRef(
 		const isThumbFocused = ! disabled && isFocused;
 
 		const isValueReset = value === null;
-		const inputSliderValue = isValueReset ? '' : value;
-		const currentInputValue = isValueReset ? '' : value || currentInput;
+		const currentValue = value !== undefined ? value : currentInput;
+
+		const inputSliderValue = isValueReset ? '' : currentValue;
 
 		const rangeFillValue = isValueReset
 			? floatClamp( max / 2, min, max )
@@ -119,12 +122,20 @@ const BaseRangeControl = forwardRef(
 		const enableTooltip = showTooltipProp !== false && isFinite( value );
 
 		const handleOnChange = ( event ) => {
-			if ( ! event.target.checkValidity() ) {
+			if (
+				event.target.checkValidity &&
+				! event.target.checkValidity()
+			) {
 				return;
 			}
 
 			const nextValue = parseFloat( event.target.value );
 
+			if ( isNaN( nextValue ) ) {
+				handleOnReset();
+				return;
+			}
+
 			setValue( nextValue );
 			onChange( nextValue );
 		};
@@ -249,7 +260,7 @@ const BaseRangeControl = forwardRef(
 							onChange={ handleOnChange }
 							step={ step }
 							type="number"
-							value={ currentInputValue }
+							value={ inputSliderValue }
 						/>
 					) }
 					{ allowReset && (
diff --git a/packages/components/src/range-control/rail.js b/packages/components/src/range-control/rail.js
index d0bedcf7ac82b6..111cf8d96972d7 100644
--- a/packages/components/src/range-control/rail.js
+++ b/packages/components/src/range-control/rail.js
@@ -1,7 +1,3 @@
-/**
- * External dependencies
- */
-import { isUndefined } from 'lodash';
 /**
  * Internal dependencies
  */
@@ -78,7 +74,7 @@ function useMarks( { marks, min = 0, max = 100, step = 1, value = 0 } ) {
 		  } ) );
 
 	const enhancedMarks = marksArray.map( ( mark, index ) => {
-		const markValue = ! isUndefined( mark.value ) ? mark.value : value;
+		const markValue = mark.value !== undefined ? mark.value : value;
 
 		const key = `mark-${ index }`;
 		const isFilled = markValue * step <= value;
diff --git a/packages/components/src/range-control/test/index.js b/packages/components/src/range-control/test/index.js
index 4c67a4f9c1a8f6..31c72383a0271e 100644
--- a/packages/components/src/range-control/test/index.js
+++ b/packages/components/src/range-control/test/index.js
@@ -1,7 +1,8 @@
 /**
  * External dependencies
  */
-import TestUtils, { act } from 'react-dom/test-utils';
+import { render, unmountComponentAtNode } from 'react-dom';
+import TestUtils, { act, Simulate } from 'react-dom/test-utils';
 
 /**
  * Internal dependencies
@@ -14,6 +15,19 @@ import RangeControl from '../';
 import { Component } from '@wordpress/element';
 import { Dashicon } from '@wordpress/components';
 
+let container = null;
+
+beforeEach( () => {
+	container = document.createElement( 'div' );
+	document.body.appendChild( container );
+} );
+
+afterEach( () => {
+	unmountComponentAtNode( container );
+	container.remove();
+	container = null;
+} );
+
 describe( 'RangeControl', () => {
 	class TestWrapper extends Component {
 		render() {
@@ -326,6 +340,79 @@ describe( 'RangeControl', () => {
 		} );
 	} );
 
+	describe( 'input field', () => {
+		it( 'should render an input field by default', () => {
+			act( () => {
+				render( <RangeControl />, container );
+			} );
+			const field = container.querySelector( 'input[type="number"]' );
+
+			expect( field ).toBeTruthy();
+		} );
+
+		it( 'should not render an input field, if disabled', () => {
+			act( () => {
+				render( <RangeControl withInputField={ false } />, container );
+			} );
+			const field = container.querySelector( 'input[type="number"]' );
+
+			expect( field ).toBeFalsy();
+		} );
+
+		it( 'should render a zero value into input range and field', () => {
+			act( () => {
+				render( <RangeControl value={ 0 } />, container );
+			} );
+			const range = container.querySelector( 'input[type="range"]' );
+			const field = container.querySelector( 'input[type="number"]' );
+
+			expect( range.value ).toBe( '0' );
+			expect( field.value ).toBe( '0' );
+		} );
+
+		it( 'should update both field and range on change', () => {
+			act( () => {
+				render( <RangeControl value={ 0 } />, container );
+			} );
+			const range = container.querySelector( 'input[type="range"]' );
+			const field = container.querySelector( 'input[type="number"]' );
+
+			act( () => {
+				Simulate.change( range, { target: { value: 13 } } );
+			} );
+
+			expect( range.value ).toBe( '13' );
+			expect( field.value ).toBe( '13' );
+
+			act( () => {
+				Simulate.change( field, { target: { value: 7 } } );
+			} );
+
+			expect( range.value ).toBe( '7' );
+			expect( field.value ).toBe( '7' );
+		} );
+
+		it( 'should reset input values if next value is removed', () => {
+			act( () => {
+				render( <RangeControl value={ 13 } />, container );
+			} );
+			const range = container.querySelector( 'input[type="range"]' );
+			const field = container.querySelector( 'input[type="number"]' );
+
+			expect( range.value ).toBe( '13' );
+			expect( field.value ).toBe( '13' );
+
+			act( () => {
+				Simulate.change( field, { target: { value: undefined } } );
+			} );
+
+			// Reset to 50. Median value of min: 0, max: 100
+			expect( range.value ).toBe( '50' );
+			// Input field should be blank
+			expect( field.value ).toBe( '' );
+		} );
+	} );
+
 	describe( 'reset', () => {
 		class StatefulTestWrapper extends Component {
 			constructor( props ) {
diff --git a/packages/components/src/range-control/utils.js b/packages/components/src/range-control/utils.js
index d1fd148ec2a3fb..9d2aecfa5e7a70 100644
--- a/packages/components/src/range-control/utils.js
+++ b/packages/components/src/range-control/utils.js
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { clamp, isFinite, noop } from 'lodash';
+import { clamp, noop } from 'lodash';
 
 /**
  * WordPress dependencies
diff --git a/storybook/test/__snapshots__/index.js.snap b/storybook/test/__snapshots__/index.js.snap
index 73dea5ec7db1bd..d150204a163422 100644
--- a/storybook/test/__snapshots__/index.js.snap
+++ b/storybook/test/__snapshots__/index.js.snap
@@ -5421,6 +5421,7 @@ input[type='number'].emotion-14 {
         onChange={[Function]}
         step={1}
         type="number"
+        value={0}
       />
     </span>
   </div>
