diff --git a/packages/block-editor/src/components/url-popover/link-viewer-url.js b/packages/block-editor/src/components/url-popover/link-viewer-url.js
new file mode 100644
index 00000000000000..5cf710f2030097
--- /dev/null
+++ b/packages/block-editor/src/components/url-popover/link-viewer-url.js
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { ExternalLink } from '@wordpress/components';
+import { safeDecodeURI, filterURLForDisplay } from '@wordpress/url';
+
+export default function LinkViewerURL( { url, urlLabel, className } ) {
+	const linkClassName = classnames(
+		className,
+		'block-editor-url-popover__link-viewer-url'
+	);
+
+	if ( ! url ) {
+		return <span className={ linkClassName }></span>;
+	}
+
+	return (
+		<ExternalLink className={ linkClassName } href={ url }>
+			{ urlLabel || filterURLForDisplay( safeDecodeURI( url ) ) }
+		</ExternalLink>
+	);
+}
diff --git a/packages/block-editor/src/components/url-popover/link-viewer.js b/packages/block-editor/src/components/url-popover/link-viewer.js
index f5cbd7459355f8..09e1487bc4dcaf 100644
--- a/packages/block-editor/src/components/url-popover/link-viewer.js
+++ b/packages/block-editor/src/components/url-popover/link-viewer.js
@@ -7,26 +7,13 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { ExternalLink, Button } from '@wordpress/components';
-import { safeDecodeURI, filterURLForDisplay } from '@wordpress/url';
+import { Button } from '@wordpress/components';
 import { pencil } from '@wordpress/icons';
 
-function LinkViewerUrl( { url, urlLabel, className } ) {
-	const linkClassName = classnames(
-		className,
-		'block-editor-url-popover__link-viewer-url'
-	);
-
-	if ( ! url ) {
-		return <span className={ linkClassName }></span>;
-	}
-
-	return (
-		<ExternalLink className={ linkClassName } href={ url }>
-			{ urlLabel || filterURLForDisplay( safeDecodeURI( url ) ) }
-		</ExternalLink>
-	);
-}
+/**
+ * Internal dependencies
+ */
+import LinkViewerURL from './link-viewer-url';
 
 export default function LinkViewer( {
 	className,
@@ -44,7 +31,7 @@ export default function LinkViewer( {
 			) }
 			{ ...props }
 		>
-			<LinkViewerUrl
+			<LinkViewerURL
 				url={ url }
 				urlLabel={ urlLabel }
 				className={ linkClassName }
