diff --git a/packages/components/src/date-time/index.js b/packages/components/src/date-time/index.js
index dae442ef0d3659..45b069b6c3d157 100644
--- a/packages/components/src/date-time/index.js
+++ b/packages/components/src/date-time/index.js
@@ -149,37 +149,30 @@ export class DateTimePicker extends Component {
 									) }
 								</li>
 							</ul>
-
-							<Button
-								isSmall
-								onClick={ this.onClickDescriptionToggle }
-							>
-								{ __( 'Close' ) }
-							</Button>
 						</div>
 					</>
 				) }
 
-				{ ! this.state.calendarHelpIsVisible && (
-					<div className="components-datetime__buttons">
-						{ currentDate && (
-							<Button
-								className="components-datetime__date-reset-button"
-								isLink
-								onClick={ () => onChange( null ) }
-							>
-								{ __( 'Reset' ) }
-							</Button>
-						) }
+				<div className="components-datetime__buttons">
+					{ ! this.state.calendarHelpIsVisible && currentDate && (
 						<Button
-							className="components-datetime__date-help-button"
+							className="components-datetime__date-reset-button"
 							isLink
-							onClick={ this.onClickDescriptionToggle }
+							onClick={ () => onChange( null ) }
 						>
-							{ __( 'Calendar Help' ) }
+							{ __( 'Reset' ) }
 						</Button>
-					</div>
-				) }
+					) }
+					<Button
+						className="components-datetime__date-help-toggle"
+						isLink
+						onClick={ this.onClickDescriptionToggle }
+					>
+						{ this.state.calendarHelpIsVisible
+							? __( 'Close' )
+							: __( 'Calendar Help' ) }
+					</Button>
+				</div>
 			</div>
 		);
 	}
diff --git a/packages/components/src/date-time/style.scss b/packages/components/src/date-time/style.scss
index 1ee2116cb5eb32..71f367ba8eeb7a 100644
--- a/packages/components/src/date-time/style.scss
+++ b/packages/components/src/date-time/style.scss
@@ -20,7 +20,7 @@
 		justify-content: space-between;
 	}
 
-	.components-datetime__date-help-button {
+	.components-datetime__date-help-toggle {
 		display: block;
 		margin-left: auto;
 	}
@@ -108,8 +108,7 @@
 		}
 	}
 
-	&.is-description-visible .DayPicker,
-	&.is-description-visible .components-datetime__date-help-button {
+	&.is-description-visible .DayPicker {
 		visibility: hidden;
 	}
 }
