diff --git a/packages/block-editor/src/components/block-switcher/test/__snapshots__/index.js.snap b/packages/block-editor/src/components/block-switcher/test/__snapshots__/index.js.snap
index 5dc22def8ff0af..c8463783bda49e 100644
--- a/packages/block-editor/src/components/block-switcher/test/__snapshots__/index.js.snap
+++ b/packages/block-editor/src/components/block-switcher/test/__snapshots__/index.js.snap
@@ -2,7 +2,7 @@
 
 exports[`BlockSwitcher should render disabled block switcher with multi block of different types when no transforms 1`] = `
 <ToolbarGroup>
-  <ToolbarButton
+  <ForwardRef(ToolbarButton)
     className="block-editor-block-switcher__no-switcher-icon"
     disabled={true}
     icon={
diff --git a/packages/block-editor/src/components/block-switcher/test/index.js b/packages/block-editor/src/components/block-switcher/test/index.js
index 5bef2c5a52ff45..824bf47d882d17 100644
--- a/packages/block-editor/src/components/block-switcher/test/index.js
+++ b/packages/block-editor/src/components/block-switcher/test/index.js
@@ -8,6 +8,7 @@ import { shallow } from 'enzyme';
  */
 import { registerBlockType, unregisterBlockType } from '@wordpress/blocks';
 import { DOWN } from '@wordpress/keycodes';
+import { ToolbarButton } from '@wordpress/components';
 
 /**
  * Internal dependencies
@@ -186,7 +187,7 @@ describe( 'BlockSwitcher', () => {
 						isOpen: false,
 					} )
 				);
-				const iconButtonClosed = toggleClosed.find( 'ToolbarButton' );
+				const iconButtonClosed = toggleClosed.find( ToolbarButton );
 
 				iconButtonClosed.simulate( 'keydown', mockKeyDown );
 
@@ -200,7 +201,7 @@ describe( 'BlockSwitcher', () => {
 						isOpen: true,
 					} )
 				);
-				const iconButtonOpen = toggleOpen.find( 'ToolbarButton' );
+				const iconButtonOpen = toggleOpen.find( ToolbarButton );
 
 				iconButtonOpen.simulate( 'keydown', mockKeyDown );
 
diff --git a/packages/components/src/toolbar-button/index.js b/packages/components/src/toolbar-button/index.js
index e0e79e755718fd..f449ecdd2d523c 100644
--- a/packages/components/src/toolbar-button/index.js
+++ b/packages/components/src/toolbar-button/index.js
@@ -2,12 +2,10 @@
  * External dependencies
  */
 import classnames from 'classnames';
-
 /**
  * WordPress dependencies
  */
-import { useContext } from '@wordpress/element';
-
+import { useContext, forwardRef } from '@wordpress/element';
 /**
  * Internal dependencies
  */
@@ -16,16 +14,19 @@ import ToolbarItem from '../toolbar-item';
 import ToolbarContext from '../toolbar-context';
 import ToolbarButtonContainer from './toolbar-button-container';
 
-function ToolbarButton( {
-	containerClassName,
-	className,
-	extraProps,
-	children,
-	title,
-	isActive,
-	isDisabled,
-	...props
-} ) {
+function ToolbarButton(
+	{
+		containerClassName,
+		className,
+		extraProps,
+		children,
+		title,
+		isActive,
+		isDisabled,
+		...props
+	},
+	ref
+) {
 	const accessibleToolbarState = useContext( ToolbarContext );
 
 	if ( ! accessibleToolbarState ) {
@@ -34,6 +35,7 @@ function ToolbarButton( {
 		return (
 			<ToolbarButtonContainer className={ containerClassName }>
 				<Button
+					ref={ ref }
 					icon={ props.icon }
 					label={ title }
 					shortcut={ props.shortcut }
@@ -68,6 +70,7 @@ function ToolbarButton( {
 			className={ classnames( 'components-toolbar-button', className ) }
 			{ ...extraProps }
 			{ ...props }
+			ref={ ref }
 		>
 			{ ( toolbarItemProps ) => (
 				<Button
@@ -83,4 +86,4 @@ function ToolbarButton( {
 	);
 }
 
-export default ToolbarButton;
+export default forwardRef( ToolbarButton );
