diff --git a/packages/block-library/src/file/block.json b/packages/block-library/src/file/block.json
index c19ba5f7143007..230942f76a6c36 100644
--- a/packages/block-library/src/file/block.json
+++ b/packages/block-library/src/file/block.json
@@ -1,4 +1,5 @@
 {
+	"apiVersion": 2,
 	"name": "core/file",
 	"category": "media",
 	"attributes": {
diff --git a/packages/block-library/src/file/edit.js b/packages/block-library/src/file/edit.js
index 0eded60708117e..0b645fb9bcb241 100644
--- a/packages/block-library/src/file/edit.js
+++ b/packages/block-library/src/file/edit.js
@@ -7,17 +7,21 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { getBlobByURL, isBlobURL, revokeBlobURL } from '@wordpress/blob';
-import { Animate, ClipboardButton, withNotices } from '@wordpress/components';
-import { compose } from '@wordpress/compose';
-import { withSelect } from '@wordpress/data';
+import {
+	__unstableUseAnimate as useAnimate,
+	ClipboardButton,
+	withNotices,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
 import {
 	BlockControls,
 	BlockIcon,
 	MediaPlaceholder,
 	MediaReplaceFlow,
 	RichText,
+	useBlockProps,
 } from '@wordpress/block-editor';
-import { Component } from '@wordpress/element';
+import { useEffect, useState } from '@wordpress/element';
 import { __, _x } from '@wordpress/i18n';
 import { file as icon } from '@wordpress/icons';
 
@@ -26,46 +30,44 @@ import { file as icon } from '@wordpress/icons';
  */
 import FileBlockInspector from './inspector';
 
-class FileEdit extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.onSelectFile = this.onSelectFile.bind( this );
-		this.confirmCopyURL = this.confirmCopyURL.bind( this );
-		this.resetCopyConfirmation = this.resetCopyConfirmation.bind( this );
-		this.changeLinkDestinationOption = this.changeLinkDestinationOption.bind(
-			this
-		);
-		this.changeOpenInNewWindow = this.changeOpenInNewWindow.bind( this );
-		this.changeShowDownloadButton = this.changeShowDownloadButton.bind(
-			this
-		);
-		this.onUploadError = this.onUploadError.bind( this );
-
-		this.state = {
-			hasError: false,
-			showCopyConfirmation: false,
-		};
-	}
-
-	componentDidMount() {
-		const {
-			attributes,
-			mediaUpload,
-			noticeOperations,
-			setAttributes,
-		} = this.props;
-		const { downloadButtonText, href } = attributes;
-
+function FileEdit( {
+	isSelected,
+	attributes,
+	setAttributes,
+	noticeUI,
+	noticeOperations,
+} ) {
+	const {
+		id,
+		fileName,
+		href,
+		textLinkHref,
+		textLinkTarget,
+		showDownloadButton,
+		downloadButtonText,
+	} = attributes;
+	const [ hasError, setHasError ] = useState( false );
+	const [ showCopyConfirmation, setShowCopyConfirmation ] = useState( false );
+	const { media, mediaUpload } = useSelect(
+		( select ) => ( {
+			media:
+				id === undefined ? undefined : select( 'core' ).getMedia( id ),
+			mediaUpload: select( 'core/block-editor' ).getSettings()
+				.mediaUpload,
+		} ),
+		[ id ]
+	);
+
+	useEffect( () => {
 		// Upload a file drag-and-dropped into the editor
 		if ( isBlobURL( href ) ) {
 			const file = getBlobByURL( href );
 
 			mediaUpload( {
 				filesList: [ file ],
-				onFileChange: ( [ media ] ) => this.onSelectFile( media ),
+				onFileChange: ( [ newMedia ] ) => onSelectFile( newMedia ),
 				onError: ( message ) => {
-					this.setState( { hasError: true } );
+					setHasError( true );
 					noticeOperations.createErrorNotice( message );
 				},
 			} );
@@ -78,80 +80,67 @@ class FileEdit extends Component {
 				downloadButtonText: _x( 'Download', 'button label' ),
 			} );
 		}
-	}
+	}, [] );
 
-	componentDidUpdate( prevProps ) {
-		// Reset copy confirmation state when block is deselected
-		if ( prevProps.isSelected && ! this.props.isSelected ) {
-			this.setState( { showCopyConfirmation: false } );
-		}
-	}
+	useEffect( () => {
+		setShowCopyConfirmation( false );
+	}, [ isSelected ] );
 
-	onSelectFile( media ) {
-		if ( media && media.url ) {
-			this.setState( { hasError: false } );
-			this.props.setAttributes( {
-				href: media.url,
-				fileName: media.title,
-				textLinkHref: media.url,
-				id: media.id,
+	function onSelectFile( newMedia ) {
+		if ( newMedia && newMedia.url ) {
+			setHasError( false );
+			setAttributes( {
+				href: newMedia.url,
+				fileName: newMedia.title,
+				textLinkHref: newMedia.url,
+				id: newMedia.id,
 			} );
 		}
 	}
 
-	onUploadError( message ) {
-		const { noticeOperations } = this.props;
-		this.setState( { hasError: true } );
+	function onUploadError( message ) {
+		setHasError( true );
 		noticeOperations.removeAllNotices();
 		noticeOperations.createErrorNotice( message );
 	}
 
-	confirmCopyURL() {
-		this.setState( { showCopyConfirmation: true } );
+	function confirmCopyURL() {
+		setShowCopyConfirmation( true );
 	}
 
-	resetCopyConfirmation() {
-		this.setState( { showCopyConfirmation: false } );
+	function resetCopyConfirmation() {
+		setShowCopyConfirmation( false );
 	}
 
-	changeLinkDestinationOption( newHref ) {
+	function changeLinkDestinationOption( newHref ) {
 		// Choose Media File or Attachment Page (when file is in Media Library)
-		this.props.setAttributes( { textLinkHref: newHref } );
+		setAttributes( { textLinkHref: newHref } );
 	}
 
-	changeOpenInNewWindow( newValue ) {
-		this.props.setAttributes( {
+	function changeOpenInNewWindow( newValue ) {
+		setAttributes( {
 			textLinkTarget: newValue ? '_blank' : false,
 		} );
 	}
 
-	changeShowDownloadButton( newValue ) {
-		this.props.setAttributes( { showDownloadButton: newValue } );
+	function changeShowDownloadButton( newValue ) {
+		setAttributes( { showDownloadButton: newValue } );
 	}
 
-	render() {
-		const {
-			className,
-			isSelected,
-			attributes,
-			setAttributes,
-			noticeUI,
-			media,
-		} = this.props;
-		const {
-			id,
-			fileName,
-			href,
-			textLinkHref,
-			textLinkTarget,
-			showDownloadButton,
-			downloadButtonText,
-		} = attributes;
-		const { hasError, showCopyConfirmation } = this.state;
-		const attachmentPage = media && media.link;
+	const attachmentPage = media && media.link;
 
-		if ( ! href || hasError ) {
-			return (
+	const blockProps = useBlockProps( {
+		className: classnames(
+			useAnimate( { type: isBlobURL( href ) ? 'loading' : null } ),
+			{
+				'is-transient': isBlobURL( href ),
+			}
+		),
+	} );
+
+	if ( ! href || hasError ) {
+		return (
+			<div { ...blockProps }>
 				<MediaPlaceholder
 					icon={ <BlockIcon icon={ icon } /> }
 					labels={ {
@@ -160,118 +149,88 @@ class FileEdit extends Component {
 							'Upload a file or pick one from your media library.'
 						),
 					} }
-					onSelect={ this.onSelectFile }
+					onSelect={ onSelectFile }
 					notices={ noticeUI }
-					onError={ this.onUploadError }
+					onError={ onUploadError }
 					accept="*"
 				/>
-			);
-		}
-
-		const classes = classnames( className, {
-			'is-transient': isBlobURL( href ),
-		} );
+			</div>
+		);
+	}
 
-		return (
-			<>
-				<FileBlockInspector
-					hrefs={ { href, textLinkHref, attachmentPage } }
-					{ ...{
-						openInNewWindow: !! textLinkTarget,
-						showDownloadButton,
-						changeLinkDestinationOption: this
-							.changeLinkDestinationOption,
-						changeOpenInNewWindow: this.changeOpenInNewWindow,
-						changeShowDownloadButton: this.changeShowDownloadButton,
-					} }
+	return (
+		<>
+			<FileBlockInspector
+				hrefs={ { href, textLinkHref, attachmentPage } }
+				{ ...{
+					openInNewWindow: !! textLinkTarget,
+					showDownloadButton,
+					changeLinkDestinationOption,
+					changeOpenInNewWindow,
+					changeShowDownloadButton,
+				} }
+			/>
+			<BlockControls>
+				<MediaReplaceFlow
+					mediaId={ id }
+					mediaURL={ href }
+					accept="*"
+					onSelect={ onSelectFile }
+					onError={ onUploadError }
 				/>
-				<BlockControls>
-					<MediaReplaceFlow
-						mediaId={ id }
-						mediaURL={ href }
-						accept="*"
-						onSelect={ this.onSelectFile }
-						onError={ this.onUploadError }
-					/>
-				</BlockControls>
-				<Animate type={ isBlobURL( href ) ? 'loading' : null }>
-					{ ( { className: animateClassName } ) => (
+			</BlockControls>
+			<div { ...blockProps }>
+				<div className={ 'wp-block-file__content-wrapper' }>
+					<div className="wp-block-file__textlink">
+						<RichText
+							tagName="div" // must be block-level or else cursor disappears
+							value={ fileName }
+							placeholder={ __( 'Write file name…' ) }
+							withoutInteractiveFormatting
+							onChange={ ( text ) =>
+								setAttributes( { fileName: text } )
+							}
+						/>
+					</div>
+					{ showDownloadButton && (
 						<div
-							className={ classnames(
-								classes,
-								animateClassName
-							) }
+							className={
+								'wp-block-file__button-richtext-wrapper'
+							}
 						>
-							<div className={ 'wp-block-file__content-wrapper' }>
-								<div className="wp-block-file__textlink">
-									<RichText
-										tagName="div" // must be block-level or else cursor disappears
-										value={ fileName }
-										placeholder={ __( 'Write file name…' ) }
-										withoutInteractiveFormatting
-										onChange={ ( text ) =>
-											setAttributes( { fileName: text } )
-										}
-									/>
-								</div>
-								{ showDownloadButton && (
-									<div
-										className={
-											'wp-block-file__button-richtext-wrapper'
-										}
-									>
-										{ /* Using RichText here instead of PlainText so that it can be styled like a button */ }
-										<RichText
-											tagName="div" // must be block-level or else cursor disappears
-											className={
-												'wp-block-file__button'
-											}
-											value={ downloadButtonText }
-											withoutInteractiveFormatting
-											placeholder={ __( 'Add text…' ) }
-											onChange={ ( text ) =>
-												setAttributes( {
-													downloadButtonText: text,
-												} )
-											}
-										/>
-									</div>
-								) }
-							</div>
-							{ isSelected && (
-								<ClipboardButton
-									isSecondary
-									text={ href }
-									className={
-										'wp-block-file__copy-url-button'
-									}
-									onCopy={ this.confirmCopyURL }
-									onFinishCopy={ this.resetCopyConfirmation }
-									disabled={ isBlobURL( href ) }
-								>
-									{ showCopyConfirmation
-										? __( 'Copied!' )
-										: __( 'Copy URL' ) }
-								</ClipboardButton>
-							) }
+							{ /* Using RichText here instead of PlainText so that it can be styled like a button */ }
+							<RichText
+								tagName="div" // must be block-level or else cursor disappears
+								className={ 'wp-block-file__button' }
+								value={ downloadButtonText }
+								withoutInteractiveFormatting
+								placeholder={ __( 'Add text…' ) }
+								onChange={ ( text ) =>
+									setAttributes( {
+										downloadButtonText: text,
+									} )
+								}
+							/>
 						</div>
 					) }
-				</Animate>
-			</>
-		);
-	}
+				</div>
+				{ isSelected && (
+					<ClipboardButton
+						isSecondary
+						text={ href }
+						className={ 'wp-block-file__copy-url-button' }
+						onCopy={ confirmCopyURL }
+						onFinishCopy={ resetCopyConfirmation }
+						disabled={ isBlobURL( href ) }
+					>
+						{ showCopyConfirmation
+							? __( 'Copied!' )
+							: __( 'Copy URL' ) }
+					</ClipboardButton>
+				) }
+			</div>
+		</>
+	);
 }
 
-export default compose( [
-	withSelect( ( select, props ) => {
-		const { getMedia } = select( 'core' );
-		const { getSettings } = select( 'core/block-editor' );
-		const { mediaUpload } = getSettings();
-		const { id } = props.attributes;
-		return {
-			media: id === undefined ? undefined : getMedia( id ),
-			mediaUpload,
-		};
-	} ),
-	withNotices,
-] )( FileEdit );
+export default withNotices( FileEdit );
diff --git a/packages/block-library/src/file/save.js b/packages/block-library/src/file/save.js
index 25102dfd0f92d0..b578300197a039 100644
--- a/packages/block-library/src/file/save.js
+++ b/packages/block-library/src/file/save.js
@@ -1,7 +1,7 @@
 /**
  * WordPress dependencies
  */
-import { RichText } from '@wordpress/block-editor';
+import { RichText, useBlockProps } from '@wordpress/block-editor';
 
 export default function save( { attributes } ) {
 	const {
@@ -15,7 +15,7 @@ export default function save( { attributes } ) {
 
 	return (
 		href && (
-			<div>
+			<div { ...useBlockProps.save() }>
 				{ ! RichText.isEmpty( fileName ) && (
 					<a
 						href={ textLinkHref }
diff --git a/packages/components/src/animate/index.js b/packages/components/src/animate/index.js
index a7cfc8d8cfa7d2..2eeed8e7be1306 100644
--- a/packages/components/src/animate/index.js
+++ b/packages/components/src/animate/index.js
@@ -3,37 +3,33 @@
  */
 import classnames from 'classnames';
 
-function Animate( { type, options = {}, children } ) {
+function getDefaultOrigin( type ) {
+	return type === 'appear' ? 'top' : 'left';
+}
+
+export function useAnimate( { type, origin = getDefaultOrigin( type ) } ) {
 	if ( type === 'appear' ) {
-		const { origin = 'top' } = options;
 		const [ yAxis, xAxis = 'center' ] = origin.split( ' ' );
-
-		return children( {
-			className: classnames( 'components-animate__appear', {
-				[ 'is-from-' + xAxis ]: xAxis !== 'center',
-				[ 'is-from-' + yAxis ]: yAxis !== 'middle',
-			} ),
+		return classnames( 'components-animate__appear', {
+			[ 'is-from-' + xAxis ]: xAxis !== 'center',
+			[ 'is-from-' + yAxis ]: yAxis !== 'middle',
 		} );
 	}
 
 	if ( type === 'slide-in' ) {
-		const { origin = 'left' } = options;
-
-		return children( {
-			className: classnames(
-				'components-animate__slide-in',
-				'is-from-' + origin
-			),
-		} );
+		return classnames(
+			'components-animate__slide-in',
+			'is-from-' + origin
+		);
 	}
 
 	if ( type === 'loading' ) {
-		return children( {
-			className: classnames( 'components-animate__loading' ),
-		} );
+		return classnames( 'components-animate__loading' );
 	}
-
-	return children( {} );
 }
 
-export default Animate;
+export default function Animate( { type, options = {}, children } ) {
+	return children( {
+		className: useAnimate( { type, ...options } ),
+	} );
+}
diff --git a/packages/components/src/index.js b/packages/components/src/index.js
index a629eb4d8efc5d..00270b85823a61 100644
--- a/packages/components/src/index.js
+++ b/packages/components/src/index.js
@@ -12,7 +12,10 @@ export {
 
 // Components
 export { default as __experimentalAlignmentMatrixControl } from './alignment-matrix-control';
-export { default as Animate } from './animate';
+export {
+	default as Animate,
+	useAnimate as __unstableUseAnimate,
+} from './animate';
 export { default as AnglePickerControl } from './angle-picker-control';
 export { default as Autocomplete } from './autocomplete';
 export { default as BaseControl } from './base-control';
