diff --git a/packages/components/src/range-control/index.js b/packages/components/src/range-control/index.js
index 7073862d413000..fb64ec8ea9b6cc 100644
--- a/packages/components/src/range-control/index.js
+++ b/packages/components/src/range-control/index.js
@@ -265,6 +265,7 @@ function RangeControl(
 						<SimpleTooltip
 							className="components-range-control__tooltip"
 							inputRef={ inputRef }
+							tooltipPosition="bottom"
 							renderTooltipContent={ renderTooltipContent }
 							show={ isCurrentlyFocused || showTooltip }
 							style={ offsetStyle }
diff --git a/packages/components/src/range-control/styles/range-control-styles.js b/packages/components/src/range-control/styles/range-control-styles.js
index 07d39dc1197a0a..4507107b9f7bbd 100644
--- a/packages/components/src/range-control/styles/range-control-styles.js
+++ b/packages/components/src/range-control/styles/range-control-styles.js
@@ -249,16 +249,16 @@ const tooltipShow = ( { show } ) => {
 };
 
 const tooltipPosition = ( { position } ) => {
-	const isTop = position === 'top';
+	const isBottom = position === 'bottom';
 
-	if ( isTop ) {
+	if ( isBottom ) {
 		return css`
-			top: -80%;
+			bottom: -80%;
 		`;
 	}
 
 	return css`
-		bottom: -80%;
+		top: -80%;
 	`;
 };
 
diff --git a/packages/components/src/range-control/tooltip.js b/packages/components/src/range-control/tooltip.js
index 8d350c31ceb7b0..07c49e927094ae 100644
--- a/packages/components/src/range-control/tooltip.js
+++ b/packages/components/src/range-control/tooltip.js
@@ -14,12 +14,10 @@ import { useCallback, useEffect, useState } from '@wordpress/element';
  */
 import { Tooltip } from './styles/range-control-styles';
 
-const TOOLTIP_OFFSET_HEIGHT = 32;
-
 export default function SimpleTooltip( {
 	className,
 	inputRef,
-	position: positionProp = 'auto',
+	tooltipPosition,
 	show = false,
 	style = {},
 	value = 0,
@@ -27,7 +25,7 @@ export default function SimpleTooltip( {
 	zIndex = 100,
 	...restProps
 } ) {
-	const position = useTooltipPosition( { inputRef, position: positionProp } );
+	const position = useTooltipPosition( { inputRef, tooltipPosition } );
 	const classes = classnames( 'components-simple-tooltip', className );
 	const styles = {
 		...style,
@@ -49,33 +47,24 @@ export default function SimpleTooltip( {
 	);
 }
 
-function useTooltipPosition( { inputRef, position: positionProp } ) {
-	const [ position, setPosition ] = useState( 'top' );
+function useTooltipPosition( { inputRef, tooltipPosition } ) {
+	const [ position, setPosition ] = useState();
 
-	const calculatePosition = useCallback( () => {
+	const setTooltipPosition = useCallback( () => {
 		if ( inputRef && inputRef.current ) {
-			let nextPosition = positionProp;
-
-			if ( positionProp === 'auto' ) {
-				const { top } = inputRef.current.getBoundingClientRect();
-				const isOffscreenTop = top - TOOLTIP_OFFSET_HEIGHT < 0;
-
-				nextPosition = isOffscreenTop ? 'bottom' : 'top';
-			}
-
-			setPosition( nextPosition );
+			setPosition( tooltipPosition );
 		}
-	}, [ positionProp ] );
+	}, [ tooltipPosition ] );
 
 	useEffect( () => {
-		calculatePosition();
-	}, [ calculatePosition ] );
+		setTooltipPosition();
+	}, [ setTooltipPosition ] );
 
 	useEffect( () => {
-		window.addEventListener( 'resize', calculatePosition );
+		window.addEventListener( 'resize', setTooltipPosition );
 
 		return () => {
-			window.removeEventListener( 'resize', calculatePosition );
+			window.removeEventListener( 'resize', setTooltipPosition );
 		};
 	} );
 
