diff --git a/docs/manifest.json b/docs/manifest.json
index 5d6a6d82a77db6..88376c838dda21 100644
--- a/docs/manifest.json
+++ b/docs/manifest.json
@@ -1223,6 +1223,12 @@
 		"markdown_source": "../packages/components/src/text-highlight/README.md",
 		"parent": "components"
 	},
+	{
+		"title": "Next",
+		"slug": "next",
+		"markdown_source": "../packages/components/src/text/next/README.md",
+		"parent": "components"
+	},
 	{
 		"title": "Text",
 		"slug": "text",
diff --git a/package-lock.json b/package-lock.json
index 09e4665d928069..c832c264bceb38 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -11950,6 +11950,7 @@
 				"dom-scroll-into-view": "^1.2.1",
 				"downshift": "^6.0.15",
 				"gradient-parser": "^0.1.5",
+				"highlight-words-core": "^1.2.2",
 				"lodash": "^4.17.19",
 				"memize": "^1.1.0",
 				"moment": "^2.22.1",
diff --git a/packages/components/package.json b/packages/components/package.json
index 3daf1361c7f807..89c40661ab3329 100644
--- a/packages/components/package.json
+++ b/packages/components/package.json
@@ -53,6 +53,7 @@
 		"dom-scroll-into-view": "^1.2.1",
 		"downshift": "^6.0.15",
 		"gradient-parser": "^0.1.5",
+		"highlight-words-core": "^1.2.2",
 		"lodash": "^4.17.19",
 		"memize": "^1.1.0",
 		"moment": "^2.22.1",
diff --git a/packages/components/src/__next/context/with-next.js b/packages/components/src/__next/context/with-next.js
index 24e65e54dfce96..d52ed5bb456ec9 100644
--- a/packages/components/src/__next/context/with-next.js
+++ b/packages/components/src/__next/context/with-next.js
@@ -2,10 +2,6 @@
  * External dependencies
  */
 import { contextConnect, useContextSystem } from '@wp-g2/components';
-/**
- * WordPress dependencies
- */
-import { forwardRef } from '@wordpress/element';
 
 export function withNext(
 	CurrentComponent = () => null,
@@ -31,5 +27,5 @@ export function withNext(
 		return contextConnect( WrappedComponent, namespace );
 	}
 
-	return forwardRef( CurrentComponent );
+	return CurrentComponent;
 }
diff --git a/packages/components/src/text/index.js b/packages/components/src/text/index.js
index e502e39a59c49a..fb815c5571d84b 100644
--- a/packages/components/src/text/index.js
+++ b/packages/components/src/text/index.js
@@ -7,6 +7,7 @@ import styled from '@emotion/styled';
  * Internal dependencies
  */
 import { text } from './styles/text-mixins';
+import { withNextComponent } from './next';
 
 const Text = styled.p(
 	`
@@ -16,4 +17,4 @@ const Text = styled.p(
 	text
 );
 
-export default Text;
+export default withNextComponent( Text );
diff --git a/packages/components/src/text/next/README.md b/packages/components/src/text/next/README.md
new file mode 100644
index 00000000000000..52471f6de9549c
--- /dev/null
+++ b/packages/components/src/text/next/README.md
@@ -0,0 +1,235 @@
+# Text
+
+`Text` is a core component that renders text in the library, using the library's typography system.
+
+## Usage
+
+`Text` can be used to render any text-content, like an HTML `p` or `span`.
+
+```jsx live
+import { Text } from '@wp-g2/components';
+
+function Example() {
+	return <Text>Where the north wind meets the sea</Text>;
+}
+```
+
+## Props
+
+##### adjustLineHeightForInnerControls
+
+**Type**: `boolean`,`"large"`,`"medium"`,`"small"`,`"xSmall"`
+
+Automatically calculate the appropriate line-height value for contents that render text and Control elements (e.g. `TextInput`).
+
+```jsx live
+import { Text, TextInput } from '@wp-g2/components';
+
+function Example() {
+	return (
+		<Text adjustLineHeightForInnerControls>
+			Where the north wind meets the <TextInput value="sea..." />
+		</Text>
+	);
+}
+```
+
+##### align
+
+**Type**: `CSS['textAlign']`
+
+Adjusts the text alignment.
+
+```jsx live
+import { Text } from '@wp-g2/components';
+
+function Example() {
+	return (
+		<Text align="center" isBlock>
+			Where the north wind meets the sea...
+		</Text>
+	);
+}
+```
+
+##### color
+
+**Type**: `CSS['color']`
+
+Adjusts the text color.
+
+##### display
+
+**Type**: `CSS['display']`
+
+Adjusts the CSS display.
+
+##### ellipsis
+
+**Type**: `string`
+
+The ellipsis string when `truncate` is set.
+
+##### ellipsizeMode
+
+**Type**: `"auto"`,`"head"`,`"tail"`,`"middle"`
+
+Determines where to truncate. For example, we can truncate text right in the middle. To do this, we need to set `ellipsizeMode` to `middle` and a text `limit`.
+
+-   `auto`: Trims content at the end automatically without a `limit`.
+-   `head`: Trims content at the beginning. Requires a `limit`.
+-   `middle`: Trims content in the middle. Requires a `limit`.
+-   `tail`: Trims content at the end. Requires a `limit`.
+
+##### highlightCaseSensitive
+
+**Type**: `boolean`
+
+Escape characters in `highlightWords` which are meaningful in regular expressions.
+
+##### highlightEscape
+
+**Type**: `boolean`
+
+Determines if `highlightWords` should be case sensitive.
+
+##### highlightSanitize
+
+**Type**: `boolean`
+
+Array of search words. String search terms are automatically cast to RegExps unless `highlightEscape` is true.
+
+##### highlightWords
+
+**Type**: `any[]`
+
+Letters or words within `Text` can be highlighted using `highlightWords`.
+
+```jsx live
+import { Text } from '@wp-g2/components';
+
+function Example() {
+	return (
+		<Text highlightWords={['the']}>
+			Where the north wind meets the sea, there's a river full of memory.
+			Sleep, my darling, safe and sound, for in this river all is found.
+			In her waters, deep and true, lay the answers and a path for you.
+			Dive down deep into her sound, but not too far or you'll be drowned
+		</Text>
+	);
+}
+```
+
+##### isBlock
+
+**Type**: `boolean`
+
+Sets `Text` to have `display: block`.
+
+##### isDestructive
+
+**Type**: `boolean`
+
+Renders a destructive color.
+
+##### limit
+
+**Type**: `number`
+
+Determines the max characters when `truncate` is set.
+
+##### lineHeight
+
+**Type**: `CSS['lineHeight']`
+
+Adjusts all text line-height based on the typography system.
+
+##### numberOfLines
+
+**Type**: `number`
+
+Clamps the text content to the specifiec `numberOfLines`, adding the `ellipsis` at the end.
+
+##### optimizeReadabilityFor
+
+**Type**: `CSS['color']`
+
+The `Text` color can be adapted to a background color for optimal readability. `optimizeReadabilityFor` can accept CSS variables, in addition to standard CSS color values (e.g. Hex, RGB, HSL, etc...).
+
+```jsx live
+import { Text, View } from '@wp-g2/components';
+
+function Example() {
+	const backgroundColor = 'blue';
+
+	return (
+		<View css={{ backgroundColor }}>
+			<Text optimizeReadabilityFor={backgroundColor}>
+				Where the north wind meets the sea, there's a river full of
+				memory.
+			</Text>
+		</View>
+	);
+}
+```
+
+##### size
+
+**Type**: `CSS['fontSize']`,`TextSize`
+
+Adjusts text size based on the typography system. `Text` can render a wide range of font sizes, which are automatically calculated and adapted to the typography system. The `size` value can be a system preset, a `number`, or a custom unit value (`string`) such as `30em`.
+
+```jsx live
+import { Text } from '@wp-g2/components';
+
+function Example() {
+	return <Text size="largeTitle">Where the north wind meets the sea...</Text>;
+}
+```
+
+##### truncate
+
+**Type**: `boolean`
+
+Enables text truncation. When `truncate` is set,we are able to truncate the long text in a variety of ways.
+
+```jsx live
+import { Text } from '@wp-g2/components';
+
+function Example() {
+	return (
+		<Text truncate>
+			Where the north wind meets the sea, there's a river full of memory.
+			Sleep, my darling, safe and sound, for in this river all is found.
+			In her waters, deep and true, lay the answers and a path for you.
+			Dive down deep into her sound, but not too far or you'll be drowned
+		</Text>
+	);
+}
+```
+
+##### upperCase
+
+**Type**: `boolean`
+
+Uppercases the text content.
+
+##### variant
+
+**Type**: `"muted"`
+
+Adjusts style variation of the text.
+
+```jsx live
+import { Text } from '@wp-g2/components';
+
+function Example() {
+	return <Text variant="muted">Where the north wind meets the sea...</Text>;
+}
+```
+
+##### weight
+
+**Type**: `CSS['fontWeight']`,`TextWeight`
+
+Adjusts font-weight of the text.
diff --git a/packages/components/src/text/next/index.js b/packages/components/src/text/next/index.js
new file mode 100644
index 00000000000000..54092b3f5155c0
--- /dev/null
+++ b/packages/components/src/text/next/index.js
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import { __unstableWithNext as withNext } from '../../__next/context';
+import NextText from './text';
+import { text } from '../styles/text-mixins';
+
+const Text = process.env.COMPONENT_SYSTEM_PHASE === 1 ? NextText : undefined;
+
+export const adapter = ( { as, variant, ...restProps } ) => ( {
+	// as has not changed
+	as,
+	// luckily `text` just returns an emotion CSS object, so we can pass the styles from that directly to the handy `css` prop
+	css: text( { variant } ).styles,
+	// mostly className
+	...restProps,
+} );
+
+export function withNextComponent( Current ) {
+	return withNext( Current, Text, 'WPComponentsText', adapter );
+}
diff --git a/packages/components/src/text/next/stories/index.js b/packages/components/src/text/next/stories/index.js
new file mode 100644
index 00000000000000..66e02a6ebfba5e
--- /dev/null
+++ b/packages/components/src/text/next/stories/index.js
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import { boolean } from '@storybook/addon-knobs';
+
+/**
+ * Internal dependencies
+ */
+import Text from '../text';
+import OldText from '../../index';
+import { adapter } from '../index';
+
+export default {
+	component: Text,
+	title: 'Components/G2/Text',
+};
+
+export const _default = () => {
+	return <Text>Hello</Text>;
+};
+
+export const truncate = () => {
+	return (
+		<Text numberOfLines={ 2 } truncate>
+			Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut
+			facilisis dictum tortor, eu tincidunt justo scelerisque tincidunt.
+			Duis semper dui id augue malesuada, ut feugiat nisi aliquam.
+			Vestibulum venenatis diam sem, finibus dictum massa semper in. Nulla
+			facilisi. Nunc vulputate faucibus diam, in lobortis arcu ornare vel.
+			In dignissim nunc sed facilisis finibus. Etiam imperdiet mattis
+			arcu, sed rutrum sapien blandit gravida. Aenean sollicitudin neque
+			eget enim blandit, sit amet rutrum leo vehicula. Nunc malesuada
+			ultricies eros ut faucibus. Aliquam erat volutpat. Nulla nec feugiat
+			risus. Vivamus iaculis dui aliquet ante ultricies feugiat.
+			Vestibulum ante ipsum primis in faucibus orci luctus et ultrices
+			posuere cubilia curae; Vivamus nec pretium velit, sit amet
+			consectetur ante. Praesent porttitor ex eget fermentum mattis.
+		</Text>
+	);
+};
+
+export const highlight = () => {
+	return (
+		<Text highlightWords={ [ 'con' ] }>
+			Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut
+			facilisis dictum tortor, eu tincidunt justo scelerisque tincidunt.
+			Duis semper dui id augue malesuada, ut feugiat nisi aliquam.
+			Vestibulum venenatis diam sem, finibus dictum massa semper in. Nulla
+			facilisi. Nunc vulputate faucibus diam, in lobortis arcu ornare vel.
+			In dignissim nunc sed facilisis finibus. Etiam imperdiet mattis
+			arcu, sed rutrum sapien blandit gravida. Aenean sollicitudin neque
+			eget enim blandit, sit amet rutrum leo vehicula. Nunc malesuada
+			ultricies eros ut faucibus. Aliquam erat volutpat. Nulla nec feugiat
+			risus. Vivamus iaculis dui aliquet ante ultricies feugiat.
+			Vestibulum ante ipsum primis in faucibus orci luctus et ultrices
+			posuere cubilia curae; Vivamus nec pretium velit, sit amet
+			consectetur ante. Praesent porttitor ex eget fermentum mattis.
+		</Text>
+	);
+};
+
+const Adapted = ( { doAdapt, ...props } ) =>
+	doAdapt ? <Text { ...adapter( props ) } /> : <OldText { ...props } />;
+
+export const adapted = () => {
+	const doAdapt = boolean( 'Use adapted component', false );
+	return (
+		<>
+			<Adapted variant="title.large" as="h1" doAdapt={ doAdapt }>
+				Title Large
+			</Adapted>
+			<Adapted variant="title.medium" as="h2" doAdapt={ doAdapt }>
+				Title Medium
+			</Adapted>
+			<Adapted variant="title.small" as="h3" doAdapt={ doAdapt }>
+				Title Small
+			</Adapted>
+			<Adapted variant="subtitle" doAdapt={ doAdapt }>
+				Subtitle
+			</Adapted>
+			<Adapted variant="subtitle.small" doAdapt={ doAdapt }>
+				Subtitle Small
+			</Adapted>
+			<Adapted variant="body" doAdapt={ doAdapt }>
+				Body
+			</Adapted>
+			<Adapted variant="body.small" doAdapt={ doAdapt }>
+				Body Small
+			</Adapted>
+			<Adapted variant="button" doAdapt={ doAdapt }>
+				Button
+			</Adapted>
+			<Adapted variant="caption" doAdapt={ doAdapt }>
+				Caption
+			</Adapted>
+			<Adapted variant="label" doAdapt={ doAdapt }>
+				Label
+			</Adapted>
+		</>
+	);
+};
diff --git a/packages/components/src/text/next/test/__snapshots__/index.js.snap b/packages/components/src/text/next/test/__snapshots__/index.js.snap
new file mode 100644
index 00000000000000..dd2f46c7d9c012
--- /dev/null
+++ b/packages/components/src/text/next/test/__snapshots__/index.js.snap
@@ -0,0 +1,99 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Text should render highlighted words with highlightCaseSensitive 1`] = `
+.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
+  box-sizing: border-box;
+  -moz-osx-font-smoothing: grayscale;
+  -webkit-font-smoothing: antialiased;
+  font-family: Inter,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",sans-serif;
+  font-family: var(--wp-g2-font-family);
+  font-size: 13px;
+  font-size: var(--wp-g2-font-size);
+  font-weight: normal;
+  font-weight: var(--wp-g2-font-weight);
+  margin: 0;
+  color: #1e1e1e;
+  color: var(--wp-g2-color-text);
+  line-height: 1.2;
+  line-height: var(--wp-g2-font-line-height-base);
+  font-size: calc(1 * 13px);
+  font-size: calc(1 * var(--wp-g2-font-size));
+  font-weight: normal;
+  font-weight: var(--wp-g2-font-weight);
+}
+
+@media (prefers-reduced-motion) {
+  .emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
+    -webkit-transition: none !important;
+    transition: none !important;
+  }
+}
+
+[data-system-ui-reduced-motion-mode="true"] .emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
+  -webkit-transition: none !important;
+  transition: none !important;
+}
+
+.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 mark {
+  background: rgba(255, 185, 0, 0.7);
+  background: var(--wp-g2-yellow-rgba-70);
+  border-radius: 2px;
+  box-shadow: 0 0 0 1px rgba( 0,0,0,0.05 ) inset, 0 -1px 0 rgba( 0,0,0,0.1 ) inset;
+}
+
+<span
+  class="components-truncate wp-components-truncate components-text wp-components-text emotion-0"
+  data-g2-c16t="true"
+  data-g2-component="Text"
+>
+  <span
+    class=""
+    data-g2-component="Text"
+  >
+    Some people are worth melting for.
+  </span>
+</span>
+`;
+
+exports[`Text snapshot tests should render correctly 1`] = `
+.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
+  box-sizing: border-box;
+  -moz-osx-font-smoothing: grayscale;
+  -webkit-font-smoothing: antialiased;
+  font-family: Inter,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",sans-serif;
+  font-family: var(--wp-g2-font-family);
+  font-size: 13px;
+  font-size: var(--wp-g2-font-size);
+  font-weight: normal;
+  font-weight: var(--wp-g2-font-weight);
+  margin: 0;
+  color: #1e1e1e;
+  color: var(--wp-g2-color-text);
+  line-height: 1.2;
+  line-height: var(--wp-g2-font-line-height-base);
+  font-size: calc(1 * 13px);
+  font-size: calc(1 * var(--wp-g2-font-size));
+  font-weight: normal;
+  font-weight: var(--wp-g2-font-weight);
+}
+
+@media (prefers-reduced-motion) {
+  .emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
+    -webkit-transition: none !important;
+    transition: none !important;
+  }
+}
+
+[data-system-ui-reduced-motion-mode="true"] .emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0.emotion-0 {
+  -webkit-transition: none !important;
+  transition: none !important;
+}
+
+<span
+  class="components-truncate wp-components-truncate components-text wp-components-text emotion-0"
+  data-g2-c16t="true"
+  data-g2-component="Text"
+>
+  Some people are worth melting for.
+</span>
+`;
diff --git a/packages/components/src/text/next/test/index.js b/packages/components/src/text/next/test/index.js
new file mode 100644
index 00000000000000..31cb96ce992839
--- /dev/null
+++ b/packages/components/src/text/next/test/index.js
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+import { getFontSize, ui } from '@wp-g2/styles';
+
+/**
+ * Internal dependencies
+ */
+import Text from '../text';
+
+describe( 'Text', () => {
+	describe( 'snapshot tests', () => {
+		test( 'should render correctly', () => {
+			const { container } = render(
+				<Text>Some people are worth melting for.</Text>
+			);
+			expect( container.firstChild ).toMatchSnapshot();
+		} );
+	} );
+
+	test( 'should render optimizeReadabilityFor', () => {
+		const { container } = render(
+			<Text optimizeReadabilityFor="blue">
+				Some people are worth melting for.
+			</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( {
+			color: ui.get( 'white' ),
+		} );
+	} );
+
+	test( 'should render truncate', () => {
+		const { container } = render(
+			<Text truncate>Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( {
+			textOverflow: 'ellipsis',
+		} );
+	} );
+
+	test( 'should render size', () => {
+		const { container } = render(
+			<Text size="title">Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( {
+			fontSize: getFontSize( 'title' ),
+		} );
+	} );
+
+	test( 'should render custom size', () => {
+		const { container } = render(
+			<Text size={ 15 }>Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( {
+			fontSize: getFontSize( 15 ),
+		} );
+	} );
+
+	test( 'should render variant', () => {
+		const { container } = render(
+			<Text variant="muted">Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( {
+			color: ui.get( 'colorTextMuted' ),
+		} );
+	} );
+
+	test( 'should render as another element', () => {
+		const { container } = render(
+			<Text as="div">Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild.nodeName ).toBe( 'DIV' );
+	} );
+
+	test( 'should render align', () => {
+		const { container } = render(
+			<Text align="center">Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( { textAlign: 'center' } );
+	} );
+
+	test( 'should render color', () => {
+		const { container } = render(
+			<Text color="orange">Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( { color: 'orange' } );
+	} );
+
+	test( 'should render display', () => {
+		const { container } = render(
+			<Text display="inline-flex">
+				Some people are worth melting for.
+			</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( {
+			display: 'inline-flex',
+		} );
+	} );
+
+	test( 'should render highlighted words', async () => {
+		const wrapper = render(
+			<Text highlightWords={ [ 'worth' ] }>
+				Some people are worth melting for.
+			</Text>
+		);
+		// It should have three child nodes: (Some people are )(worth)( melting for.)
+		expect( wrapper.container.firstChild.childNodes ).toHaveLength( 3 );
+		const worth = await wrapper.findAllByText( 'worth' );
+		expect( worth ).toHaveLength( 1 );
+		expect( worth[ 0 ].dataset.g2Component ).toBe( 'TextHighlight' );
+	} );
+
+	test( 'should render highlighted words with undefined passed', () => {
+		const { container } = render(
+			<Text highlightWords={ undefined }>
+				Some people are worth melting for.
+			</Text>
+		);
+		// It'll have a length of 1 because there shouldn't be anything but the single span being rendered
+		expect( container.firstChild.childNodes ).toHaveLength( 1 );
+	} );
+
+	test( 'should render highlighted words with highlightCaseSensitive', () => {
+		const { container } = render(
+			<Text highlightCaseSensitive highlightWords={ [ 'WORTH' ] }>
+				Some people are worth melting for.
+			</Text>
+		);
+
+		expect( container.firstChild ).toMatchSnapshot();
+		// It'll have a length of 1 because there shouldn't be anything but the single span being rendered
+		expect( container.firstChild.childNodes ).toHaveLength( 1 );
+	} );
+
+	test( 'should render isBlock', () => {
+		const { container } = render(
+			<Text isBlock>Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( {
+			display: 'block',
+		} );
+	} );
+
+	test( 'should render lineHeight', () => {
+		const { container } = render(
+			<Text lineHeight={ 1.5 }>Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( { lineHeight: '1.5' } );
+	} );
+
+	test( 'should render upperCase', () => {
+		const { container } = render(
+			<Text upperCase>Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( {
+			textTransform: 'uppercase',
+		} );
+	} );
+
+	test( 'should render weight', () => {
+		const { container } = render(
+			<Text weight={ 700 }>Some people are worth melting for.</Text>
+		);
+		expect( container.firstChild ).toHaveStyle( { fontWeight: '700' } );
+	} );
+} );
diff --git a/packages/components/src/text/next/text-styles.js b/packages/components/src/text/next/text-styles.js
new file mode 100644
index 00000000000000..947677a1b198b2
--- /dev/null
+++ b/packages/components/src/text/next/text-styles.js
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { css, ui } from '@wp-g2/styles';
+
+export const Text = css`
+	color: ${ ui.color.text };
+	line-height: ${ ui.get( 'fontLineHeightBase' ) };
+`;
+
+export const block = css`
+	display: block;
+`;
+
+export const positive = css`
+	color: ${ ui.color.positive };
+`;
+
+export const destructive = css`
+	color: ${ ui.color.destructive };
+`;
+
+export const muted = css`
+	color: ${ ui.get( 'colorTextMuted' ) };
+`;
+
+export const highlighterText = css`
+	mark {
+		background: ${ ui.get( 'yellowRgba70' ) };
+		border-radius: 2px;
+		box-shadow: 0 0 0 1px rgba( 0, 0, 0, 0.05 ) inset,
+			0 -1px 0 rgba( 0, 0, 0, 0.1 ) inset;
+	}
+`;
+
+export const upperCase = css`
+	text-transform: uppercase;
+`;
diff --git a/packages/components/src/text/next/text-utils.js b/packages/components/src/text/next/text-utils.js
new file mode 100644
index 00000000000000..72941af4f2f737
--- /dev/null
+++ b/packages/components/src/text/next/text-utils.js
@@ -0,0 +1,145 @@
+/**
+ * External dependencies
+ */
+import { ui } from '@wp-g2/styles';
+import { memoize } from '@wp-g2/utils';
+import { findAll } from 'highlight-words-core';
+
+/**
+ * WordPress dependencies
+ */
+import { createElement } from '@wordpress/element';
+
+/**
+ * Source:
+ * https://github.com/bvaughn/react-highlight-words/blob/master/src/Highlighter.js
+ */
+
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @typedef Options
+ * @property {string} [activeClassName=''] Classname for active highlighted areas.
+ * @property {number} [activeIndex=-1] The index of the active highlighted area.
+ * @property {import('react').AllHTMLAttributes['style']} [activeStyle] Styles to apply to the active highlighted area.
+ * @property {boolean} [autoEscape] Whether to automatically escape text.
+ * @property {boolean} [caseSensitive=false] Whether to highlight in a case-sensitive manner.
+ * @property {string} children Children to highlight.
+ * @property {import('highlight-words-core').FindAllArgs['findChunks']} [findChunks] Custom `findChunks` function to pass to `highlight-words-core`.
+ * @property {string | Record<string, string>} [highlightClassName=''] Classname to apply to highlighted text or a Record of classnames to apply to given text (which should be the key).
+ * @property {import('react').AllHTMLAttributes['style']} [highlightStyle={}] Styles to apply to highlighted text.
+ * @property {keyof JSX.IntrinsicElements} [highlightTag='mark'] Tag to use for the highlighted text.
+ * @property {import('highlight-words-core').FindAllArgs['sanitize']} [sanitize] Custom `santize` function to pass to `highlight-words-core`.
+ * @property {string[]} [searchWords=[]] Words to search for and highlight.
+ * @property {string} [unhighlightClassName=''] Classname to apply to unhighlighted text.
+ * @property {import('react').AllHTMLAttributes['style']} [unhighlightStyle] Style to apply to unhighlighted text.
+ */
+/* eslint-enable jsdoc/valid-types */
+
+/**
+ *
+ * @param {Options} options
+ */
+export function createHighlighterText( {
+	activeClassName = '',
+	activeIndex = -1,
+	activeStyle,
+	autoEscape,
+	caseSensitive = false,
+	children,
+	findChunks,
+	highlightClassName = '',
+	highlightStyle = {},
+	highlightTag = 'mark',
+	sanitize,
+	searchWords = [],
+	unhighlightClassName = '',
+	unhighlightStyle,
+} ) {
+	if ( ! children ) return null;
+	if ( typeof children !== 'string' ) return children;
+
+	const textToHighlight = children;
+
+	const chunks = findAll( {
+		autoEscape,
+		caseSensitive,
+		findChunks,
+		sanitize,
+		searchWords,
+		textToHighlight,
+	} );
+	const HighlightTag = highlightTag;
+	let highlightIndex = -1;
+	let highlightClassNames = '';
+	let highlightStyles;
+
+	const lowercaseProps = ( object ) => {
+		const mapped = {};
+		for ( const key in object ) {
+			mapped[ key.toLowerCase() ] = object[ key ];
+		}
+		return mapped;
+	};
+
+	const memoizedLowercaseProps = memoize( lowercaseProps );
+
+	const textContent = chunks.map( ( chunk, index ) => {
+		const text = textToHighlight.substr(
+			chunk.start,
+			chunk.end - chunk.start
+		);
+
+		if ( chunk.highlight ) {
+			highlightIndex++;
+
+			let highlightClass;
+			if ( typeof highlightClassName === 'object' ) {
+				if ( ! caseSensitive ) {
+					highlightClassName = memoizedLowercaseProps(
+						highlightClassName
+					);
+					highlightClass = highlightClassName[ text.toLowerCase() ];
+				} else {
+					highlightClass = highlightClassName[ text ];
+				}
+			} else {
+				highlightClass = highlightClassName;
+			}
+
+			const isActive = highlightIndex === +activeIndex;
+
+			highlightClassNames = `${ highlightClass } ${
+				isActive ? activeClassName : ''
+			}`;
+			highlightStyles =
+				isActive === true && activeStyle !== null
+					? Object.assign( {}, highlightStyle, activeStyle )
+					: highlightStyle;
+
+			const props = {
+				...ui.$( 'TextHighlight' ),
+				children: text,
+				className: highlightClassNames,
+				key: index,
+				style: highlightStyles,
+			};
+
+			// Don't attach arbitrary props to DOM elements; this triggers React DEV warnings (https://fb.me/react-unknown-prop)
+			// Only pass through the highlightIndex attribute for custom components.
+			if ( typeof HighlightTag !== 'string' ) {
+				props.highlightIndex = highlightIndex;
+			}
+
+			return createElement( HighlightTag, props );
+		}
+		return createElement( 'span', {
+			...ui.$( 'Text' ),
+			children: text,
+			className: unhighlightClassName,
+			key: index,
+			style: unhighlightStyle,
+		} );
+	} );
+
+	return textContent;
+}
diff --git a/packages/components/src/text/next/text.js b/packages/components/src/text/next/text.js
new file mode 100644
index 00000000000000..b7a54f106cb9fb
--- /dev/null
+++ b/packages/components/src/text/next/text.js
@@ -0,0 +1,11 @@
+/**
+ * Internal dependencies
+ */
+import { createComponent } from '../../utils';
+import { useText } from './use-text';
+
+export default createComponent( {
+	as: 'span',
+	useHook: useText,
+	name: 'Text',
+} );
diff --git a/packages/components/src/text/next/types.ts b/packages/components/src/text/next/types.ts
new file mode 100644
index 00000000000000..6fbb38ffefd3fc
--- /dev/null
+++ b/packages/components/src/text/next/types.ts
@@ -0,0 +1,188 @@
+import type { Props as TruncateProps } from '../../truncate/use-truncate';
+import type { CSSProperties } from 'react';
+
+type TextAdjustLineHeightForInnerControls =
+	| boolean
+	| 'large'
+	| 'medium'
+	| 'small'
+	| 'xSmall';
+
+export type TextSize =
+	| 'body'
+	| 'caption'
+	| 'footnote'
+	| 'largeTitle'
+	| 'subheadline'
+	| 'title';
+
+type TextVariant = 'muted';
+
+type TextWeight = 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900;
+
+export type Props = TruncateProps & {
+	/**
+	 * Adjusts the text alignment.
+	 *
+	 * @example
+	 * ```jsx
+	 * import { Text } from `@wp-g2/components`
+	 *
+	 * function Example() {
+	 * 	return (
+	 * 		<Text align="center" isBlock>
+	 * 			Where the north wind meets the sea...
+	 * 		</Text>
+	 * 	)
+	 * }
+	 *```
+	 */
+	align?: CSSProperties[ 'textAlign' ];
+	/**
+	 * Automatically calculate the appropriate line-height value for contents that render text and Control elements (e.g. `TextInput`).
+	 *
+	 * @example
+	 * ```jsx
+	 * import { Text, TextInput } from `@wp-g2/components`
+	 *
+	 * function Example() {
+	 * 	return (
+	 * 		<Text adjustLineHeightForInnerControls>
+	 * 			Where the north wind meets the <TextInput value="sea..." />
+	 * 		</Text>
+	 * 	)
+	 * }
+	 *```
+	 */
+	adjustLineHeightForInnerControls?: TextAdjustLineHeightForInnerControls;
+	/**
+	 * Adjusts the text color.
+	 */
+	color?: CSSProperties[ 'color' ];
+	/**
+	 * Adjusts the CSS display.
+	 */
+	display?: CSSProperties[ 'display' ];
+	/**
+	 * Renders a destructive color.
+	 *
+	 * @default false
+	 */
+	isDestructive?: boolean;
+	/**
+	 * Escape characters in `highlightWords` which are meaningful in regular expressions.
+	 */
+	highlightEscape?: boolean;
+	/**
+	 * Determines if `highlightWords` should be case sensitive.
+	 */
+	highlightCaseSensitive?: boolean;
+	/**
+	 * Letters or words within `Text` can be highlighted using `highlightWords`.
+	 *
+	 * @example
+	 * ```jsx
+	 * import { Text } from `@wp-g2/components`
+	 *
+	 * function Example() {
+	 *   return (
+	 *     <Text highlightWords={["the"]}>
+	 *       Where the north wind meets the sea, there's a river full of memory. Sleep,
+	 *       my darling, safe and sound, for in this river all is found. In her waters,
+	 *       deep and true, lay the answers and a path for you. Dive down deep into her
+	 *       sound, but not too far or you'll be drowned
+	 *     </Text>
+	 *   )
+	 * }
+	 * ```
+	 */
+	highlightWords?: string[];
+	/**
+	 * Array of search words. String search terms are automatically cast to RegExps unless `highlightEscape` is true.
+	 */
+	highlightSanitize?: import( 'highlight-words-core' ).FindAllArgs[ 'sanitize' ];
+	/**
+	 * Sets `Text` to have `display: block`.
+	 */
+	isBlock?: boolean;
+	/**
+	 * Adjusts all text line-height based on the typography system.
+	 */
+	lineHeight?: CSSProperties[ 'lineHeight' ];
+	/**
+	 * The `Text` color can be adapted to a background color for optimal readability. `optimizeReadabilityFor` can accept CSS variables, in addition to standard CSS color values (e.g. Hex, RGB, HSL, etc...).
+	 *
+	 * @example
+	 * ```jsx
+	 * import { Text, View } from `@wp-g2/components`
+	 *
+	 * function Example() {
+	 *   const backgroundColor = "blue"
+	 *
+	 *   return (
+	 *     <View css={{ backgroundColor }}>
+	 *       <Text optimizeReadabilityFor={backgroundColor}>
+	 *         Where the north wind meets the sea, there's a river full of memory.
+	 *       </Text>
+	 *     </View>
+	 *   )
+	 * }
+	 * ```
+	 */
+	optimizeReadabilityFor?: CSSProperties[ 'color' ];
+	/**
+	 * Adjusts text size based on the typography system. `Text` can render a wide range of font sizes, which are automatically calculated and adapted to the typography system. The `size` value can be a system preset, a `number`, or a custom unit value (`string`) such as `30em`.
+	 *
+	 * @example
+	 * ```jsx
+	 * import { Text } from `@wp-g2/components`
+	 *
+	 * function Example() {
+	 *   return <Text size="largeTitle">Where the north wind meets the sea...</Text>
+	 * }
+	 * ```
+	 */
+	size?: CSSProperties[ 'fontSize' ] | TextSize;
+	/**
+	 * Enables text truncation. When `truncate` is set,we are able to truncate the long text in a variety of ways.
+	 *
+	 * @example
+	 *
+	 * ```jsx
+	 * import { Text } from `@wp-g2/components`
+	 *
+	 * function Example() {
+	 *   return (
+	 *     <Text truncate>
+	 *       Where the north wind meets the sea, there's a river full of memory. Sleep,
+	 *       my darling, safe and sound, for in this river all is found. In her waters,
+	 *       deep and true, lay the answers and a path for you. Dive down deep into her
+	 *       sound, but not too far or you'll be drowned
+	 *     </Text>
+	 *   )
+	 * }
+	 * ```
+	 */
+	truncate?: boolean;
+	/**
+	 * Uppercases the text content.
+	 */
+	upperCase?: boolean;
+	/**
+	 * Adjusts style variation of the text.
+	 *
+	 * @example
+	 * ```jsx
+	 * import { Text } from `@wp-g2/components`
+	 *
+	 * function Example() {
+	 *   return <Text variant="muted">Where the north wind meets the sea...</Text>
+	 * }
+	 * ```
+	 */
+	variant?: TextVariant;
+	/**
+	 * Adjusts font-weight of the text.
+	 */
+	weight?: CSSProperties[ 'fontWeight' ] | TextWeight;
+};
diff --git a/packages/components/src/text/next/use-text.js b/packages/components/src/text/next/use-text.js
new file mode 100644
index 00000000000000..88f7fdca6bcdb7
--- /dev/null
+++ b/packages/components/src/text/next/use-text.js
@@ -0,0 +1,200 @@
+/**
+ * External dependencies
+ */
+import { hasNamespace, useContextSystem } from '@wp-g2/context';
+import { css, cx, getFontSize, ui } from '@wp-g2/styles';
+import { isPlainObject, isNil } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo, Children, cloneElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useTruncate } from '../../truncate';
+import { getOptimalTextShade } from '../../utils/colors';
+import * as styles from './text-styles';
+import { createHighlighterText } from './text-utils';
+
+/**
+ * @param {import('@wp-g2/create-styles').ViewOwnProps<import('./types').Props, 'span'>} props
+ */
+export function useText( props ) {
+	const {
+		adjustLineHeightForInnerControls,
+		align,
+		children,
+		className,
+		color,
+		ellipsizeMode,
+		isDestructive = false,
+		display,
+		highlightEscape = false,
+		highlightCaseSensitive = false,
+		highlightWords = [],
+		highlightSanitize,
+		isBlock = false,
+		letterSpacing,
+		lineHeight: lineHeightProp,
+		optimizeReadabilityFor,
+		size,
+		truncate = false,
+		upperCase = false,
+		variant,
+		weight = ui.get( 'fontWeight' ),
+		...otherProps
+	} = useContextSystem( props, 'Text' );
+
+	let content = children;
+	const isHighlighter =
+		Array.isArray( highlightWords ) && highlightWords.length;
+	const isCaption = size === 'caption';
+
+	if ( isHighlighter ) {
+		content = createHighlighterText( {
+			autoEscape: highlightEscape,
+			children,
+			caseSensitive: highlightCaseSensitive,
+			searchWords: highlightWords,
+			sanitize: highlightSanitize,
+		} );
+	}
+
+	const classes = useMemo( () => {
+		const sx = {};
+
+		const lineHeight = getLineHeight( {
+			lineHeight: lineHeightProp,
+			adjustLineHeightForInnerControls,
+		} );
+
+		sx.Base = css( {
+			color,
+			display,
+			fontSize: getFontSize( size ),
+			fontWeight: weight,
+			lineHeight,
+			letterSpacing,
+			textAlign: align,
+		} );
+
+		sx.upperCase = css( { textTransform: 'uppercase' } );
+
+		sx.optimalTextColor = null;
+
+		if ( optimizeReadabilityFor ) {
+			const isOptimalTextColorDark =
+				getOptimalTextShade( optimizeReadabilityFor ) === 'dark';
+
+			sx.optimalTextColor = isOptimalTextColorDark
+				? css( { color: ui.get( 'black' ) } )
+				: css( { color: ui.get( 'white' ) } );
+		}
+
+		return cx(
+			styles.Text,
+			sx.Base,
+			sx.optimalTextColor,
+			isDestructive && styles.destructive,
+			!! isHighlighter && styles.highlighterText,
+			isBlock && styles.block,
+			isCaption && styles.muted,
+			variant && styles[ variant ],
+			upperCase && sx.upperCase,
+			className
+		);
+	}, [
+		adjustLineHeightForInnerControls,
+		align,
+		className,
+		color,
+		display,
+		isBlock,
+		isCaption,
+		isDestructive,
+		isHighlighter,
+		letterSpacing,
+		lineHeightProp,
+		optimizeReadabilityFor,
+		size,
+		upperCase,
+		variant,
+		weight,
+	] );
+
+	/** @type {undefined | 'auto' | 'none'} */
+	let finalEllipsizeMode;
+	if ( truncate === true ) {
+		finalEllipsizeMode = 'auto';
+	}
+	if ( truncate === false ) {
+		finalEllipsizeMode = 'none';
+	}
+
+	const finalComponentProps = {
+		...otherProps,
+		className: classes,
+		children,
+		ellipsizeMode: ellipsizeMode || finalEllipsizeMode,
+	};
+
+	const truncateProps = useTruncate( finalComponentProps );
+
+	/**
+	 * Enhance child `<Link />` components to inherit font size.
+	 */
+	if ( ! truncate && Array.isArray( children ) ) {
+		content = Children.map( children, ( child ) => {
+			// @ts-ignore
+			if ( ! isPlainObject( child ) || ! ( 'props' in child ) ) {
+				return child;
+			}
+
+			const isLink = hasNamespace( child, [ 'Link' ] );
+			if ( isLink ) {
+				return cloneElement( child, {
+					size: child.props.size || 'inherit',
+				} );
+			}
+
+			return child;
+		} );
+	}
+
+	return {
+		...truncateProps,
+		children: truncate ? truncateProps.children : content,
+	};
+}
+
+function getLineHeight( { adjustLineHeightForInnerControls, lineHeight } ) {
+	if ( ! isNil( lineHeight ) ) return lineHeight;
+
+	if ( ! adjustLineHeightForInnerControls ) return;
+
+	let value = `calc(${ ui.get( 'controlHeight' ) } + ${ ui.space( 2 ) })`;
+
+	switch ( adjustLineHeightForInnerControls ) {
+		case 'large':
+			value = `calc(${ ui.get( 'controlHeightLarge' ) } + ${ ui.space(
+				2
+			) })`;
+			break;
+		case 'small':
+			value = `calc(${ ui.get( 'controlHeightSmall' ) } + ${ ui.space(
+				2
+			) })`;
+			break;
+		case 'xSmall':
+			value = `calc(${ ui.get( 'controlHeightXSmall' ) } + ${ ui.space(
+				2
+			) })`;
+			break;
+		default:
+			break;
+	}
+
+	return value;
+}
diff --git a/packages/components/src/utils/colors.js b/packages/components/src/utils/colors.js
index 6815964f57a879..b7d6938c2e9a41 100644
--- a/packages/components/src/utils/colors.js
+++ b/packages/components/src/utils/colors.js
@@ -3,6 +3,7 @@
  */
 import { get } from 'lodash';
 import tinycolor from 'tinycolor2';
+import memoize from 'memize';
 
 /**
  * Internal dependencies
@@ -39,3 +40,146 @@ export function color( value ) {
 	const fallbackColor = '#000';
 	return get( COLORS, value, fallbackColor );
 }
+
+/** @type {HTMLDivElement} */
+let __colorComputationNode;
+
+/**
+ * @return {HTMLDivElement | undefined} The HTML element for color computation.
+ */
+function getColorComputationNode() {
+	if ( typeof document === 'undefined' ) return;
+
+	if ( ! __colorComputationNode ) {
+		// Create a temporary element for style computation.
+		const el = document.createElement( 'div' );
+		el.setAttribute( 'data-g2-color-computation-node', '' );
+		// Inject for window computed style.
+		document.body.appendChild( el );
+		__colorComputationNode = el;
+	}
+
+	return __colorComputationNode;
+}
+
+/**
+ * @param {string | unknown} value
+ *
+ * @return {boolean} Whether the value is a valid color.
+ */
+export function isColor( value ) {
+	if ( typeof value !== 'string' ) return false;
+	const test = tinycolor( value );
+
+	return test.isValid();
+}
+
+/**
+ * Retrieves the computed background color. This is useful for getting the
+ * value of a CSS variable color.
+ *
+ * @param {string | unknown} backgroundColor The background color to compute.
+ *
+ * @return {string} The computed background color.
+ */
+function __getComputedBackgroundColor( backgroundColor ) {
+	if ( typeof backgroundColor !== 'string' ) return '';
+
+	if ( isColor( backgroundColor ) ) return backgroundColor;
+
+	if ( ! backgroundColor.includes( 'var(' ) ) return '';
+	if ( typeof document === 'undefined' ) return '';
+
+	// Attempts to gracefully handle CSS variables color values.
+	const el = getColorComputationNode();
+	if ( ! el ) return '';
+
+	el.style.background = backgroundColor;
+	// Grab the style
+	const computedColor = window?.getComputedStyle( el ).background;
+	// Reset
+	el.style.background = '';
+
+	return computedColor || '';
+}
+
+/**
+ * Retrieves the computed background color. This is useful for getting the
+ * value of a CSS variable color.
+ *
+ * @param {string | unknown} color The background color to compute.
+ *
+ * @return {string} The computed background color.
+ */
+export const getComputedBackgroundColor = memoize(
+	__getComputedBackgroundColor
+);
+
+/**
+ * Retrieves the computed text color. This is useful for getting the
+ * value of a CSS variable color.
+ *
+ * @param {string | unknown} textColor
+ *
+ * @return {string} The computed text color.
+ */
+function __getComputedColor( textColor ) {
+	if ( typeof textColor !== 'string' ) return '';
+
+	if ( isColor( textColor ) ) return textColor;
+
+	if ( ! textColor.includes( 'var(' ) ) return '';
+	if ( typeof document === 'undefined' ) return '';
+
+	// Attempts to gracefully handle CSS variables color values.
+	const el = getColorComputationNode();
+	if ( ! el ) return '';
+
+	el.style.color = textColor;
+	// Grab the style
+	const computedColor = window?.getComputedStyle( el ).color;
+	// Reset
+	el.style.color = '';
+
+	return computedColor || '';
+}
+
+/**
+ * Retrieves the computed text color. This is useful for getting the
+ * value of a CSS variable color.
+ *
+ * @param {string | unknown} color
+ *
+ * @return {string} The computed text color.
+ */
+export const getComputedColor = memoize( __getComputedColor );
+
+/**
+ * Get the text shade optimized for readability, based on a background color.
+ *
+ * @param {string | unknown} backgroundColor  The background color.
+ *
+ * @return {string} The optimized text color (black or white).
+ */
+export function getOptimalTextColor( backgroundColor ) {
+	const background = getComputedBackgroundColor( backgroundColor );
+	const isReadableWithBlackText = tinycolor.isReadable(
+		background,
+		'#000000'
+	);
+
+	return isReadableWithBlackText ? '#000000' : '#ffffff';
+}
+
+/**
+ * Get the text shade optimized for readability, based on a background color.
+ *
+ * @param {string | unknown} backgroundColor The background color.
+ *
+ * @return {string} The optimized text shade (dark or light).
+ */
+export function getOptimalTextShade( backgroundColor ) {
+	const result = getOptimalTextColor( backgroundColor );
+
+	return result === '#000000' ? 'dark' : 'light';
+}
