diff --git a/packages/components/src/index.native.js b/packages/components/src/index.native.js
index afe0c37691e405..3422650608f186 100644
--- a/packages/components/src/index.native.js
+++ b/packages/components/src/index.native.js
@@ -75,6 +75,7 @@ export { default as ReadableContentView } from './mobile/readable-content-view';
 export { default as CycleSelectControl } from './mobile/cycle-select-control';
 export { default as Gradient } from './mobile/gradient';
 export { default as ColorSettings } from './mobile/color-settings';
+export { default as SegmentedControls } from './mobile/segmented-control';
 export { default as FocalPointSettingsPanel } from './mobile/focal-point-settings-panel';
 export { LinkPicker } from './mobile/link-picker';
 export { default as LinkPickerScreen } from './mobile/link-picker/link-picker-screen';
