diff --git a/packages/components/src/ui/context/context-connect.js b/packages/components/src/ui/context/context-connect.js
index 3c77cd339d1888..e03fb68eb7c1f7 100644
--- a/packages/components/src/ui/context/context-connect.js
+++ b/packages/components/src/ui/context/context-connect.js
@@ -23,7 +23,7 @@ import { CONNECT_STATIC_NAMESPACE } from './constants';
  * The hope is that we can improve render performance by removing functional
  * component wrappers.
  *
- * @template {import('../create-styles').ViewOwnProps<{}, any>} P
+ * @template {import('./polymorphic-component').ViewOwnProps<{}, any>} P
  * @param {(props: P, ref: import('react').Ref<any>) => JSX.Element | null} Component The component to register into the Context system.
  * @param {Array<string>|string} namespace The namespace to register the component under.
  * @param {Object} options
