diff --git a/packages/components/src/custom-gradient-bar/control-points.js b/packages/components/src/custom-gradient-bar/control-points.js
index e5edfc88820144..fc63c5896a5873 100644
--- a/packages/components/src/custom-gradient-bar/control-points.js
+++ b/packages/components/src/custom-gradient-bar/control-points.js
@@ -10,6 +10,7 @@ import { useInstanceId } from '@wordpress/compose';
 import { useEffect, useRef, useState } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 import { plus } from '@wordpress/icons';
+import { LEFT, RIGHT } from '@wordpress/keycodes';
 
 /**
  * Internal dependencies
@@ -17,7 +18,6 @@ import { plus } from '@wordpress/icons';
 import Button from '../button';
 import ColorPicker from '../color-picker';
 import Dropdown from '../dropdown';
-import KeyboardShortcuts from '../keyboard-shortcuts';
 import { VisuallyHidden } from '../visually-hidden';
 
 import {
@@ -36,46 +36,11 @@ import {
 	KEYBOARD_CONTROL_POINT_VARIATION,
 } from './constants';
 
-function ControlPointKeyboardMove( { value: position, onChange, children } ) {
-	const shortcuts = {
-		right( event ) {
-			// Stop propagation of the key press event to avoid focus moving
-			// to another editor area.
-			event.stopPropagation();
-			const newPosition = clampPercent(
-				position + KEYBOARD_CONTROL_POINT_VARIATION
-			);
-			onChange( newPosition );
-		},
-		left( event ) {
-			// Stop propagation of the key press event to avoid focus moving
-			// to another editor area.
-			event.stopPropagation();
-			const newPosition = clampPercent(
-				position - KEYBOARD_CONTROL_POINT_VARIATION
-			);
-			onChange( newPosition );
-		},
-	};
-
-	return (
-		<KeyboardShortcuts shortcuts={ shortcuts }>
-			{ children }
-		</KeyboardShortcuts>
-	);
-}
-
-function ControlPointButton( {
-	isOpen,
-	position,
-	color,
-	onChange,
-	...additionalProps
-} ) {
+function ControlPointButton( { isOpen, position, color, ...additionalProps } ) {
 	const instanceId = useInstanceId( ControlPointButton );
 	const descriptionId = `components-custom-gradient-picker__control-point-button-description-${ instanceId }`;
 	return (
-		<ControlPointKeyboardMove value={ position } onChange={ onChange }>
+		<>
 			<Button
 				aria-label={ sprintf(
 					// translators: %1$s: gradient position e.g: 70, %2$s: gradient color code e.g: rgb(52,121,151).
@@ -104,7 +69,7 @@ function ControlPointButton( {
 					'Use your left or right arrow keys or drag and drop with the mouse to change the gradient position. Press the button to change the color or remove the control point.'
 				) }
 			</VisuallyHidden>
-		</ControlPointKeyboardMove>
+		</>
 	);
 }
 
@@ -208,18 +173,40 @@ function ControlPoints( {
 									);
 								}
 							} }
+							onKeyDown={ ( event ) => {
+								if ( event.keyCode === LEFT ) {
+									// Stop propagation of the key press event to avoid focus moving
+									// to another editor area.
+									event.stopPropagation();
+									onChange(
+										updateControlPointPosition(
+											controlPoints,
+											index,
+											clampPercent(
+												point.position -
+													KEYBOARD_CONTROL_POINT_VARIATION
+											)
+										)
+									);
+								} else if ( event.keyCode === RIGHT ) {
+									// Stop propagation of the key press event to avoid focus moving
+									// to another editor area.
+									event.stopPropagation();
+									onChange(
+										updateControlPointPosition(
+											controlPoints,
+											index,
+											clampPercent(
+												point.position +
+													KEYBOARD_CONTROL_POINT_VARIATION
+											)
+										)
+									);
+								}
+							} }
 							isOpen={ isOpen }
 							position={ point.position }
 							color={ point.color }
-							onChange={ ( newPosition ) => {
-								onChange(
-									updateControlPointPosition(
-										controlPoints,
-										index,
-										newPosition
-									)
-								);
-							} }
 						/>
 					) }
 					renderContent={ ( { onClose } ) => (
