diff --git a/packages/components/src/truncate/component.js b/packages/components/src/truncate/component.js
index 5cb453b778c8de..aa8ab965007e00 100644
--- a/packages/components/src/truncate/component.js
+++ b/packages/components/src/truncate/component.js
@@ -1,9 +1,20 @@
 /**
  * Internal dependencies
  */
-import { createComponent } from '../ui/utils';
+import { contextConnect } from '../ui/context';
+import { View } from '../view';
 import useTruncate from './hook';
 
+/**
+ * @param {import('../ui/context').WordPressComponentProps<import('./types').Props, 'span'>} props
+ * @param {import('react').Ref<any>}                                                         forwardedRef
+ */
+function Truncate( props, forwardedRef ) {
+	const truncateProps = useTruncate( props );
+
+	return <View as="span" { ...truncateProps } ref={ forwardedRef } />;
+}
+
 /**
  * `Truncate` is a typography primitive that trims text content.
  * For almost all cases, it is recommended that `Text`, `Heading`, or
@@ -25,10 +36,6 @@ import useTruncate from './hook';
  * }
  * ```
  */
-const Truncate = createComponent( {
-	as: 'span',
-	useHook: useTruncate,
-	name: 'Truncate',
-} );
+const ConnectedTruncate = contextConnect( Truncate, 'Truncate' );
 
-export default Truncate;
+export default ConnectedTruncate;
