diff --git a/packages/components/src/navigator/navigator-provider/component.js b/packages/components/src/navigator/navigator-provider/component.js
index a49553ac5377c4..e4963762c1b836 100644
--- a/packages/components/src/navigator/navigator-provider/component.js
+++ b/packages/components/src/navigator/navigator-provider/component.js
@@ -1,20 +1,32 @@
 /**
  * WordPress dependencies
  */
-import { useState } from '@wordpress/element';
+import { useReducer } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import { useCx } from '../../utils/hooks/use-cx';
 import { NavigatorContext } from '../context';
+import { gridDeck } from '../styles';
+
+function locationReducer( holding, incoming ) {
+	const isBack = incoming.isBack ?? holding.lastPath === incoming.path;
+	return { ...incoming, isBack, lastPath: holding.path, isFirst: false };
+}
 
 function NavigatorProvider( { initialPath, children } ) {
-	const [ path, setPath ] = useState( { path: initialPath } );
+	const [ location, navigate ] = useReducer( locationReducer, {
+		path: initialPath,
+		isFirst: true,
+	} );
 
 	return (
-		<NavigatorContext.Provider value={ [ path, setPath ] }>
-			{ children }
-		</NavigatorContext.Provider>
+		<div className={ useCx()( gridDeck ) }>
+			<NavigatorContext.Provider value={ [ location, navigate ] }>
+				{ children }
+			</NavigatorContext.Provider>
+		</div>
 	);
 }
 
diff --git a/packages/components/src/navigator/navigator-screen/component.js b/packages/components/src/navigator/navigator-screen/component.js
index 3a5e33605a0b3e..fa60eb8f2210f2 100644
--- a/packages/components/src/navigator/navigator-screen/component.js
+++ b/packages/components/src/navigator/navigator-screen/component.js
@@ -1,40 +1,44 @@
-/**
- * External dependencies
- */
-// eslint-disable-next-line no-restricted-imports
-import { motion } from 'framer-motion';
-
 /**
  * WordPress dependencies
  */
-import { useContext, useEffect, useState } from '@wordpress/element';
+import { useContext, useReducer } from '@wordpress/element';
 import { useReducedMotion, useFocusOnMount } from '@wordpress/compose';
-import { isRTL } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
+import { useCx } from '../../utils/hooks/use-cx';
 import { NavigatorContext } from '../context';
+import { animation, pointerNone } from '../styles';
 
-const animationEnterDelay = 0;
-const animationEnterDuration = 0.14;
-const animationExitDuration = 0.14;
-const animationExitDelay = 0;
+function usePresence( shouldPresent ) {
+	const [ { isPresent }, dispatch ] = useReducer( usePresence.reducer, {} );
+	if ( shouldPresent && ! isPresent ) {
+		dispatch( true );
+	}
+	return [ isPresent, dispatch ];
+}
 
-function NavigatorScreen( { children, path } ) {
+usePresence.reducer = ( holding, predicate ) => {
+	if ( predicate === true ) {
+		// Mutates holding and returns it since a render isn't wanted.
+		holding.isPresent = true;
+		return holding;
+	}
+	return { isPresent: false };
+};
+
+function NavigatorScreen( { children, className, path } ) {
 	const prefersReducedMotion = useReducedMotion();
-	const [ currentPath ] = useContext( NavigatorContext );
-	const isMatch = currentPath.path === path;
+	const [ location ] = useContext( NavigatorContext );
+	const isCurrent = location.path === path;
+	const [ isPresent, exit ] = usePresence( isCurrent );
+
 	const ref = useFocusOnMount();
 
-	// This flag is used to only apply the focus on mount when the actual path changes.
-	// It avoids the focus to happen on the first render.
-	const [ hasPathChanged, setHasPathChanged ] = useState( false );
-	useEffect( () => {
-		setHasPathChanged( true );
-	}, [ path ] );
+	const cx = useCx();
 
-	if ( ! isMatch ) {
+	if ( ! isPresent || ( prefersReducedMotion && ! isCurrent ) ) {
 		return null;
 	}
 
@@ -42,51 +46,21 @@ function NavigatorScreen( { children, path } ) {
 		return <div>{ children }</div>;
 	}
 
-	const animate = {
-		opacity: 1,
-		transition: {
-			delay: animationEnterDelay,
-			duration: animationEnterDuration,
-			ease: 'easeInOut',
-		},
-		x: 0,
-	};
-	const initial = {
-		opacity: 0,
-		x:
-			( isRTL() && currentPath.isBack ) ||
-			( ! isRTL() && ! currentPath.isBack )
-				? 50
-				: -50,
-	};
-	const exit = {
-		delay: animationExitDelay,
-		opacity: 0,
-		x:
-			( ! isRTL() && currentPath.isBack ) ||
-			( isRTL() && ! currentPath.isBack )
-				? 50
-				: -50,
-		transition: {
-			duration: animationExitDuration,
-			ease: 'easeInOut',
-		},
-	};
+	const props = { className: [ className ] };
+
+	if ( ! isCurrent ) {
+		props.ariaHidden = true;
+		props.onAnimationEnd = exit;
+		props.className.push( pointerNone );
+	}
+	if ( ! location.isFirst ) {
+		props.ref = ref;
+		props.className.push( animation( isCurrent, location.isBack ) );
+	}
 
-	const animatedProps = {
-		animate,
-		exit,
-		initial,
-	};
+	props.className = cx( ...props.className );
 
-	return (
-		<motion.div
-			ref={ hasPathChanged ? ref : undefined }
-			{ ...animatedProps }
-		>
-			{ children }
-		</motion.div>
-	);
+	return <div { ...props }>{ children }</div>;
 }
 
 export default NavigatorScreen;
diff --git a/packages/components/src/navigator/styles.js b/packages/components/src/navigator/styles.js
new file mode 100644
index 00000000000000..756d81ac95bf10
--- /dev/null
+++ b/packages/components/src/navigator/styles.js
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { css, keyframes } from '@emotion/react';
+
+/**
+ * WordPress dependencies
+ */
+import { isRTL } from '@wordpress/i18n';
+
+const animationDuration = 0.14;
+const animationEase = 'cubic-bezier(0, 0, 0.2, 1)';
+const skimUnit = 'px';
+const skimVector = isRTL() ? 50 : -50;
+
+const fadeKeyframes = ( phase ) => keyframes`${ phase } { opacity: 0 }`;
+
+const skimKeyframes = ( phase, vector ) =>
+	keyframes`${ phase } { transform: translateX(${ vector }${ skimUnit }); }`;
+
+export const animation = ( isCurrent, isBack ) => {
+	const phase = isCurrent ? 'from' : 'to';
+	let vector = isBack ? -skimVector : skimVector;
+	vector *= isCurrent ? -1 : 1;
+	const fade = fadeKeyframes( phase );
+	const skim = skimKeyframes( phase, vector );
+	return css`
+		label: ${ phase }-${ vector === skimVector ? 'aft' : 'fore' };
+		animation-name: ${ fade }, ${ skim };
+		animation-duration: ${ animationDuration }s;
+		animation-timing-function: ${ animationEase };
+		animation-iteration-count: 1;
+	`;
+};
+
+export const pointerNone = css`
+	pointer-events: none;
+`;
+
+export const gridDeck = css`
+	display: grid;
+	grid-template: 1fr/1fr;
+	> * {
+		grid-area: 1/1;
+	}
+`;
