diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index 839980fd15745e..9d617bed8b71eb 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -34,6 +34,7 @@
 -   Add support for rendering the `ColorPalette` in a `Dropdown` when opened in the sidebar ([#37067](https://github.com/WordPress/gutenberg/pull/37067))
 -   Show an incremental sequence of numbers (1/2/3/4/5) as a label of the font size, when we have at most five font sizes, where at least one the them contains a complex css value(clamp, var, etc..). We do this because complex css values cannot be calculated properly and the incremental sequence of numbers as labels can help the user better mentally map the different available font sizes. ([#37038](https://github.com/WordPress/gutenberg/pull/37038))
 -   Add support for proper borders to color indicators ([#37500](https://github.com/WordPress/gutenberg/pull/37500))
+-   Refactor `SuggestionsList` class component to functional component([#36924](https://github.com/WordPress/gutenberg/pull/36924/))
 
 ## 19.1.4 (2021-12-13)
 
diff --git a/packages/components/src/form-token-field/suggestions-list.js b/packages/components/src/form-token-field/suggestions-list.js
index 1ad0ace0a3b89e..b7eee38e95e243 100644
--- a/packages/components/src/form-token-field/suggestions-list.js
+++ b/packages/components/src/form-token-field/suggestions-list.js
@@ -2,157 +2,146 @@
  * External dependencies
  */
 import { map } from 'lodash';
-import scrollIntoView from 'dom-scroll-into-view';
+import scrollView from 'dom-scroll-into-view';
 import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { withSafeTimeout } from '@wordpress/compose';
+import { useState } from '@wordpress/element';
+import { withSafeTimeout, useRefEffect } from '@wordpress/compose';
 
-class SuggestionsList extends Component {
-	constructor() {
-		super( ...arguments );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.bindList = this.bindList.bind( this );
-	}
+const emptyList = Object.freeze( [] );
 
-	componentDidUpdate() {
-		// only have to worry about scrolling selected suggestion into view
-		// when already expanded
-		if (
-			this.props.selectedIndex > -1 &&
-			this.props.scrollIntoView &&
-			this.list.children[ this.props.selectedIndex ]
-		) {
-			this.scrollingIntoView = true;
-			scrollIntoView(
-				this.list.children[ this.props.selectedIndex ],
-				this.list,
-				{
-					onlyScrollIfNeeded: true,
-				}
-			);
+const handleMouseDown = ( e ) => {
+	// By preventing default here, we will not lose focus of <input> when clicking a suggestion
+	e.preventDefault();
+};
 
-			this.props.setTimeout( () => {
-				this.scrollingIntoView = false;
-			}, 100 );
-		}
-	}
+function SuggestionsList( {
+	selectedIndex,
+	scrollIntoView,
+	match = '',
+	onHover,
+	onSelect,
+	suggestions = emptyList,
+	displayTransform,
+	instanceId,
+	setTimeout,
+} ) {
+	const [ scrollingIntoView, setScrollingIntoView ] = useState( false );
 
-	bindList( ref ) {
-		this.list = ref;
-	}
+	const listRef = useRefEffect(
+		( listNode ) => {
+			// only have to worry about scrolling selected suggestion into view
+			// when already expanded
+			if (
+				selectedIndex > -1 &&
+				scrollIntoView &&
+				listNode.children[ selectedIndex ]
+			) {
+				setScrollingIntoView( true );
+				scrollView( listNode.children[ selectedIndex ], listNode, {
+					onlyScrollIfNeeded: true,
+				} );
+				setTimeout( () => {
+					setScrollingIntoView( false );
+				}, 100 );
+			}
+		},
+		[ selectedIndex, scrollIntoView ]
+	);
 
-	handleHover( suggestion ) {
+	const handleHover = ( suggestion ) => {
 		return () => {
-			if ( ! this.scrollingIntoView ) {
-				this.props.onHover( suggestion );
+			if ( ! scrollingIntoView ) {
+				onHover?.( suggestion );
 			}
 		};
-	}
+	};
 
-	handleClick( suggestion ) {
+	const handleClick = ( suggestion ) => {
 		return () => {
-			this.props.onSelect( suggestion );
+			onSelect?.( suggestion );
 		};
-	}
+	};
 
-	handleMouseDown( e ) {
-		// By preventing default here, we will not lose focus of <input> when clicking a suggestion
-		e.preventDefault();
-	}
-
-	computeSuggestionMatch( suggestion ) {
-		const match = this.props
-			.displayTransform( this.props.match || '' )
-			.toLocaleLowerCase();
-		if ( match.length === 0 ) {
+	const computeSuggestionMatch = ( suggestion ) => {
+		const matchText = displayTransform( match || '' ).toLocaleLowerCase();
+		if ( matchText.length === 0 ) {
 			return null;
 		}
 
-		suggestion = this.props.displayTransform( suggestion );
-		const indexOfMatch = suggestion.toLocaleLowerCase().indexOf( match );
+		suggestion = displayTransform( suggestion );
+		const indexOfMatch = suggestion
+			.toLocaleLowerCase()
+			.indexOf( matchText );
 
 		return {
 			suggestionBeforeMatch: suggestion.substring( 0, indexOfMatch ),
 			suggestionMatch: suggestion.substring(
 				indexOfMatch,
-				indexOfMatch + match.length
+				indexOfMatch + matchText.length
 			),
 			suggestionAfterMatch: suggestion.substring(
-				indexOfMatch + match.length
+				indexOfMatch + matchText.length
 			),
 		};
-	}
+	};
 
-	render() {
-		// We set `tabIndex` here because otherwise Firefox sets focus on this
-		// div when tabbing off of the input in `TokenField` -- not really sure
-		// why, since usually a div isn't focusable by default
-		// TODO does this still apply now that it's a <ul> and not a <div>?
-		return (
-			<ul
-				ref={ this.bindList }
-				className="components-form-token-field__suggestions-list"
-				id={ `components-form-token-suggestions-${ this.props.instanceId }` }
-				role="listbox"
-			>
-				{ map( this.props.suggestions, ( suggestion, index ) => {
-					const match = this.computeSuggestionMatch( suggestion );
-					const classeName = classnames(
-						'components-form-token-field__suggestion',
-						{
-							'is-selected': index === this.props.selectedIndex,
-						}
-					);
+	// We set `tabIndex` here because otherwise Firefox sets focus on this
+	// div when tabbing off of the input in `TokenField` -- not really sure
+	// why, since usually a div isn't focusable by default
+	// TODO does this still apply now that it's a <ul> and not a <div>?
+	return (
+		<ul
+			ref={ listRef }
+			className="components-form-token-field__suggestions-list"
+			id={ `components-form-token-suggestions-${ instanceId }` }
+			role="listbox"
+		>
+			{ map( suggestions, ( suggestion, index ) => {
+				const matchText = computeSuggestionMatch( suggestion );
+				const className = classnames(
+					'components-form-token-field__suggestion',
+					{
+						'is-selected': index === selectedIndex,
+					}
+				);
 
-					/* eslint-disable jsx-a11y/click-events-have-key-events */
-					return (
-						<li
-							id={ `components-form-token-suggestions-${ this.props.instanceId }-${ index }` }
-							role="option"
-							className={ classeName }
-							key={
-								suggestion?.value
-									? suggestion.value
-									: this.props.displayTransform( suggestion )
-							}
-							onMouseDown={ this.handleMouseDown }
-							onClick={ this.handleClick( suggestion ) }
-							onMouseEnter={ this.handleHover( suggestion ) }
-							aria-selected={ index === this.props.selectedIndex }
-						>
-							{ match ? (
-								<span
-									aria-label={ this.props.displayTransform(
-										suggestion
-									) }
-								>
-									{ match.suggestionBeforeMatch }
-									<strong className="components-form-token-field__suggestion-match">
-										{ match.suggestionMatch }
-									</strong>
-									{ match.suggestionAfterMatch }
-								</span>
-							) : (
-								this.props.displayTransform( suggestion )
-							) }
-						</li>
-					);
-					/* eslint-enable jsx-a11y/click-events-have-key-events */
-				} ) }
-			</ul>
-		);
-	}
+				/* eslint-disable jsx-a11y/click-events-have-key-events */
+				return (
+					<li
+						id={ `components-form-token-suggestions-${ instanceId }-${ index }` }
+						role="option"
+						className={ className }
+						key={
+							suggestion?.value
+								? suggestion.value
+								: displayTransform( suggestion )
+						}
+						onMouseDown={ handleMouseDown }
+						onClick={ handleClick( suggestion ) }
+						onMouseEnter={ handleHover( suggestion ) }
+						aria-selected={ index === selectedIndex }
+					>
+						{ matchText ? (
+							<span aria-label={ displayTransform( suggestion ) }>
+								{ matchText.suggestionBeforeMatch }
+								<strong className="components-form-token-field__suggestion-match">
+									{ matchText.suggestionMatch }
+								</strong>
+								{ matchText.suggestionAfterMatch }
+							</span>
+						) : (
+							displayTransform( suggestion )
+						) }
+					</li>
+				);
+				/* eslint-enable jsx-a11y/click-events-have-key-events */
+			} ) }
+		</ul>
+	);
 }
 
-SuggestionsList.defaultProps = {
-	match: '',
-	onHover: () => {},
-	onSelect: () => {},
-	suggestions: Object.freeze( [] ),
-};
-
 export default withSafeTimeout( SuggestionsList );
diff --git a/packages/components/src/form-token-field/test/index.js b/packages/components/src/form-token-field/test/index.js
index f166b5b201c09f..0a9ee88d9c4d67 100644
--- a/packages/components/src/form-token-field/test/index.js
+++ b/packages/components/src/form-token-field/test/index.js
@@ -2,7 +2,7 @@
  * External dependencies
  */
 import { filter, map } from 'lodash';
-import TestUtils from 'react-dom/test-utils';
+import TestUtils, { act } from 'react-dom/test-utils';
 import ReactDOM from 'react-dom';
 
 /**
@@ -262,7 +262,9 @@ describe( 'FormTokenField', () => {
 
 			// before sending a hover event, we need to wait for
 			// SuggestionList#_scrollingIntoView to become false
-			jest.advanceTimersByTime( 100 );
+			act( () => {
+				jest.advanceTimersByTime( 100 );
+			} );
 
 			TestUtils.Simulate.mouseEnter( hoverSuggestion );
 			expect( getSelectedSuggestion() ).toEqual( [ 'wi', 'th' ] );
