diff --git a/packages/components/src/snackbar/index.js b/packages/components/src/snackbar/index.js
index 486ec5366db176..6f7dd1a05b4f3c 100644
--- a/packages/components/src/snackbar/index.js
+++ b/packages/components/src/snackbar/index.js
@@ -65,7 +65,7 @@ function Snackbar(
 		}
 
 		// Prevent focus loss by moving it to the list element.
-		listRef.current.focus();
+		listRef?.current?.focus();
 
 		onDismiss();
 		onRemove();
