diff --git a/packages/block-editor/src/components/letter-spacing-control/index.js b/packages/block-editor/src/components/letter-spacing-control/index.js
index dac792b2eb3bd4..002ef618e6f626 100644
--- a/packages/block-editor/src/components/letter-spacing-control/index.js
+++ b/packages/block-editor/src/components/letter-spacing-control/index.js
@@ -12,6 +12,20 @@ import { __ } from '@wordpress/i18n';
  */
 import useSetting from '../../components/use-setting';
 
+/**
+ * Minimum space between letters
+ *
+ * @type {number}
+ */
+const MIN_LETTER_SPACE = 0.1;
+
+/**
+ * Maximum space between letters
+ *
+ * @type {number}
+ */
+const MAX_LETTER_SPACE = 100;
+
 /**
  * Control for letter-spacing.
  *
@@ -38,6 +52,8 @@ export default function LetterSpacingControl( {
 			__unstableInputWidth={ __unstableInputWidth }
 			units={ units }
 			onChange={ onChange }
+			min={ MIN_LETTER_SPACE }
+			max={ MAX_LETTER_SPACE }
 		/>
 	);
 }
