diff --git a/.eslintrc.js b/.eslintrc.js
index 611ac94827640b..a56df2632f2126 100644
--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -115,6 +115,7 @@ module.exports = {
 							'toString',
 							'trim',
 							'uniqWith',
+							'values',
 						],
 						message:
 							'This Lodash method is not recommended. Please use native functionality instead. If using `memoize`, please use `memize` instead.',
diff --git a/packages/components/src/mobile/segmented-control/index.native.js b/packages/components/src/mobile/segmented-control/index.native.js
index 5440ffffae86e5..b34b8d158a0899 100644
--- a/packages/components/src/mobile/segmented-control/index.native.js
+++ b/packages/components/src/mobile/segmented-control/index.native.js
@@ -9,7 +9,6 @@ import {
 	Animated,
 	Easing,
 } from 'react-native';
-import { values, map, reduce } from 'lodash';
 /**
  * WordPress dependencies
  */
@@ -101,11 +100,13 @@ const SegmentedControls = ( {
 		const { paddingLeft: offset } = isIOS
 			? styles.containerIOS
 			: styles.container;
-		const widths = map( values( segmentsDimensions ), 'width' );
+		const widths = Object.values( segmentsDimensions ).map(
+			( dimension ) => dimension.width
+		);
 		const widthsDistance = widths.slice( 0, index );
-		const widthsDistanceSum = reduce(
-			widthsDistance,
-			( sum, n ) => sum + n
+		const widthsDistanceSum = widthsDistance.reduce(
+			( sum, n ) => sum + n,
+			0
 		);
 
 		const endValue = index === 0 ? 0 : widthsDistanceSum;
diff --git a/packages/edit-post/src/store/selectors.js b/packages/edit-post/src/store/selectors.js
index 74d42691114c93..b84e2b6887431a 100644
--- a/packages/edit-post/src/store/selectors.js
+++ b/packages/edit-post/src/store/selectors.js
@@ -2,7 +2,6 @@
  * External dependencies
  */
 import createSelector from 'rememo';
-import { includes, some, values } from 'lodash';
 
 /**
  * WordPress dependencies
@@ -43,8 +42,7 @@ export const isEditorSidebarOpened = createRegistrySelector(
 			select( interfaceStore ).getActiveComplementaryArea(
 				'core/edit-post'
 			);
-		return includes(
-			[ 'edit-post/document', 'edit-post/block' ],
+		return [ 'edit-post/document', 'edit-post/block' ].includes(
 			activeGeneralSidebar
 		);
 	}
@@ -65,8 +63,7 @@ export const isPluginSidebarOpened = createRegistrySelector(
 			);
 		return (
 			!! activeGeneralSidebar &&
-			! includes(
-				[ 'edit-post/document', 'edit-post/block' ],
+			! [ 'edit-post/document', 'edit-post/block' ].includes(
 				activeGeneralSidebar
 			)
 		);
@@ -254,7 +251,7 @@ export function isPublishSidebarOpened( state ) {
  * @return {boolean} Whether or not the panel is removed.
  */
 export function isEditorPanelRemoved( state, panelName ) {
-	return includes( state.removedPanels, panelName );
+	return state.removedPanels.includes( panelName );
 }
 
 /**
@@ -369,7 +366,7 @@ export const getActiveMetaBoxLocations = createSelector(
 export function isMetaBoxLocationVisible( state, location ) {
 	return (
 		isMetaBoxLocationActive( state, location ) &&
-		some( getMetaBoxesPerLocation( state, location ), ( { id } ) => {
+		getMetaBoxesPerLocation( state, location )?.some( ( { id } ) => {
 			return isEditorPanelEnabled( state, `meta-box-${ id }` );
 		} )
 	);
@@ -410,7 +407,7 @@ export function getMetaBoxesPerLocation( state, location ) {
  */
 export const getAllMetaBoxes = createSelector(
 	( state ) => {
-		return values( state.metaBoxes.locations ).flat();
+		return Object.values( state.metaBoxes.locations ).flat();
 	},
 	( state ) => [ state.metaBoxes.locations ]
 );
