diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index e4bac1d2dbac8a..4f26fe1e22dee8 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -51,6 +51,7 @@
 -   `BorderControl` and `BorderBoxControl`: replace temporary types with `Popover`'s types ([#43823](https://github.com/WordPress/gutenberg/pull/43823/)).
 -   `DimensionControl`: Refactor tests to `@testing-library/react` ([#43916](https://github.com/WordPress/gutenberg/pull/43916)).
 -   `withFilters`: Refactor tests to `@testing-library/react` ([#44017](https://github.com/WordPress/gutenberg/pull/44017)).
+-   `IsolatedEventContainer`: Refactor tests to `@testing-library/react` ([#44073](https://github.com/WordPress/gutenberg/pull/44073)).
 -   `KeyboardShortcuts`: Refactor tests to `@testing-library/react` ([#44075](https://github.com/WordPress/gutenberg/pull/44075)).
 
 ## 20.0.0 (2022-08-24)
diff --git a/packages/components/src/isolated-event-container/test/index.js b/packages/components/src/isolated-event-container/test/index.js
index 9a83885a61c1bf..1667d69b4df0dd 100644
--- a/packages/components/src/isolated-event-container/test/index.js
+++ b/packages/components/src/isolated-event-container/test/index.js
@@ -1,7 +1,8 @@
 /**
  * External dependencies
  */
-import { shallow } from 'enzyme';
+import { render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 /**
  * Internal dependencies
@@ -9,21 +10,62 @@ import { shallow } from 'enzyme';
 import IsolatedEventContainer from '../';
 
 describe( 'IsolatedEventContainer', () => {
-	it( 'should pass props to container', () => {
-		const isolated = shallow(
-			<IsolatedEventContainer className="test" onClick="click" />
+	it( 'should pass props to container', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+		const clickHandler = jest.fn();
+		render(
+			<IsolatedEventContainer
+				title="Container"
+				className="test"
+				onClick={ clickHandler }
+			/>
 		);
 
-		expect( isolated.hasClass( 'test' ) ).toBe( true );
-		expect( isolated.prop( 'onClick' ) ).toBe( 'click' );
+		const container = screen.getByTitle( 'Container' );
+		expect( container ).toHaveClass( 'test' );
+
+		await user.click( container );
+
+		expect( clickHandler ).toHaveBeenCalledTimes( 1 );
 		expect( console ).toHaveWarned();
 	} );
 
-	it( 'should stop mousedown event propagation', () => {
-		const isolated = shallow( <IsolatedEventContainer /> );
-		const event = { stopPropagation: jest.fn() };
+	it( 'should render children', async () => {
+		render(
+			<IsolatedEventContainer title="Container">
+				<p>Child</p>
+			</IsolatedEventContainer>
+		);
+
+		expect(
+			within( screen.getByTitle( 'Container' ) ).getByText( 'Child' )
+		).toBeVisible();
+	} );
+
+	it( 'should stop event propagation only for mousedown, but not for keydown', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+
+		const mousedownHandler = jest.fn();
+		const keydownHandler = jest.fn();
+		render(
+			<button
+				onMouseDown={ mousedownHandler }
+				onKeyDown={ keydownHandler }
+			>
+				<IsolatedEventContainer title="Container" />
+			</button>
+		);
+
+		const container = screen.getByTitle( 'Container' );
+
+		await user.click( container );
+		await user.keyboard( '[Enter]' );
 
-		isolated.simulate( 'mousedown', event );
-		expect( event.stopPropagation ).toHaveBeenCalled();
+		expect( mousedownHandler ).not.toHaveBeenCalled();
+		expect( keydownHandler ).toHaveBeenCalledTimes( 1 );
 	} );
 } );
