diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index e5290611a93e2c..ca8ac8060ca9fe 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -53,6 +53,7 @@
 -   `TextControl`: Set Storybook control types on `help`, `label` and `type` ([#45405](https://github.com/WordPress/gutenberg/pull/45405)).
 -   `Autocomplete`: use Popover's new `placement` prop instead of legacy `position` prop ([#44396](https://github.com/WordPress/gutenberg/pull/44396/)).
 -   `FontSizePicker`: Add more comprehensive tests ([#45298](https://github.com/WordPress/gutenberg/pull/45298)).
+-   `FontSizePicker`: Refactor to use components instead of helper functions ([#44891](https://github.com/WordPress/gutenberg/pull/44891)).
 
 ## 21.3.0 (2022-10-19)
 
diff --git a/packages/components/src/font-size-picker/font-size-picker-select.tsx b/packages/components/src/font-size-picker/font-size-picker-select.tsx
new file mode 100644
index 00000000000000..192a0f5632f715
--- /dev/null
+++ b/packages/components/src/font-size-picker/font-size-picker-select.tsx
@@ -0,0 +1,87 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import CustomSelectControl from '../custom-select-control';
+import { parseQuantityAndUnitFromRawValue } from '../unit-control';
+import type {
+	FontSizePickerSelectProps,
+	FontSizePickerSelectOption,
+} from './types';
+
+const DEFAULT_OPTION: FontSizePickerSelectOption = {
+	key: 'default',
+	name: __( 'Default' ),
+	value: undefined,
+};
+
+const CUSTOM_OPTION: FontSizePickerSelectOption = {
+	key: 'custom',
+	name: __( 'Custom' ),
+};
+
+const FontSizePickerSelect = ( props: FontSizePickerSelectProps ) => {
+	const {
+		fontSizes,
+		value,
+		disableCustomFontSizes,
+		size,
+		onChange,
+		onSelectCustom,
+	} = props;
+
+	const options: FontSizePickerSelectOption[] = [
+		DEFAULT_OPTION,
+		...fontSizes.map( ( fontSize ) => {
+			const [ quantity ] = parseQuantityAndUnitFromRawValue(
+				fontSize.size
+			);
+			return {
+				key: fontSize.slug,
+				name: fontSize.name || fontSize.slug,
+				value: fontSize.size,
+				__experimentalHint:
+					quantity !== undefined ? String( quantity ) : undefined,
+			};
+		} ),
+		...( disableCustomFontSizes ? [] : [ CUSTOM_OPTION ] ),
+	];
+
+	const selectedOption = value
+		? options.find( ( option ) => option.value === value ) ?? CUSTOM_OPTION
+		: DEFAULT_OPTION;
+
+	return (
+		<CustomSelectControl
+			__nextUnconstrainedWidth
+			className="components-font-size-picker__select"
+			label={ __( 'Font size' ) }
+			hideLabelFromVision
+			describedBy={ sprintf(
+				// translators: %s: Currently selected font size.
+				__( 'Currently selected font size: %s' ),
+				selectedOption.name
+			) }
+			options={ options }
+			value={ selectedOption }
+			onChange={ ( {
+				selectedItem,
+			}: {
+				selectedItem: FontSizePickerSelectOption;
+			} ) => {
+				if ( selectedItem === CUSTOM_OPTION ) {
+					onSelectCustom();
+				} else {
+					onChange( selectedItem.value );
+				}
+			} }
+			size={ size }
+		/>
+	);
+};
+
+export default FontSizePickerSelect;
diff --git a/packages/components/src/font-size-picker/font-size-picker-toggle-group.tsx b/packages/components/src/font-size-picker/font-size-picker-toggle-group.tsx
new file mode 100644
index 00000000000000..a7aed96e76ae6f
--- /dev/null
+++ b/packages/components/src/font-size-picker/font-size-picker-toggle-group.tsx
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ToggleGroupControl,
+	ToggleGroupControlOption,
+} from '../toggle-group-control';
+import type { FontSizePickerToggleGroupProps } from './types';
+
+/**
+ * In case we have at most five font sizes, show a `T-shirt size` alias as a
+ * label of the font size. The label assumes that the font sizes are ordered
+ * accordingly - from smallest to largest.
+ */
+const FONT_SIZES_ALIASES = [
+	/* translators: S stands for 'small' and is a size label. */
+	__( 'S' ),
+	/* translators: M stands for 'medium' and is a size label. */
+	__( 'M' ),
+	/* translators: L stands for 'large' and is a size label. */
+	__( 'L' ),
+	/* translators: XL stands for 'extra large' and is a size label. */
+	__( 'XL' ),
+	/* translators: XXL stands for 'extra extra large' and is a size label. */
+	__( 'XXL' ),
+];
+
+const FontSizePickerToggleGroup = ( props: FontSizePickerToggleGroupProps ) => {
+	const { fontSizes, value, __nextHasNoMarginBottom, size, onChange } = props;
+	return (
+		<ToggleGroupControl
+			__nextHasNoMarginBottom={ __nextHasNoMarginBottom }
+			label={ __( 'Font size' ) }
+			hideLabelFromVision
+			value={ value }
+			onChange={ onChange }
+			isBlock
+			size={ size }
+		>
+			{ fontSizes.map( ( fontSize, index ) => (
+				<ToggleGroupControlOption
+					key={ fontSize.slug }
+					value={ fontSize.size }
+					label={ FONT_SIZES_ALIASES[ index ] }
+					aria-label={ fontSize.name || FONT_SIZES_ALIASES[ index ] }
+					showTooltip
+				/>
+			) ) }
+		</ToggleGroupControl>
+	);
+};
+
+export default FontSizePickerToggleGroup;
diff --git a/packages/components/src/font-size-picker/index.tsx b/packages/components/src/font-size-picker/index.tsx
index 5adb5b75e9abda..9c4c1cac988850 100644
--- a/packages/components/src/font-size-picker/index.tsx
+++ b/packages/components/src/font-size-picker/index.tsx
@@ -7,7 +7,7 @@ import type { ForwardedRef } from 'react';
  * WordPress dependencies
  */
 import deprecated from '@wordpress/deprecated';
-import { __, sprintf } from '@wordpress/i18n';
+import { __ } from '@wordpress/i18n';
 import { settings } from '@wordpress/icons';
 import { useState, useMemo, forwardRef } from '@wordpress/element';
 
@@ -22,24 +22,10 @@ import {
 	parseQuantityAndUnitFromRawValue,
 	useCustomUnits,
 } from '../unit-control';
-import CustomSelectControl from '../custom-select-control';
 import { VisuallyHidden } from '../visually-hidden';
-import {
-	ToggleGroupControl,
-	ToggleGroupControlOption,
-} from '../toggle-group-control';
-import {
-	getFontSizeOptions,
-	getSelectedOption,
-	isSimpleCssValue,
-	CUSTOM_FONT_SIZE,
-} from './utils';
+import { isSimpleCssValue } from './utils';
 import { HStack } from '../h-stack';
-import type {
-	FontSizePickerProps,
-	FontSizeSelectOption,
-	FontSizeToggleGroupOption,
-} from './types';
+import type { FontSizePickerProps } from './types';
 import {
 	Container,
 	HeaderHint,
@@ -48,6 +34,8 @@ import {
 	ResetButton,
 } from './styles';
 import { Spacer } from '../spacer';
+import FontSizePickerSelect from './font-size-picker-select';
+import FontSizePickerToggleGroup from './font-size-picker-toggle-group';
 
 const UnforwardedFontSizePicker = (
 	props: FontSizePickerProps,
@@ -65,6 +53,7 @@ const UnforwardedFontSizePicker = (
 		withSlider = false,
 		withReset = true,
 	} = props;
+
 	if ( ! __nextHasNoMarginBottom ) {
 		deprecated( 'Bottom margin styles for wp.components.FontSizePicker', {
 			since: '6.1',
@@ -77,28 +66,16 @@ const UnforwardedFontSizePicker = (
 		availableUnits: [ 'px', 'em', 'rem' ],
 	} );
 
-	/**
-	 * The main font size UI displays a toggle group when the presets are less
-	 * than six and a select control when they are more.
-	 */
-	const fontSizesContainComplexValues = fontSizes.some(
-		( { size: sizeArg } ) => ! isSimpleCssValue( sizeArg )
-	);
 	const shouldUseSelectControl = fontSizes.length > 5;
-	const options = useMemo(
-		() =>
-			getFontSizeOptions(
-				shouldUseSelectControl,
-				fontSizes,
-				disableCustomFontSizes
-			),
-		[ shouldUseSelectControl, fontSizes, disableCustomFontSizes ]
+	const selectedFontSize = fontSizes.find(
+		( fontSize ) => fontSize.size === value
 	);
-	const selectedOption = getSelectedOption( fontSizes, value );
-	const isCustomValue = selectedOption.slug === CUSTOM_FONT_SIZE;
+	const isCustomValue = !! value && ! selectedFontSize;
+
 	const [ showCustomValueControl, setShowCustomValueControl ] = useState(
 		! disableCustomFontSizes && isCustomValue
 	);
+
 	const headerHint = useMemo( () => {
 		if ( showCustomValueControl ) {
 			return `(${ __( 'Custom' ) })`;
@@ -113,22 +90,30 @@ const UnforwardedFontSizePicker = (
 				`(${ value })`
 			);
 		}
+
 		if ( shouldUseSelectControl ) {
 			return (
-				selectedOption?.size !== undefined &&
-				isSimpleCssValue( selectedOption?.size ) &&
-				`(${ selectedOption?.size })`
+				selectedFontSize?.size !== undefined &&
+				isSimpleCssValue( selectedFontSize?.size ) &&
+				`(${ selectedFontSize?.size })`
 			);
 		}
 
+		if ( ! selectedFontSize ) {
+			return __( 'Default' );
+		}
+
 		// Calculate the `hint` for toggle group control.
-		let hint = selectedOption?.name || selectedOption.slug;
+		let hint = selectedFontSize.name || selectedFontSize.slug;
+		const fontSizesContainComplexValues = fontSizes.some(
+			( fontSize ) => ! isSimpleCssValue( fontSize.size )
+		);
 		if (
 			! fontSizesContainComplexValues &&
-			typeof selectedOption.size === 'string'
+			typeof selectedFontSize.size === 'string'
 		) {
 			const [ , unit ] = parseQuantityAndUnitFromRawValue(
-				selectedOption.size,
+				selectedFontSize.size,
 				units
 			);
 			hint += `(${ unit })`;
@@ -136,26 +121,17 @@ const UnforwardedFontSizePicker = (
 		return hint;
 	}, [
 		showCustomValueControl,
-		selectedOption?.name,
-		selectedOption?.size,
-		value,
 		isCustomValue,
+		selectedFontSize,
+		value,
 		shouldUseSelectControl,
-		fontSizesContainComplexValues,
+		fontSizes,
 	] );
 
-	if ( ! options ) {
+	if ( fontSizes.length === 0 && disableCustomFontSizes ) {
 		return null;
 	}
 
-	// This is used for select control only. We need to add support
-	// for ToggleGroupControl.
-	const currentFontSizeSR = sprintf(
-		// translators: %s: Currently selected font size.
-		__( 'Currently selected font size: %s' ),
-		selectedOption.name
-	);
-
 	// If neither the value or first font size is a string, then FontSizePicker
 	// operates in a legacy "unitless" mode where UnitControl can only be used
 	// to select px values and onChange() is always called with number values.
@@ -210,43 +186,31 @@ const UnforwardedFontSizePicker = (
 				{ !! fontSizes.length &&
 					shouldUseSelectControl &&
 					! showCustomValueControl && (
-						<CustomSelectControl
-							__nextUnconstrainedWidth
-							className="components-font-size-picker__select"
-							label={ __( 'Font size' ) }
-							hideLabelFromVision
-							describedBy={ currentFontSizeSR }
-							options={ options as FontSizeSelectOption[] }
-							value={ ( options as FontSizeSelectOption[] ).find(
-								( option ) => option.key === selectedOption.slug
-							) }
-							onChange={ ( {
-								selectedItem,
-							}: {
-								selectedItem: FontSizeSelectOption;
-							} ) => {
-								if ( selectedItem.size === undefined ) {
+						<FontSizePickerSelect
+							fontSizes={ fontSizes }
+							value={ value }
+							disableCustomFontSizes={ disableCustomFontSizes }
+							size={ size }
+							onChange={ ( newValue ) => {
+								if ( newValue === undefined ) {
 									onChange?.( undefined );
 								} else {
 									onChange?.(
-										hasUnits
-											? selectedItem.size
-											: Number( selectedItem.size )
+										hasUnits ? newValue : Number( newValue )
 									);
 								}
-								if ( selectedItem.key === CUSTOM_FONT_SIZE ) {
-									setShowCustomValueControl( true );
-								}
 							} }
-							size={ size }
+							onSelectCustom={ () =>
+								setShowCustomValueControl( true )
+							}
 						/>
 					) }
 				{ ! shouldUseSelectControl && ! showCustomValueControl && (
-					<ToggleGroupControl
-						__nextHasNoMarginBottom={ __nextHasNoMarginBottom }
-						label={ __( 'Font size' ) }
-						hideLabelFromVision
+					<FontSizePickerToggleGroup
+						fontSizes={ fontSizes }
 						value={ value }
+						__nextHasNoMarginBottom={ __nextHasNoMarginBottom }
+						size={ size }
 						onChange={ ( newValue ) => {
 							if ( newValue === undefined ) {
 								onChange?.( undefined );
@@ -256,21 +220,7 @@ const UnforwardedFontSizePicker = (
 								);
 							}
 						} }
-						isBlock
-						size={ size }
-					>
-						{ ( options as FontSizeToggleGroupOption[] ).map(
-							( option ) => (
-								<ToggleGroupControlOption
-									key={ option.key }
-									value={ option.value }
-									label={ option.label }
-									aria-label={ option.name }
-									showTooltip={ true }
-								/>
-							)
-						) }
-					</ToggleGroupControl>
+					/>
 				) }
 				{ ! disableCustomFontSizes && showCustomValueControl && (
 					<Flex className="components-font-size-picker__custom-size-control">
diff --git a/packages/components/src/font-size-picker/test/index.tsx b/packages/components/src/font-size-picker/test/index.tsx
index 25a21da92527e0..cdcce2db1cc715 100644
--- a/packages/components/src/font-size-picker/test/index.tsx
+++ b/packages/components/src/font-size-picker/test/index.tsx
@@ -126,6 +126,26 @@ describe( 'FontSizePicker', () => {
 			expect( options[ 7 ] ).toHaveAccessibleName( 'Custom' );
 		} );
 
+		test.each( [
+			{ value: undefined, option: 'Default' },
+			{ value: '', option: 'Default' },
+			{ value: '8px', option: 'Tiny' },
+		] )(
+			'defaults to $option when value is $value',
+			( { value, option } ) => {
+				render(
+					<FontSizePicker
+						__nextHasNoMarginBottom
+						fontSizes={ fontSizes }
+						value={ value }
+					/>
+				);
+				expect(
+					screen.getByRole( 'button', { name: 'Font size' } )
+				).toHaveTextContent( option );
+			}
+		);
+
 		test.each( [
 			{ value: undefined, expectedLabel: 'Size' },
 			{ value: '8px', expectedLabel: 'Size (8px)' },
@@ -204,8 +224,7 @@ describe( 'FontSizePicker', () => {
 				screen.getByRole( 'option', { name: 'Custom' } )
 			);
 			expect( screen.getByLabelText( 'Custom' ) ).toBeInTheDocument();
-			// TODO: onChange() shouldn't be called.
-			//expect( onChange ).not.toHaveBeenCalled();
+			expect( onChange ).not.toHaveBeenCalled();
 		} );
 
 		commonTests( fontSizes );
@@ -298,6 +317,7 @@ describe( 'FontSizePicker', () => {
 			expect( onChange ).toHaveBeenCalledWith( '16px' );
 		} );
 
+		commonToggleGroupTests( fontSizes );
 		commonTests( fontSizes );
 	} );
 
@@ -399,10 +419,54 @@ describe( 'FontSizePicker', () => {
 			}
 		);
 
+		commonToggleGroupTests( fontSizes );
 		commonTests( fontSizes );
 	} );
 
+	function commonToggleGroupTests( fontSizes: FontSize[] ) {
+		it( 'defaults to M when value is 16px', () => {
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+					value={ fontSizes[ 0 ].size }
+				/>
+			);
+			expect(
+				screen.getByRole( 'radio', { checked: true } )
+			).toHaveTextContent( 'S' );
+		} );
+
+		test.each( [ undefined, '' ] )(
+			'has no selection when value is %p',
+			( value ) => {
+				render(
+					<FontSizePicker
+						__nextHasNoMarginBottom
+						fontSizes={ fontSizes }
+						value={ value }
+					/>
+				);
+				expect( screen.getByRole( 'radiogroup' ) ).toBeInTheDocument();
+				expect(
+					screen.queryByRole( 'radio', { checked: true } )
+				).not.toBeInTheDocument();
+			}
+		);
+	}
+
 	function commonTests( fontSizes: FontSize[] ) {
+		it( 'shows custom input when value is unknown', () => {
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+					value="3px"
+				/>
+			);
+			expect( screen.getByLabelText( 'Custom' ) ).toBeInTheDocument();
+		} );
+
 		it( 'allows custom values by default', async () => {
 			const user = userEvent.setup( {
 				advanceTimers: jest.advanceTimersByTime,
diff --git a/packages/components/src/font-size-picker/test/utils.ts b/packages/components/src/font-size-picker/test/utils.ts
index 829927051b4622..91205062d57021 100644
--- a/packages/components/src/font-size-picker/test/utils.ts
+++ b/packages/components/src/font-size-picker/test/utils.ts
@@ -1,7 +1,7 @@
 /**
  * Internal dependencies
  */
-import { isSimpleCssValue, getToggleGroupOptions } from '../utils';
+import { isSimpleCssValue } from '../utils';
 
 const simpleCSSCases: [ number | string, boolean ][] = [
 	// Test integers and non-integers.
@@ -36,74 +36,3 @@ describe( 'isSimpleCssValue', () => {
 		}
 	);
 } );
-
-describe( 'getToggleGroupOptions', () => {
-	test( 'should assign t-shirt sizes by default up until the aliases length', () => {
-		const optionsArray = [
-			{
-				slug: '1',
-				size: '1',
-				name: '1',
-			},
-			{
-				slug: '2',
-				size: '2',
-				name: '2',
-			},
-			{
-				slug: '3',
-				size: '3',
-				name: '3',
-			},
-			{
-				slug: '4',
-				size: '4',
-				name: '4',
-			},
-			{
-				slug: '5',
-				size: '5',
-			},
-		];
-		expect(
-			getToggleGroupOptions( optionsArray, [
-				'S',
-				'M',
-				'L',
-				'XL',
-				'XXL',
-			] )
-		).toEqual( [
-			{
-				key: '1',
-				label: 'S',
-				name: '1',
-				value: '1',
-			},
-			{
-				key: '2',
-				label: 'M',
-				name: '2',
-				value: '2',
-			},
-			{
-				key: '3',
-				label: 'L',
-				name: '3',
-				value: '3',
-			},
-			{
-				key: '4',
-				label: 'XL',
-				name: '4',
-				value: '4',
-			},
-			{
-				key: '5',
-				label: 'XXL',
-				name: 'XXL',
-				value: '5',
-			},
-		] );
-	} );
-} );
diff --git a/packages/components/src/font-size-picker/types.ts b/packages/components/src/font-size-picker/types.ts
index e61988b6d4fbfc..c410ac172d63dc 100644
--- a/packages/components/src/font-size-picker/types.ts
+++ b/packages/components/src/font-size-picker/types.ts
@@ -1,8 +1,3 @@
-/**
- * External dependencies
- */
-import type { ReactNode } from 'react';
-
 export type FontSizePickerProps = {
 	/**
 	 * If `true`, it will not be possible to choose a custom fontSize. The user
@@ -81,18 +76,29 @@ export type FontSize = {
 	slug: string;
 };
 
-export type FontSizeOption = Omit< FontSize, 'size' > &
-	Partial< Pick< FontSize, 'size' > >;
-
-export type FontSizeSelectOption = Pick< FontSizeOption, 'size' > & {
-	key: string;
-	name?: string;
-	__experimentalHint: ReactNode;
+export type FontSizePickerSelectProps = Pick<
+	FontSizePickerProps,
+	'value' | 'size'
+> & {
+	fontSizes: NonNullable< FontSizePickerProps[ 'fontSizes' ] >;
+	disableCustomFontSizes: NonNullable<
+		FontSizePickerProps[ 'disableCustomFontSizes' ]
+	>;
+	onChange: NonNullable< FontSizePickerProps[ 'onChange' ] >;
+	onSelectCustom: () => void;
 };
 
-export type FontSizeToggleGroupOption = {
+export type FontSizePickerSelectOption = {
 	key: string;
-	value: number | string;
-	label: string;
 	name: string;
+	value?: FontSize[ 'size' ];
+	__experimentalHint?: string;
+};
+
+export type FontSizePickerToggleGroupProps = Pick<
+	FontSizePickerProps,
+	'value' | 'size' | '__nextHasNoMarginBottom'
+> & {
+	fontSizes: NonNullable< FontSizePickerProps[ 'fontSizes' ] >;
+	onChange: NonNullable< FontSizePickerProps[ 'onChange' ] >;
 };
diff --git a/packages/components/src/font-size-picker/utils.ts b/packages/components/src/font-size-picker/utils.ts
index dfd578b06827a3..3f0c6019da8067 100644
--- a/packages/components/src/font-size-picker/utils.ts
+++ b/packages/components/src/font-size-picker/utils.ts
@@ -6,42 +6,7 @@ import { __ } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
-import type {
-	FontSize,
-	FontSizeOption,
-	FontSizeSelectOption,
-	FontSizeToggleGroupOption,
-	FontSizePickerProps,
-} from './types';
-
-const DEFAULT_FONT_SIZE = 'default';
-const DEFAULT_FONT_SIZE_OPTION = {
-	slug: DEFAULT_FONT_SIZE,
-	name: __( 'Default' ),
-};
-export const CUSTOM_FONT_SIZE = 'custom';
-const CUSTOM_FONT_SIZE_OPTION = {
-	slug: CUSTOM_FONT_SIZE,
-	name: __( 'Custom' ),
-};
-
-/**
- * In case we have at most five font sizes, show a `T-shirt size`
- * alias as a label of the font size. The label assumes that the font sizes
- * are ordered accordingly - from smallest to largest.
- */
-const FONT_SIZES_ALIASES = [
-	/* translators: S stands for 'small' and is a size label. */
-	__( 'S' ),
-	/* translators: M stands for 'medium' and is a size label. */
-	__( 'M' ),
-	/* translators: L stands for 'large' and is a size label. */
-	__( 'L' ),
-	/* translators: XL stands for 'extra large' and is a size label. */
-	__( 'XL' ),
-	/* translators: XXL stands for 'extra extra large' and is a size label. */
-	__( 'XXL' ),
-];
+import type { FontSizePickerProps } from './types';
 
 /**
  * Some themes use css vars for their font sizes, so until we
@@ -56,77 +21,3 @@ export function isSimpleCssValue(
 	const sizeRegex = /^[\d\.]+(px|em|rem|vw|vh|%)?$/i;
 	return sizeRegex.test( String( value ) );
 }
-
-/**
- * Return font size options in the proper format depending
- * on the currently used control (select, toggle group).
- *
- * @param  useSelectControl       Whether to use a select control.
- * @param  optionsArray           Array of available font sizes objects.
- * @param  disableCustomFontSizes Flag that indicates if custom font sizes are disabled.
- * @return Array of font sizes in proper format for the used control.
- */
-export function getFontSizeOptions(
-	useSelectControl: boolean,
-	optionsArray: FontSize[],
-	disableCustomFontSizes: boolean
-): FontSizeSelectOption[] | FontSizeToggleGroupOption[] | null {
-	if ( disableCustomFontSizes && ! optionsArray.length ) {
-		return null;
-	}
-	return useSelectControl
-		? getSelectOptions( optionsArray, disableCustomFontSizes )
-		: getToggleGroupOptions( optionsArray );
-}
-
-function getSelectOptions(
-	optionsArray: FontSize[],
-	disableCustomFontSizes: boolean
-): FontSizeSelectOption[] {
-	const options: FontSizeOption[] = [
-		DEFAULT_FONT_SIZE_OPTION,
-		...optionsArray,
-		...( disableCustomFontSizes ? [] : [ CUSTOM_FONT_SIZE_OPTION ] ),
-	];
-	return options.map( ( { slug, name, size } ) => ( {
-		key: slug,
-		name: name || slug,
-		size,
-		__experimentalHint:
-			size && isSimpleCssValue( size ) && parseFloat( String( size ) ),
-	} ) );
-}
-
-/**
- * Build options for the toggle group options.
- *
- * @param  optionsArray An array of font size options.
- * @param  labelAliases An array of alternative labels.
- * @return Remapped optionsArray.
- */
-export function getToggleGroupOptions(
-	optionsArray: FontSize[],
-	labelAliases: string[] = FONT_SIZES_ALIASES
-): FontSizeToggleGroupOption[] {
-	return optionsArray.map( ( { slug, size, name }, index ) => {
-		return {
-			key: slug,
-			value: size,
-			label: labelAliases[ index ],
-			name: name || labelAliases[ index ],
-		};
-	} );
-}
-
-export function getSelectedOption(
-	fontSizes: FontSize[],
-	value: FontSizePickerProps[ 'value' ]
-): FontSizeOption {
-	if ( ! value ) {
-		return DEFAULT_FONT_SIZE_OPTION;
-	}
-	return (
-		fontSizes.find( ( font ) => font.size === value ) ||
-		CUSTOM_FONT_SIZE_OPTION
-	);
-}
