diff --git a/packages/block-editor/src/components/block-selection-clearer/test/index.js b/packages/block-editor/src/components/block-selection-clearer/test/index.js
index 78b1ab84a2dd77..3886055aebaf4f 100644
--- a/packages/block-editor/src/components/block-selection-clearer/test/index.js
+++ b/packages/block-editor/src/components/block-selection-clearer/test/index.js
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { fireEvent, render } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 
 /**
  * WordPress dependencies
@@ -42,12 +42,12 @@ describe( 'BlockSelectionClearer component', () => {
 			clearSelectedBlock: mockClearSelectedBlock,
 		} ) );
 
-		const { queryByRole } = render(
+		render(
 			<BlockSelectionClearer>
 				<button>Not a block</button>
 			</BlockSelectionClearer>
 		);
-		const button = queryByRole( 'button' );
+		const button = screen.getByRole( 'button' );
 		fireEvent.mouseDown( button.parentElement );
 
 		expect( mockClearSelectedBlock ).toBeCalled();
@@ -63,12 +63,12 @@ describe( 'BlockSelectionClearer component', () => {
 			clearSelectedBlock: mockClearSelectedBlock,
 		} ) );
 
-		const { queryByRole } = render(
+		render(
 			<BlockSelectionClearer>
 				<button>Not a block</button>
 			</BlockSelectionClearer>
 		);
-		const button = queryByRole( 'button' );
+		const button = screen.getByRole( 'button' );
 		fireEvent.mouseDown( button.parentElement );
 
 		expect( mockClearSelectedBlock ).toBeCalled();
@@ -81,12 +81,12 @@ describe( 'BlockSelectionClearer component', () => {
 			clearSelectedBlock: mockClearSelectedBlock,
 		} ) );
 
-		const { queryByRole } = render(
+		render(
 			<BlockSelectionClearer>
 				<button>Not a block</button>
 			</BlockSelectionClearer>
 		);
-		const button = queryByRole( 'button' );
+		const button = screen.getByRole( 'button' );
 		fireEvent.mouseDown( button.parentElement );
 
 		expect( mockClearSelectedBlock ).not.toBeCalled();
@@ -105,12 +105,12 @@ describe( 'BlockSelectionClearer component', () => {
 			clearSelectedBlock: mockClearSelectedBlock,
 		} ) );
 
-		const { queryByRole } = render(
+		render(
 			<BlockSelectionClearer>
 				<button>Not a block</button>
 			</BlockSelectionClearer>
 		);
-		const button = queryByRole( 'button' );
+		const button = screen.getByRole( 'button' );
 		fireEvent.mouseDown( button.parentElement );
 
 		expect( mockClearSelectedBlock ).not.toBeCalled();
diff --git a/packages/components/src/sandbox/test/index.js b/packages/components/src/sandbox/test/index.js
index c1c096e0df5262..d0dadd107cddd9 100644
--- a/packages/components/src/sandbox/test/index.js
+++ b/packages/components/src/sandbox/test/index.js
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { act, fireEvent, render } from '@testing-library/react';
+import { act, fireEvent, render, screen } from '@testing-library/react';
 
 /**
  * WordPress dependencies
@@ -55,7 +55,7 @@ describe( 'Sandbox', () => {
 		);
 
 		act( () => {
-			fireEvent.click( result.getByRole( 'button' ) );
+			fireEvent.click( screen.getByRole( 'button' ) );
 		} );
 
 		sandboxedIframe =
diff --git a/packages/core-data/src/hooks/test/use-query-select.js b/packages/core-data/src/hooks/test/use-query-select.js
index 3a63ff010b7446..217a76c57f533e 100644
--- a/packages/core-data/src/hooks/test/use-query-select.js
+++ b/packages/core-data/src/hooks/test/use-query-select.js
@@ -10,7 +10,7 @@ import {
 /**
  * External dependencies
  */
-import { act, render } from '@testing-library/react';
+import { act, render, screen } from '@testing-library/react';
 
 /**
  * Internal dependencies
@@ -52,7 +52,7 @@ describe( 'useQuerySelect', () => {
 		const TestComponent = jest
 			.fn()
 			.mockImplementation( getTestComponent( selectSpy, 'keyName' ) );
-		const testInstance = render(
+		render(
 			<RegistryProvider value={ registry }>
 				<TestComponent keyName="foo" />
 			</RegistryProvider>
@@ -64,7 +64,7 @@ describe( 'useQuerySelect', () => {
 		expect( TestComponent ).toHaveBeenCalledTimes( 2 );
 
 		// ensure expected state was rendered
-		expect( testInstance.getByText( 'bar' ) ).toBeInTheDocument();
+		expect( screen.getByText( 'bar' ) ).toBeInTheDocument();
 	} );
 
 	it( 'uses memoized selectors', () => {
diff --git a/packages/data/src/components/use-select/test/suspense.js b/packages/data/src/components/use-select/test/suspense.js
index c75d368de44ff1..032db07bdd1bdb 100644
--- a/packages/data/src/components/use-select/test/suspense.js
+++ b/packages/data/src/components/use-select/test/suspense.js
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 
 /**
  * WordPress dependencies
@@ -109,8 +109,8 @@ describe( 'useSuspenseSelect', () => {
 			</RegistryProvider>
 		);
 
-		const rendered = render( <App /> );
-		await rendered.findByLabelText( 'loaded' );
+		render( <App /> );
+		await screen.findByLabelText( 'loaded' );
 
 		// Verify there were 3 attempts to render. Suspended twice because of
 		// `getToken` and `getData` selectors not being resolved, and then finally
@@ -156,8 +156,8 @@ describe( 'useSuspenseSelect', () => {
 			</RegistryProvider>
 		);
 
-		const rendered = render( <App /> );
-		const label = await rendered.findByLabelText( 'error' );
+		render( <App /> );
+		const label = await screen.findByLabelText( 'error' );
 		expect( label ).toHaveTextContent( 'resolution failed' );
 		expect( console ).toHaveErrored();
 	} );
