diff --git a/packages/components/src/sandbox/test/index.js b/packages/components/src/sandbox/test/index.js
index d91dce1842199a..f25bdca675cbfd 100644
--- a/packages/components/src/sandbox/test/index.js
+++ b/packages/components/src/sandbox/test/index.js
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { fireEvent, render, screen } from '@testing-library/react';
+import { fireEvent, render, screen, within } from '@testing-library/react';
 
 /**
  * WordPress dependencies
@@ -16,15 +16,15 @@ import Sandbox from '../';
 describe( 'Sandbox', () => {
 	const TestWrapper = () => {
 		const [ html, setHtml ] = useState(
-			// MuatationObserver implementation from JSDom does not work as intended
+			// MutationObserver implementation from JSDom does not work as intended
 			// with iframes so we need to ignore it for the time being.
 			'<script type="text/javascript">window.MutationObserver = null;</script>' +
-				'<iframe class="mock-iframe" src="https://super.embed"></iframe>'
+				'<iframe title="Mock Iframe" src="https://super.embed"></iframe>'
 		);
 
 		const updateHtml = () => {
 			setHtml(
-				'<iframe class="mock-iframe" src="https://another.super.embed"></iframe>'
+				'<iframe title="Mock Iframe" src="https://another.super.embed"></iframe>'
 			);
 		};
 
@@ -33,18 +33,19 @@ describe( 'Sandbox', () => {
 				<button onClick={ updateHtml } className="mock-button">
 					Mock Button
 				</button>
-				<Sandbox html={ html } />
+				<Sandbox html={ html } title="Sandbox Title" />
 			</div>
 		);
 	};
 
 	it( 'should rerender with new emdeded content if html prop changes', () => {
-		const { container } = render( <TestWrapper /> );
+		render( <TestWrapper /> );
 
-		const iframe = container.querySelector( '.components-sandbox' );
+		const iframe = screen.getByTitle( 'Sandbox Title' );
 
-		let sandboxedIframe =
-			iframe.contentWindow.document.body.querySelector( '.mock-iframe' );
+		let sandboxedIframe = within(
+			iframe.contentWindow.document.body
+		).getByTitle( 'Mock Iframe' );
 
 		expect( sandboxedIframe ).toHaveAttribute(
 			'src',
@@ -53,8 +54,9 @@ describe( 'Sandbox', () => {
 
 		fireEvent.click( screen.getByRole( 'button' ) );
 
-		sandboxedIframe =
-			iframe.contentWindow.document.body.querySelector( '.mock-iframe' );
+		sandboxedIframe = within(
+			iframe.contentWindow.document.body
+		).getByTitle( 'Mock Iframe' );
 
 		expect( sandboxedIframe ).toHaveAttribute(
 			'src',
