diff --git a/packages/components/src/toolbar/stories/index.js b/packages/components/src/toolbar/stories/index.js
index a186c1c0de50ed..39e69da286a9e3 100644
--- a/packages/components/src/toolbar/stories/index.js
+++ b/packages/components/src/toolbar/stories/index.js
@@ -12,15 +12,42 @@ import {
 	link,
 	more,
 	paragraph,
+	arrowUp,
+	arrowDown,
+	arrowLeft,
+	arrowRight,
+	chevronDown,
 } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
-import { Toolbar, ToolbarButton, ToolbarGroup, ToolbarItem } from '..';
-import { SVG, Path, DropdownMenu } from '../../..';
+import {
+	Toolbar,
+	ToolbarButton,
+	ToolbarGroup,
+	ToolbarItem,
+	ToolbarDropdownMenu,
+} from '..';
+import { SVG, Path, DropdownMenu } from '../../';
 
-export default { title: 'Components/Toolbar', component: Toolbar };
+export default {
+	title: 'Components/Toolbar',
+	component: Toolbar,
+	subcomponents: {
+		ToolbarButton,
+		ToolbarGroup,
+		ToolbarItem,
+		ToolbarDropdownMenu,
+	},
+	argTypes: {
+		children: { control: { type: null } },
+	},
+	parameters: {
+		controls: { expanded: true },
+		docs: { source: { state: 'open' } },
+	},
+};
 
 function InlineImageIcon() {
 	return (
@@ -30,13 +57,20 @@ function InlineImageIcon() {
 	);
 }
 
-/* eslint-disable no-restricted-syntax */
-export const _default = () => {
-	return (
-		// id is required for server side rendering
-		<Toolbar label="Options" id="options-toolbar">
+const Template = ( props ) => (
+	<div style={ { height: 280 } }>
+		<Toolbar { ...props } />
+	</div>
+);
+
+export const Default = Template.bind( {} );
+Default.args = {
+	label: 'Options',
+	id: 'options-toolbar',
+	children: (
+		<>
 			<ToolbarGroup>
-				<ToolbarButton icon={ paragraph } label="Paragraph" />
+				<ToolbarButton icon={ paragraph } text="Paragraph" />
 			</ToolbarGroup>
 			<ToolbarGroup>
 				<ToolbarItem>
@@ -98,72 +132,41 @@ export const _default = () => {
 					{ icon: alignRight, title: 'Align right' },
 				] }
 			/>
-		</Toolbar>
-	);
+			<ToolbarDropdownMenu
+				icon={ chevronDown }
+				label="Select a direction"
+				controls={ [
+					{
+						title: 'Up',
+						icon: arrowUp,
+					},
+					{
+						title: 'Right',
+						icon: arrowRight,
+					},
+					{
+						title: 'Down',
+						icon: arrowDown,
+					},
+					{
+						title: 'Left',
+						icon: arrowLeft,
+					},
+				] }
+			/>
+		</>
+	),
 };
 
-export const withoutGroup = () => {
-	return (
-		// id is required for server side rendering
-		<Toolbar label="Options" id="options-toolbar-without-group">
+export const WithoutGroup = Template.bind( {} );
+WithoutGroup.args = {
+	label: 'Options',
+	id: 'options-toolbar-without-group',
+	children: (
+		<>
 			<ToolbarButton icon={ formatBold } label="Bold" isPressed />
 			<ToolbarButton icon={ formatItalic } label="Italic" />
 			<ToolbarButton icon={ link } label="Link" />
-		</Toolbar>
-	);
-};
-/* eslint-enable no-restricted-syntax */
-
-export const toolbars = () => {
-	return (
-		<div>
-			<div style={ { padding: '20px' } }>
-				<h2>Icon-only Toolbar</h2>
-				<Toolbar>
-					<ToolbarButton icon={ formatBold } title="Bold" />
-					<ToolbarButton
-						icon={ formatItalic }
-						title="Italic"
-						isActive
-					/>
-					<ToolbarButton icon={ link } title="Link" />
-				</Toolbar>
-			</div>
-
-			<div style={ { padding: '20px' } }>
-				<h2>Text-only Toolbar</h2>
-				<Toolbar>
-					<ToolbarButton>Bold Format</ToolbarButton>
-					<ToolbarButton isActive>Italic Format</ToolbarButton>
-					<ToolbarButton>Link Format</ToolbarButton>
-				</Toolbar>
-			</div>
-
-			<div style={ { padding: '20px' } }>
-				<h2>Text and Icon Toolbar</h2>
-				<Toolbar>
-					<ToolbarButton icon={ formatBold } title="Bold" />
-					<ToolbarButton isActive>Bold Format</ToolbarButton>
-					<ToolbarButton icon={ formatItalic } title="Italic" />
-					<ToolbarButton>Italic Format</ToolbarButton>
-					<ToolbarButton icon={ link } title="Link" />
-					<ToolbarButton>Link Format</ToolbarButton>
-				</Toolbar>
-			</div>
-
-			<div style={ { padding: '20px' } }>
-				<h2>Single Icon Button Toolbar</h2>
-				<Toolbar>
-					<ToolbarButton icon={ formatBold } title="Bold" />
-				</Toolbar>
-			</div>
-
-			<div style={ { padding: '20px' } }>
-				<h2>Single Text Button toolbar</h2>
-				<Toolbar>
-					<ToolbarButton>Bold Toolbar</ToolbarButton>
-				</Toolbar>
-			</div>
-		</div>
-	);
+		</>
+	),
 };
diff --git a/packages/components/src/toolbar/stories/toolbar-button.js b/packages/components/src/toolbar/stories/toolbar-button.js
deleted file mode 100644
index 3740ed8b2bc10f..00000000000000
--- a/packages/components/src/toolbar/stories/toolbar-button.js
+++ /dev/null
@@ -1,32 +0,0 @@
-/**
- * External dependencies
- */
-import { text } from '@storybook/addon-knobs';
-
-/**
- * WordPress dependencies
- */
-import { wordpress } from '@wordpress/icons';
-
-/**
- * Internal dependencies
- */
-import { Toolbar, ToolbarButton } from '..';
-
-export default {
-	title: 'Components/ToolbarButton',
-	component: ToolbarButton,
-	parameters: {
-		knobs: { disable: false },
-	},
-};
-
-export const _default = () => {
-	const label = text( 'Label', 'This is an example label.' );
-
-	return (
-		<Toolbar label="Example Toolbar">
-			<ToolbarButton icon={ wordpress } label={ label } />
-		</Toolbar>
-	);
-};
diff --git a/packages/components/src/toolbar/stories/toolbar-group.js b/packages/components/src/toolbar/stories/toolbar-group.js
deleted file mode 100644
index d967fc72a63e43..00000000000000
--- a/packages/components/src/toolbar/stories/toolbar-group.js
+++ /dev/null
@@ -1,33 +0,0 @@
-/**
- * WordPress dependencies
- */
-import { formatBold, formatItalic, link } from '@wordpress/icons';
-
-/**
- * Internal dependencies
- */
-import { ToolbarButton, ToolbarGroup } from '..';
-
-export default { title: 'Components/ToolbarGroup', component: ToolbarGroup };
-
-export const _default = () => {
-	return (
-		<ToolbarGroup>
-			<ToolbarButton icon={ formatBold } title="Bold" isActive />
-			<ToolbarButton icon={ formatItalic } title="Italic" />
-			<ToolbarButton icon={ link } title="Link" />
-		</ToolbarGroup>
-	);
-};
-
-export const withControlsProp = () => {
-	return (
-		<ToolbarGroup
-			controls={ [
-				{ icon: formatBold, title: 'Bold', isActive: true },
-				{ icon: formatItalic, title: 'Italic' },
-				{ icon: link, title: 'Link' },
-			] }
-		/>
-	);
-};
