diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index b542190b19912d..d2fb7406d7ea29 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -15,6 +15,8 @@
 -   `SandBox`: Convert to TypeScript ([#46478](https://github.com/WordPress/gutenberg/pull/46478)).
 -   `ResponsiveWrapper`: Convert to TypeScript ([#46480](https://github.com/WordPress/gutenberg/pull/46480)).
 -   `ItemGroup`: migrate Storybook to controls, refactor to TypeScript ([46945](https://github.com/WordPress/gutenberg/pull/46945)).
+-   `Toolbar`: unify Storybook examples under one file, migrate from knobs to controls ([47117](https://github.com/WordPress/gutenberg/pull/47117)).
+-   `DropdownMenu`: migrate Storybook to controls ([47149](https://github.com/WordPress/gutenberg/pull/47149)).
 
 ### Bug Fix
 
diff --git a/packages/components/src/dropdown-menu/README.md b/packages/components/src/dropdown-menu/README.md
index aa930f22edcf18..fc6fc4ba708c9e 100644
--- a/packages/components/src/dropdown-menu/README.md
+++ b/packages/components/src/dropdown-menu/README.md
@@ -205,6 +205,6 @@ Use this object to modify props available for the `NavigableMenu` component that
 
 In some contexts, the arrow down key used to open the dropdown menu might need to be disabled—for example when that key is used to perform another action.
 
--   Type: `Object`
+-   Type: `boolean`
 -   Required: No
 -   Default: `false`
diff --git a/packages/components/src/dropdown-menu/stories/index.js b/packages/components/src/dropdown-menu/stories/index.js
index c70ef559387644..33477854721ee0 100644
--- a/packages/components/src/dropdown-menu/stories/index.js
+++ b/packages/components/src/dropdown-menu/stories/index.js
@@ -1,52 +1,121 @@
-/**
- * External dependencies
- */
-import { text, boolean } from '@storybook/addon-knobs';
-
 /**
  * Internal dependencies
  */
 import DropdownMenu from '../';
+import { MenuGroup, MenuItem } from '../../';
 
 /**
  * WordPress dependencies
  */
-import { menu, arrowUp, arrowDown } from '@wordpress/icons';
+import {
+	menu,
+	arrowUp,
+	arrowDown,
+	chevronDown,
+	more,
+	trash,
+} from '@wordpress/icons';
 
 export default {
 	title: 'Components/DropdownMenu',
 	component: DropdownMenu,
+	argTypes: {
+		className: { control: { type: 'text' } },
+		children: { control: { type: null } },
+		disableOpenOnArrowDown: { control: { type: 'boolean' } },
+		icon: {
+			options: [ 'menu', 'chevronDown', 'more' ],
+			mapping: { menu, chevronDown, more },
+			control: { type: 'select' },
+		},
+		menuProps: {
+			control: { type: 'object' },
+		},
+		noIcons: { control: { type: 'boolean' } },
+		popoverProps: {
+			control: { type: 'object' },
+		},
+		text: { control: { type: 'text' } },
+		toggleProps: {
+			control: { type: 'object' },
+		},
+	},
 	parameters: {
-		knobs: { disable: false },
+		controls: { expanded: true },
+		docs: { source: { state: 'open' } },
 	},
 };
 
-export const _default = () => {
-	const label = text( 'Label', 'Select a direction.' );
-	const firstMenuItemLabel = text( 'First Menu Item Label', 'Up' );
-	const secondMenuItemLabel = text( 'First Menu Item Label', 'Down' );
-	const toggleButtonTootip = boolean(
-		'Show tooltip on a toggle button',
-		true
-	);
+const Template = ( props ) => (
+	<div style={ { height: 150 } }>
+		<DropdownMenu { ...props } />
+	</div>
+);
 
-	const controls = [
+export const Default = Template.bind( {} );
+Default.args = {
+	label: 'Select a direction.',
+	icon: menu,
+	controls: [
 		{
-			title: firstMenuItemLabel,
+			title: 'First Menu Item Label',
 			icon: arrowUp,
+			// eslint-disable-next-line no-console
+			onClick: () => console.log( 'up!' ),
 		},
 		{
-			title: secondMenuItemLabel,
+			title: 'Second Menu Item Label',
 			icon: arrowDown,
+			// eslint-disable-next-line no-console
+			onClick: () => console.log( 'down!' ),
 		},
-	];
+	],
+};
 
-	return (
-		<DropdownMenu
-			icon={ menu }
-			label={ label }
-			controls={ controls }
-			toggleProps={ { showTooltip: toggleButtonTootip } }
-		/>
-	);
+export const WithChildren = Template.bind( {} );
+// Adding custom source because Storybook is not able to show the contents of
+// the `children` prop correctly in the code snippet.
+WithChildren.parameters = {
+	docs: {
+		source: {
+			code: `<DropdownMenu label="Select a direction." icon={ more }>
+  <MenuGroup>
+    <MenuItem icon={ arrowUp } onClick={ onClose }>
+      Move Up
+    </MenuItem>
+    <MenuItem icon={ arrowDown } onClick={ onClose }>
+      Move Down
+    </MenuItem>
+  </MenuGroup>
+  <MenuGroup>
+    <MenuItem icon={ trash } onClick={ onClose }>
+      Remove
+    </MenuItem>
+  </MenuGroup>
+</DropdownMenu>`,
+			language: 'jsx',
+			type: 'auto',
+		},
+	},
+};
+WithChildren.args = {
+	label: 'Select a direction.',
+	icon: more,
+	children: ( { onClose } ) => (
+		<>
+			<MenuGroup>
+				<MenuItem icon={ arrowUp } onClick={ onClose }>
+					Move Up
+				</MenuItem>
+				<MenuItem icon={ arrowDown } onClick={ onClose }>
+					Move Down
+				</MenuItem>
+			</MenuGroup>
+			<MenuGroup>
+				<MenuItem icon={ trash } onClick={ onClose }>
+					Remove
+				</MenuItem>
+			</MenuGroup>
+		</>
+	),
 };
