diff --git a/packages/block-editor/src/components/list-view/block-select-button.js b/packages/block-editor/src/components/list-view/block-select-button.js
index 39baf78d5a7a88..21759d205845aa 100644
--- a/packages/block-editor/src/components/list-view/block-select-button.js
+++ b/packages/block-editor/src/components/list-view/block-select-button.js
@@ -35,6 +35,9 @@ function ListViewBlockSelectButton(
 		onDragStart,
 		onDragEnd,
 		draggable,
+		isExpanded,
+		ariaLabel,
+		ariaDescribedBy,
 	},
 	ref
 ) {
@@ -76,7 +79,9 @@ function ListViewBlockSelectButton(
 				onDragEnd={ onDragEnd }
 				draggable={ draggable }
 				href={ `#block-${ clientId }` }
-				aria-hidden={ true }
+				aria-label={ ariaLabel }
+				aria-describedby={ ariaDescribedBy }
+				aria-expanded={ isExpanded }
 			>
 				<ListViewExpander onClick={ onToggleExpanded } />
 				<BlockIcon icon={ blockInformation?.icon } showColors />
diff --git a/packages/block-editor/src/components/list-view/block.js b/packages/block-editor/src/components/list-view/block.js
index b9ccfdecdfcf90..fb8054bee7986a 100644
--- a/packages/block-editor/src/components/list-view/block.js
+++ b/packages/block-editor/src/components/list-view/block.js
@@ -53,7 +53,6 @@ function ListViewBlock( {
 	path,
 	isExpanded,
 	selectedClientIds,
-	preventAnnouncement,
 	isSyncedBranch,
 } ) {
 	const cellRef = useRef( null );
@@ -90,6 +89,7 @@ function ListViewBlock( {
 	const { toggleBlockHighlight } = useDispatch( blockEditorStore );
 
 	const blockInformation = useBlockDisplayInformation( clientId );
+	const blockTitle = blockInformation?.title || __( 'Untitled' );
 	const blockName = useSelect(
 		( select ) => select( blockEditorStore ).getBlockName( clientId ),
 		[ clientId ]
@@ -111,28 +111,19 @@ function ListViewBlock( {
 		level
 	);
 
-	let blockAriaLabel = __( 'Link' );
-	if ( blockInformation ) {
-		blockAriaLabel = isLocked
-			? sprintf(
-					// translators: %s: The title of the block. This string indicates a link to select the locked block.
-					__( '%s link (locked)' ),
-					blockInformation.title
-			  )
-			: sprintf(
-					// translators: %s: The title of the block. This string indicates a link to select the block.
-					__( '%s link' ),
-					blockInformation.title
-			  );
-	}
-
-	const settingsAriaLabel = blockInformation
+	const blockAriaLabel = isLocked
 		? sprintf(
-				// translators: %s: The title of the block.
-				__( 'Options for %s block' ),
-				blockInformation.title
+				// translators: %s: The title of the block. This string indicates a link to select the locked block.
+				__( '%s (locked)' ),
+				blockTitle
 		  )
-		: __( 'Options' );
+		: blockTitle;
+
+	const settingsAriaLabel = sprintf(
+		// translators: %s: The title of the block.
+		__( 'Options for %s' ),
+		blockTitle
+	);
 
 	const { isTreeGridMounted, expand, collapse, BlockSettingsMenu } =
 		useListViewContext();
@@ -249,18 +240,13 @@ function ListViewBlock( {
 			id={ `list-view-block-${ clientId }` }
 			data-block={ clientId }
 			data-expanded={ canExpand ? isExpanded : undefined }
-			isExpanded={ canExpand ? isExpanded : undefined }
-			aria-selected={ !! isSelected || forceSelectionContentLock }
 			ref={ rowRef }
 		>
 			<TreeGridCell
 				className="block-editor-list-view-block__contents-cell"
 				colSpan={ colSpan }
 				ref={ cellRef }
-				aria-label={ blockAriaLabel }
 				aria-selected={ !! isSelected || forceSelectionContentLock }
-				aria-expanded={ canExpand ? isExpanded : undefined }
-				aria-describedby={ descriptionId }
 			>
 				{ ( { ref, tabIndex, onFocus } ) => (
 					<div className="block-editor-list-view-block__contents-container">
@@ -277,9 +263,10 @@ function ListViewBlock( {
 								currentlyEditingBlockInCanvas ? 0 : tabIndex
 							}
 							onFocus={ onFocus }
-							isExpanded={ isExpanded }
+							isExpanded={ canExpand ? isExpanded : undefined }
 							selectedClientIds={ selectedClientIds }
-							preventAnnouncement={ preventAnnouncement }
+							ariaLabel={ blockAriaLabel }
+							ariaDescribedBy={ descriptionId }
 						/>
 						<div
 							className="block-editor-list-view-block-select-button__description"
diff --git a/packages/block-editor/src/components/list-view/leaf.js b/packages/block-editor/src/components/list-view/leaf.js
index 63c154fd620379..c42d696f32b3c1 100644
--- a/packages/block-editor/src/components/list-view/leaf.js
+++ b/packages/block-editor/src/components/list-view/leaf.js
@@ -51,6 +51,7 @@ const ListViewLeaf = forwardRef(
 				level={ level }
 				positionInSet={ position }
 				setSize={ rowCount }
+				isExpanded={ undefined }
 				{ ...props }
 			>
 				{ children }
diff --git a/packages/block-editor/src/components/list-view/style.scss b/packages/block-editor/src/components/list-view/style.scss
index a04572da98e5f3..3784f3be63eaa1 100644
--- a/packages/block-editor/src/components/list-view/style.scss
+++ b/packages/block-editor/src/components/list-view/style.scss
@@ -15,6 +15,18 @@
 	// Use position relative for row animation.
 	position: relative;
 
+	.components-button {
+		// When a row is expanded, retain the dark color.
+		&[aria-expanded="true"] {
+			color: $gray-900;
+		}
+
+		// Ensure that on hover, the admin color is still used.
+		&:hover {
+			color: var(--wp-admin-theme-color);
+		}
+	}
+
 	// The background has to be applied to the td, not tr, or border-radius won't work.
 	&.is-selected td {
 		background: var(--wp-admin-theme-color);
@@ -93,7 +105,7 @@
 	&.is-branch-selected.is-first-selected td:last-child {
 		border-top-right-radius: $radius-block-ui;
 	}
-	&[aria-expanded="false"] {
+	&[data-expanded="false"] {
 		&.is-branch-selected.is-first-selected td:first-child {
 			border-top-left-radius: $radius-block-ui;
 		}
@@ -380,7 +392,7 @@ $block-navigation-max-indent: 8;
 }
 
 // Point downwards when open.
-.block-editor-list-view-leaf[aria-expanded="true"] .block-editor-list-view__expander svg {
+.block-editor-list-view-leaf[data-expanded="true"] .block-editor-list-view__expander svg {
 	visibility: visible;
 	transition: transform 0.2s ease;
 	transform: rotate(90deg);
@@ -388,7 +400,7 @@ $block-navigation-max-indent: 8;
 }
 
 // Point rightwards when closed
-.block-editor-list-view-leaf[aria-expanded="false"] .block-editor-list-view__expander svg {
+.block-editor-list-view-leaf[data-expanded="false"] .block-editor-list-view__expander svg {
 	visibility: visible;
 	transform: rotate(0deg);
 	transition: transform 0.2s ease;
diff --git a/packages/block-editor/src/components/list-view/use-block-selection.js b/packages/block-editor/src/components/list-view/use-block-selection.js
index e612ccaab6329f..716995edbdd53f 100644
--- a/packages/block-editor/src/components/list-view/use-block-selection.js
+++ b/packages/block-editor/src/components/list-view/use-block-selection.js
@@ -144,7 +144,7 @@ export default function useBlockSelection() {
 			}
 
 			if ( label ) {
-				speak( label );
+				speak( label, 'assertive' );
 			}
 		},
 		[
diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index f7056713b817a0..005e757e2f4418 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Enhancements
+
+-   `TreeGrid`: Modify keyboard navigation code to use a data-expanded attribute if aria-expanded is to be controlled outside of the TreeGrid component ([#48461](https://github.com/WordPress/gutenberg/pull/48461)).
+
 ### Documentation
 
 -   `Autocomplete`: Add heading and fix type for `onReplace` in README. ([#49798](https://github.com/WordPress/gutenberg/pull/49798)).
diff --git a/packages/components/src/tree-grid/index.tsx b/packages/components/src/tree-grid/index.tsx
index 7087bafc86f70b..6a258eb9b1e319 100644
--- a/packages/components/src/tree-grid/index.tsx
+++ b/packages/components/src/tree-grid/index.tsx
@@ -91,7 +91,8 @@ function UnforwardedTreeGrid(
 			const canExpandCollapse = 0 === currentColumnIndex;
 			const cannotFocusNextColumn =
 				canExpandCollapse &&
-				activeRow.getAttribute( 'aria-expanded' ) === 'false' &&
+				( activeRow.getAttribute( 'data-expanded' ) === 'false' ||
+					activeRow.getAttribute( 'aria-expanded' ) === 'false' ) &&
 				keyCode === RIGHT;
 
 			if ( ( [ LEFT, RIGHT ] as number[] ).includes( keyCode ) ) {
@@ -112,6 +113,8 @@ function UnforwardedTreeGrid(
 						// Left:
 						// If a row is focused, and it is expanded, collapses the current row.
 						if (
+							activeRow.getAttribute( 'data-expanded' ) ===
+								'true' ||
 							activeRow.getAttribute( 'aria-expanded' ) === 'true'
 						) {
 							onCollapseRow( activeRow );
@@ -151,8 +154,10 @@ function UnforwardedTreeGrid(
 						// Right:
 						// If a row is focused, and it is collapsed, expands the current row.
 						if (
+							activeRow.getAttribute( 'data-expanded' ) ===
+								'false' ||
 							activeRow.getAttribute( 'aria-expanded' ) ===
-							'false'
+								'false'
 						) {
 							onExpandRow( activeRow );
 							event.preventDefault();
diff --git a/test/e2e/specs/editor/blocks/columns.spec.js b/test/e2e/specs/editor/blocks/columns.spec.js
index 58acffa5fee053..c2078c861c4a89 100644
--- a/test/e2e/specs/editor/blocks/columns.spec.js
+++ b/test/e2e/specs/editor/blocks/columns.spec.js
@@ -26,7 +26,7 @@ test.describe( 'Columns', () => {
 		// block column add
 		await page
 			.locator(
-				'role=treegrid[name="Block navigation structure"i] >> role=gridcell[name="Column link"i]'
+				'role=treegrid[name="Block navigation structure"i] >> role=gridcell[name="Column"i]'
 			)
 			.first()
 			.click();
diff --git a/test/e2e/specs/editor/various/list-view.spec.js b/test/e2e/specs/editor/various/list-view.spec.js
index e7437bc2acbf5d..b7021752ea8cfb 100644
--- a/test/e2e/specs/editor/various/list-view.spec.js
+++ b/test/e2e/specs/editor/various/list-view.spec.js
@@ -27,7 +27,8 @@ test.describe( 'List View', () => {
 		// The last inserted block should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Paragraph link',
+				name: 'Paragraph',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
@@ -43,10 +44,12 @@ test.describe( 'List View', () => {
 
 		// Drag the paragraph above the heading.
 		const paragraphBlockItem = listView.getByRole( 'gridcell', {
-			name: 'Paragraph link',
+			name: 'Paragraph',
+			exact: true,
 		} );
 		const headingBlockItem = listView.getByRole( 'gridcell', {
-			name: 'Heading link',
+			name: 'Heading',
+			exact: true,
 		} );
 		await paragraphBlockItem.dragTo( headingBlockItem, { x: 0, y: 0 } );
 
@@ -80,7 +83,8 @@ test.describe( 'List View', () => {
 		// The last inserted block should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Paragraph link',
+				name: 'Paragraph',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
@@ -88,11 +92,9 @@ test.describe( 'List View', () => {
 		// Go to the image block in List View.
 		await pageUtils.pressKeys( 'ArrowUp', { times: 2 } );
 		await expect(
-			listView
-				.getByRole( 'gridcell', {
-					name: 'Image link',
-				} )
-				.getByRole( 'link', { includeHidden: true } )
+			listView.getByRole( 'link', {
+				name: 'Image',
+			} )
 		).toBeFocused();
 
 		// Select the image block in the canvas.
@@ -110,9 +112,7 @@ test.describe( 'List View', () => {
 		await page.keyboard.press( 'Backspace' );
 
 		// List View should have two rows.
-		await expect(
-			listView.getByRole( 'gridcell', { name: /link/i } )
-		).toHaveCount( 2 );
+		await expect( listView.getByRole( 'row' ) ).toHaveCount( 2 );
 	} );
 
 	// Check for regression of https://github.com/WordPress/gutenberg/issues/39026.
@@ -135,14 +135,15 @@ test.describe( 'List View', () => {
 		// The last inserted block should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Paragraph link',
+				name: 'Paragraph',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
 
 		// Remove the Paragraph block via its options menu in List View.
 		await listView
-			.getByRole( 'button', { name: 'Options for Paragraph block' } )
+			.getByRole( 'button', { name: 'Options for Paragraph' } )
 			.click();
 		await page.getByRole( 'menuitem', { name: /Delete/i } ).click();
 
@@ -174,7 +175,8 @@ test.describe( 'List View', () => {
 		// The last inserted block should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Paragraph link',
+				name: 'Paragraph',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
@@ -182,17 +184,15 @@ test.describe( 'List View', () => {
 		// Select the image block in List View.
 		await pageUtils.pressKeys( 'ArrowUp', { times: 2 } );
 		await expect(
-			listView
-				.getByRole( 'gridcell', {
-					name: 'Image link',
-				} )
-				.getByRole( 'link', { includeHidden: true } )
+			listView.getByRole( 'link', {
+				name: 'Image',
+			} )
 		).toBeFocused();
 		await page.keyboard.press( 'Enter' );
 
 		// Remove the Image block via its options menu in List View.
 		await listView
-			.getByRole( 'button', { name: 'Options for Image block' } )
+			.getByRole( 'button', { name: 'Options for Image' } )
 			.click();
 		await page.getByRole( 'menuitem', { name: /Delete/i } ).click();
 
@@ -227,7 +227,8 @@ test.describe( 'List View', () => {
 		// The last inserted block should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Heading link',
+				name: 'Heading',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
@@ -236,14 +237,15 @@ test.describe( 'List View', () => {
 		await pageUtils.pressKeys( 'shift+ArrowUp' );
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Image link',
+				name: 'Image',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
 
 		// Remove both blocks.
 		await listView
-			.getByRole( 'button', { name: 'Options for Image block' } )
+			.getByRole( 'button', { name: 'Options for Image' } )
 			.click();
 		await page.getByRole( 'menuitem', { name: /Delete blocks/i } ).click();
 
@@ -276,8 +278,8 @@ test.describe( 'List View', () => {
 
 		// Things start off expanded.
 		await expect(
-			listView.getByRole( 'gridcell', {
-				name: 'Cover link',
+			listView.getByRole( 'link', {
+				name: 'Cover',
 				expanded: true,
 			} )
 		).toBeVisible();
@@ -285,28 +287,27 @@ test.describe( 'List View', () => {
 		// The child paragraph block should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Paragraph link',
+				name: 'Paragraph',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
 
 		// Collapse the Cover block.
 		await listView
-			.getByRole( 'gridcell', { name: 'Cover link' } )
+			.getByRole( 'gridcell', { name: 'Cover', exact: true } )
 			.getByTestId( 'list-view-expander', { includeHidden: true } )
 			// Force the click to bypass the visibility check. The expander is
 			// intentionally aria-hidden. See the implementation for details.
 			.click( { force: true } );
 
 		// Check that we're collapsed.
-		await expect(
-			listView.getByRole( 'gridcell', { name: /link/i } )
-		).toHaveCount( 1 );
+		await expect( listView.getByRole( 'row' ) ).toHaveCount( 1 );
 
 		// Click the Cover block List View item.
 		await listView
-			.getByRole( 'gridcell', {
-				name: 'Cover link',
+			.getByRole( 'link', {
+				name: 'Cover',
 				expanded: false,
 			} )
 			.click();
@@ -320,7 +321,8 @@ test.describe( 'List View', () => {
 		// The child paragraph block in List View should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Paragraph link',
+				name: 'Paragraph',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
@@ -347,7 +349,8 @@ test.describe( 'List View', () => {
 		// The last inserted block should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Group link',
+				name: 'Group',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
@@ -355,35 +358,32 @@ test.describe( 'List View', () => {
 		// Press Home to go to the first inserted block (image).
 		await page.keyboard.press( 'Home' );
 		await expect(
-			listView
-				.getByRole( 'gridcell', {
-					name: 'Image link',
-				} )
-				.getByRole( 'link', { includeHidden: true } )
+			listView.getByRole( 'link', {
+				name: 'Image',
+			} )
 		).toBeFocused();
 
 		// Press End followed by Arrow Up to go to the second to last block (columns).
 		await page.keyboard.press( 'End' );
 		await page.keyboard.press( 'ArrowUp' );
 		await expect(
-			listView
-				.getByRole( 'gridcell', {
-					name: 'Columns link',
-				} )
-				.getByRole( 'link', { includeHidden: true } )
+			listView.getByRole( 'link', {
+				name: 'Columns',
+				exact: true,
+			} )
 		).toBeFocused();
 
 		// Navigate the right column to image block options button via Home key.
 		await page.keyboard.press( 'ArrowRight' );
 		await page.keyboard.press( 'Home' );
 		await expect(
-			listView.getByRole( 'button', { name: 'Options for Image block' } )
+			listView.getByRole( 'button', { name: 'Options for Image' } )
 		).toBeFocused();
 
 		// Navigate the right column to group block options button.
 		await page.keyboard.press( 'End' );
 		await expect(
-			listView.getByRole( 'button', { name: 'Options for Group block' } )
+			listView.getByRole( 'button', { name: 'Options for Group' } )
 		).toBeFocused();
 	} );
 
@@ -415,18 +415,17 @@ test.describe( 'List View', () => {
 		// The paragraph item should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Paragraph link',
+				name: 'Paragraph',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
 
 		// Navigate to the image block item.
 		await page.keyboard.press( 'ArrowUp' );
-		const imageItem = listView
-			.getByRole( 'gridcell', {
-				name: 'Image link',
-			} )
-			.getByRole( 'link', { includeHidden: true } );
+		const imageItem = listView.getByRole( 'link', {
+			name: 'Image',
+		} );
 
 		await expect( imageItem ).toBeFocused();
 
@@ -527,7 +526,8 @@ test.describe( 'List View', () => {
 		// The last inserted block should be selected.
 		await expect(
 			listView.getByRole( 'gridcell', {
-				name: 'Paragraph link',
+				name: 'Paragraph',
+				exact: true,
 				selected: true,
 			} )
 		).toBeVisible();
@@ -535,11 +535,9 @@ test.describe( 'List View', () => {
 		// Go to the image block in List View.
 		await pageUtils.pressKeys( 'ArrowUp', { times: 2 } );
 		await expect(
-			listView
-				.getByRole( 'gridcell', {
-					name: 'Image link',
-				} )
-				.getByRole( 'link', { includeHidden: true } )
+			listView.getByRole( 'link', {
+				name: 'Image',
+			} )
 		).toBeFocused();
 
 		// Select the image block in the canvas.
@@ -554,11 +552,9 @@ test.describe( 'List View', () => {
 		// Triggering the List View shortcut should result in the image block gaining focus.
 		await pageUtils.pressKeys( 'access+o' );
 		await expect(
-			listView
-				.getByRole( 'gridcell', {
-					name: 'Image link',
-				} )
-				.getByRole( 'link', { includeHidden: true } )
+			listView.getByRole( 'link', {
+				name: 'Image',
+			} )
 		).toBeFocused();
 	} );
 } );
diff --git a/test/e2e/specs/editor/various/multi-block-selection.spec.js b/test/e2e/specs/editor/various/multi-block-selection.spec.js
index 3e03a0c2805620..d72ab2c6bf7dbf 100644
--- a/test/e2e/specs/editor/various/multi-block-selection.spec.js
+++ b/test/e2e/specs/editor/various/multi-block-selection.spec.js
@@ -916,8 +916,8 @@ test.describe( 'Multi-block selection', () => {
 		const listView = page.getByRole( 'treegrid', {
 			name: 'Block navigation structure',
 		} );
-		const navButtons = listView.getByRole( 'gridcell', {
-			name: 'Paragraph link',
+		const navButtons = listView.getByRole( 'link', {
+			name: 'Paragraph',
 		} );
 
 		await navButtons.nth( 1 ).click();
@@ -954,9 +954,7 @@ test.describe( 'Multi-block selection', () => {
 
 		// Move focus to the list view link to prepare for the keyboard navigation.
 		await navButtons.nth( 3 ).click();
-		await expect(
-			navButtons.nth( 3 ).getByRole( 'link', { includeHidden: true } )
-		).toBeFocused();
+		await expect( navButtons.nth( 3 ) ).toBeFocused();
 		// Press Up twice to highlight the second block.
 		await pageUtils.pressKeys( 'ArrowUp', { times: 2 } );
 
