diff --git a/packages/components/src/ui/context/get-styled-class-name-from-key.ts b/packages/components/src/ui/context/get-styled-class-name-from-key.ts
index 76dba53001d822..c2f5ebec5df6ec 100644
--- a/packages/components/src/ui/context/get-styled-class-name-from-key.ts
+++ b/packages/components/src/ui/context/get-styled-class-name-from-key.ts
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { kebabCase } from 'lodash';
+import { paramCase as kebabCase } from 'change-case';
 import memoize from 'memize';
 
 /**
