diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index 0f4c8ebb8a08cf..a7a1755e7a9eb5 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -2,7 +2,7 @@
 
 ## Unreleased
 
-## 25.13.0 (2023-11-29)
+- `BorderControlDropdown`: Switches from appending state to the aria-label and adds an aria-describedBy attribute with hidden text (via display: none;, as the text will be read via focus on the button and doesn't need to announced otherwise ([#54469](https://github.com/WordPress/gutenberg/pull/54469)).
 
 ### Enhancements
 
diff --git a/packages/components/src/border-box-control/test/index.tsx b/packages/components/src/border-box-control/test/index.tsx
index fe5f7e7a4f1fff..0b3a7ebc8cab06 100644
--- a/packages/components/src/border-box-control/test/index.tsx
+++ b/packages/components/src/border-box-control/test/index.tsx
@@ -46,8 +46,8 @@ const props = {
 	value: undefined,
 };
 
-const toggleLabelRegex = /Border color( and style)* picker/;
-const colorPickerRegex = /Border color picker/;
+const toggleLabelRegex = /The currently selected color has/;
+const colorPickerRegex = /The currently selected color is called/;
 
 describe( 'BorderBoxControl', () => {
 	describe( 'Linked view rendering', () => {
diff --git a/packages/components/src/border-control/border-control-dropdown/component.tsx b/packages/components/src/border-control/border-control-dropdown/component.tsx
index 4f43a6ed0ce55d..9b3fb923671bbc 100644
--- a/packages/components/src/border-control/border-control-dropdown/component.tsx
+++ b/packages/components/src/border-control/border-control-dropdown/component.tsx
@@ -6,6 +6,7 @@ import type { CSSProperties } from 'react';
 /**
  * WordPress dependencies
  */
+import { useInstanceId } from '@wordpress/compose';
 import { __, sprintf } from '@wordpress/i18n';
 import { closeSmall } from '@wordpress/icons';
 
@@ -24,13 +25,12 @@ import { contextConnect } from '../../context';
 import { useBorderControlDropdown } from './hook';
 import { StyledLabel } from '../../base-control/styles/base-control-styles';
 import DropdownContentWrapper from '../../dropdown/dropdown-content-wrapper';
-
 import type { ColorObject } from '../../color-palette/types';
 import { isMultiplePaletteArray } from '../../color-palette/utils';
 import type { DropdownProps as DropdownComponentProps } from '../../dropdown/types';
 import type { ColorProps, DropdownProps } from '../types';
 
-const getAriaLabelColorValue = ( colorValue: string ) => {
+const getAriaDescriptionColorValue = ( colorValue: string ) => {
 	// Leave hex values as-is. Remove the `var()` wrapper from CSS vars.
 	return colorValue.replace( /^var\((.+)\)$/, '$1' );
 };
@@ -65,7 +65,7 @@ const getColorObject = (
 	return colors.find( ( color ) => color.color === colorValue );
 };
 
-const getToggleAriaLabel = (
+const getToggleDescription = (
 	colorValue: CSSProperties[ 'borderColor' ],
 	colorObject: ColorObject | undefined,
 	style: CSSProperties[ 'borderStyle' ],
@@ -73,60 +73,69 @@ const getToggleAriaLabel = (
 ) => {
 	if ( isStyleEnabled ) {
 		if ( colorObject ) {
-			const ariaLabelValue = getAriaLabelColorValue( colorObject.color );
+			const ariaDescriptionValue = getAriaDescriptionColorValue(
+				colorObject.color
+			);
 			return style
 				? sprintf(
 						// translators: %1$s: The name of the color e.g. "vivid red". %2$s: The color's hex code e.g.: "#f00:". %3$s: The current border style selection e.g. "solid".
-						'Border color and style picker. The currently selected color is called "%1$s" and has a value of "%2$s". The currently selected style is "%3$s".',
+						'The currently selected color is called "%1$s" and has a value of "%2$s". The currently selected style is "%3$s".',
 						colorObject.name,
-						ariaLabelValue,
+						ariaDescriptionValue,
 						style
 				  )
 				: sprintf(
 						// translators: %1$s: The name of the color e.g. "vivid red". %2$s: The color's hex code e.g.: "#f00:".
-						'Border color and style picker. The currently selected color is called "%1$s" and has a value of "%2$s".',
+						'The currently selected color is called "%1$s" and has a value of "%2$s".',
 						colorObject.name,
-						ariaLabelValue
+						ariaDescriptionValue
 				  );
 		}
 
 		if ( colorValue ) {
-			const ariaLabelValue = getAriaLabelColorValue( colorValue );
+			const ariaDescriptionValue =
+				getAriaDescriptionColorValue( colorValue );
 			return style
 				? sprintf(
 						// translators: %1$s: The color's hex code e.g.: "#f00:". %2$s: The current border style selection e.g. "solid".
-						'Border color and style picker. The currently selected color has a value of "%1$s". The currently selected style is "%2$s".',
-						ariaLabelValue,
+						'The currently selected color has a value of "%1$s". The currently selected style is "%2$s".',
+						ariaDescriptionValue,
 						style
 				  )
 				: sprintf(
 						// translators: %1$s: The color's hex code e.g: "#f00".
-						'Border color and style picker. The currently selected color has a value of "%1$s".',
-						ariaLabelValue
+						'The currently selected color has a value of "%1$s".',
+						ariaDescriptionValue
 				  );
 		}
 
-		return __( 'Border color and style picker.' );
+		if ( style ) {
+			return sprintf(
+				// translators: %1$s: The current border style selection e.g. "solid".
+				'The currently selected style is "%1$s". No color is selected. Select a color to display the border.',
+				style
+			);
+		}
 	}
 
 	if ( colorObject ) {
 		return sprintf(
 			// translators: %1$s: The name of the color e.g. "vivid red". %2$s: The color's hex code e.g: "#f00".
-			'Border color picker. The currently selected color is called "%1$s" and has a value of "%2$s".',
+			'The currently selected color is called "%1$s" and has a value of "%2$s".',
 			colorObject.name,
-			getAriaLabelColorValue( colorObject.color )
+			getAriaDescriptionColorValue( colorObject.color )
 		);
 	}
 
 	if ( colorValue ) {
 		return sprintf(
 			// translators: %1$s: The color's hex code e.g: "#f00".
-			'Border color picker. The currently selected color has a value of "%1$s".',
-			getAriaLabelColorValue( colorValue )
+			'The currently selected color has a value of "%1$s".',
+			getAriaDescriptionColorValue( colorValue )
 		);
 	}
 
-	return __( 'Border color picker.' );
+	return __( 'Select a border color and style.' );
 };
 
 const BorderControlDropdown = (
@@ -155,8 +164,12 @@ const BorderControlDropdown = (
 
 	const { color, style } = border || {};
 	const colorObject = getColorObject( color, colors );
+	const toggleDescriptionId = useInstanceId(
+		BorderControlDropdown,
+		'border-control-dropdown'
+	);
 
-	const toggleAriaLabel = getToggleAriaLabel(
+	const toggleDescription = getToggleDescription(
 		color,
 		colorObject,
 		style,
@@ -174,10 +187,10 @@ const BorderControlDropdown = (
 		<Button
 			onClick={ onToggle }
 			variant="tertiary"
-			aria-label={ toggleAriaLabel }
 			tooltipPosition={ dropdownPosition }
 			label={ __( 'Border color and style picker' ) }
 			showTooltip={ true }
+			aria-describedby={ toggleDescriptionId }
 		>
 			<span className={ indicatorWrapperClassName }>
 				<ColorIndicator
@@ -243,15 +256,20 @@ const BorderControlDropdown = (
 	);
 
 	return (
-		<Dropdown
-			renderToggle={ renderToggle }
-			renderContent={ renderContent }
-			popoverProps={ {
-				...__unstablePopoverProps,
-			} }
-			{ ...otherProps }
-			ref={ forwardedRef }
-		/>
+		<>
+			<div id={ toggleDescriptionId } style={ { display: 'none' } }>
+				{ toggleDescription }
+			</div>
+			<Dropdown
+				renderToggle={ renderToggle }
+				renderContent={ renderContent }
+				popoverProps={ {
+					...__unstablePopoverProps,
+				} }
+				{ ...otherProps }
+				ref={ forwardedRef }
+			/>
+		</>
 	);
 };
 
diff --git a/packages/components/src/border-control/test/index.js b/packages/components/src/border-control/test/index.js
index c41dce687cc522..c6ac1215597847 100644
--- a/packages/components/src/border-control/test/index.js
+++ b/packages/components/src/border-control/test/index.js
@@ -186,7 +186,7 @@ describe( 'BorderControl', () => {
 				render( <BorderControl { ...props } /> );
 
 				expect(
-					screen.getByLabelText( 'Border color and style picker.' )
+					screen.getByText( 'Select a border color and style.' )
 				).toBeInTheDocument();
 			} );
 
@@ -195,8 +195,8 @@ describe( 'BorderControl', () => {
 				render( <BorderControl { ...props } /> );
 
 				expect(
-					screen.getByLabelText(
-						'Border color and style picker. The currently selected color is called "Blue" and has a value of "#72aee6".'
+					screen.getByText(
+						'The currently selected color is called "Blue" and has a value of "#72aee6".'
 					)
 				).toBeInTheDocument();
 			} );
@@ -206,8 +206,8 @@ describe( 'BorderControl', () => {
 				render( <BorderControl { ...props } /> );
 
 				expect(
-					screen.getByLabelText(
-						'Border color and style picker. The currently selected color has a value of "#4b1d80".'
+					screen.getByText(
+						'The currently selected color has a value of "#4b1d80".'
 					)
 				).toBeInTheDocument();
 			} );
@@ -219,8 +219,8 @@ describe( 'BorderControl', () => {
 				render( <BorderControl { ...props } /> );
 
 				expect(
-					screen.getByLabelText(
-						'Border color and style picker. The currently selected color is called "Blue" and has a value of "#72aee6". The currently selected style is "dotted".'
+					screen.getByText(
+						'The currently selected color is called "Blue" and has a value of "#72aee6". The currently selected style is "dotted".'
 					)
 				).toBeInTheDocument();
 			} );
@@ -232,8 +232,8 @@ describe( 'BorderControl', () => {
 				render( <BorderControl { ...props } /> );
 
 				expect(
-					screen.getByLabelText(
-						'Border color and style picker. The currently selected color has a value of "#4b1d80". The currently selected style is "dashed".'
+					screen.getByText(
+						'The currently selected color has a value of "#4b1d80". The currently selected style is "dashed".'
 					)
 				).toBeInTheDocument();
 			} );
@@ -248,7 +248,7 @@ describe( 'BorderControl', () => {
 				render( <BorderControl { ...props } /> );
 
 				expect(
-					screen.getByLabelText( 'Border color picker.' )
+					screen.getByText( 'Select a border color and style.' )
 				).toBeInTheDocument();
 			} );
 
@@ -260,8 +260,8 @@ describe( 'BorderControl', () => {
 				render( <BorderControl { ...props } /> );
 
 				expect(
-					screen.getByLabelText(
-						'Border color picker. The currently selected color is called "Blue" and has a value of "#72aee6".'
+					screen.getByText(
+						'The currently selected color is called "Blue" and has a value of "#72aee6".'
 					)
 				).toBeInTheDocument();
 			} );
@@ -274,8 +274,8 @@ describe( 'BorderControl', () => {
 				render( <BorderControl { ...props } /> );
 
 				expect(
-					screen.getByLabelText(
-						'Border color picker. The currently selected color has a value of "#4b1d80".'
+					screen.getByText(
+						'The currently selected color has a value of "#4b1d80".'
 					)
 				).toBeInTheDocument();
 			} );
