diff --git a/packages/components/src/custom-select/index.tsx b/packages/components/src/custom-select/index.tsx
new file mode 100644
index 00000000000000..505f5c46871825
--- /dev/null
+++ b/packages/components/src/custom-select/index.tsx
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CustomSelectButton,
+	CustomSelectPopover,
+	CustomSelectItem as SelectItem,
+} from './styles';
+import { type CustomSelectProps } from './types';
+
+export function CustomSelect( props: CustomSelectProps ) {
+	const { label, onChange, options } = props;
+
+	const store = Ariakit.useSelectStore( {
+		setValue: ( value ) => onChange?.( value ),
+	} );
+
+	return (
+		<>
+			<Ariakit.SelectLabel store={ store }>{ label }</Ariakit.SelectLabel>
+			<CustomSelectButton store={ store } />
+			<CustomSelectPopover sameWidth store={ store }>
+				{ options?.map( ( { name, key, ...rest } ) => {
+					return (
+						<SelectItem value={ name } key={ key } { ...rest } />
+					);
+				} ) }
+			</CustomSelectPopover>
+		</>
+	);
+}
diff --git a/packages/components/src/custom-select/stories/index.story.tsx b/packages/components/src/custom-select/stories/index.story.tsx
new file mode 100644
index 00000000000000..66403ee294203f
--- /dev/null
+++ b/packages/components/src/custom-select/stories/index.story.tsx
@@ -0,0 +1,59 @@
+//@ts-nocheck
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CustomSelect, CustomSelectItem } from '..';
+
+const meta: Meta< typeof CustomSelect > = {
+	title: 'Components/CustomSelect',
+	component: CustomSelect,
+	argTypes: {},
+	parameters: {
+		controls: { expanded: true },
+		docs: { canvas: { sourceState: 'shown' } },
+	},
+};
+export default meta;
+
+const Template: StoryFn = () => {
+	const options = [
+		{
+			key: 'small',
+			name: 'Small',
+			style: { fontSize: '50%' },
+		},
+		{
+			key: 'normal',
+			name: 'Normal',
+			style: { fontSize: '100%' },
+		},
+		{
+			key: 'large',
+			name: 'Large',
+			style: { fontSize: '200%' },
+		},
+		{
+			key: 'huge',
+			name: 'Huge',
+			style: { fontSize: '300%' },
+		},
+	];
+
+	return (
+		<>
+			<CustomSelect label="Font Size" options={ options } />
+		</>
+	);
+};
+
+export const Default = Template.bind( {} );
diff --git a/packages/components/src/custom-select/styles.ts b/packages/components/src/custom-select/styles.ts
new file mode 100644
index 00000000000000..75ac50f59a2dcb
--- /dev/null
+++ b/packages/components/src/custom-select/styles.ts
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * Internal dependencies
+ */
+import { COLORS } from '../utils';
+import { space } from '../utils/space';
+
+export const CustomSelectButton = styled( Ariakit.Select )`
+	display: flex;
+	align-items: center;
+	justify-content: space-between;
+	gap: ${ space( 1 ) };
+	white-space: nowrap;
+	font-size: 1rem;
+	border-style: none;
+	border-radius: ${ space( 1 ) };
+	min-width: 250px;
+	height: auto;
+	margin-top: 1rem;
+	padding: ${ space( 4 ) };
+	background: ${ COLORS.white };
+	box-shadow: 0 0 0 var( --wp-admin-border-width-focus )
+		${ COLORS.gray[ 400 ] };
+	&:hover {
+		background-color: ${ COLORS.gray[ 100 ] };
+	}
+`;
+
+export const CustomSelectPopover = styled( Ariakit.SelectPopover )`
+	z-index: 50;
+	display: flex;
+	flex-direction: column;
+	border-radius: ${ space( 1 ) };
+	background: ${ COLORS.white };
+	box-shadow: 0 0 0 var( --wp-admin-border-width-focus )
+		${ COLORS.gray[ 400 ] };
+`;
+export const CustomSelectItem = styled( Ariakit.SelectItem )`
+	cursor: pointer;
+	padding: ${ space( 2 ) };
+	&:hover {
+		background-color: ${ COLORS.theme.accent };
+		color: ${ COLORS.white };
+	}
+`;
diff --git a/packages/components/src/custom-select/types.ts b/packages/components/src/custom-select/types.ts
new file mode 100644
index 00000000000000..2a1da135ae9290
--- /dev/null
+++ b/packages/components/src/custom-select/types.ts
@@ -0,0 +1,17 @@
+export type CustomSelectProps = {
+	label?: string;
+	/**
+	 * Function called with the control's internal state changes. The `selectedItem` property contains the next selected item.
+	 */
+	onChange?: Function;
+	/**
+	 * The options that can be chosen from.
+	 */
+	options: Array< {
+		key: string;
+		name: string;
+		style?: {};
+		className?: string;
+		__experimentalHint?: string;
+	} >;
+};
