diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index 19feebd911b456..0448da86c22740 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -20,6 +20,7 @@
 ### Experimental
 
 -   `Tabs`: do not render hidden content ([#57046](https://github.com/WordPress/gutenberg/pull/57046)).
+-   `Tabs`: improve hover and text alignment styles ([#57275](https://github.com/WordPress/gutenberg/pull/57275)).
 -   `Tabs`: make sure `Tab`s are associated to the right `TabPanel`s, regardless of the order they're rendered in ([#57033](https://github.com/WordPress/gutenberg/pull/57033)).
 
 ## 25.14.0 (2023-12-13)
diff --git a/packages/components/src/tabs/styles.ts b/packages/components/src/tabs/styles.ts
index cb735f3177662a..6abca79aa51aed 100644
--- a/packages/components/src/tabs/styles.ts
+++ b/packages/components/src/tabs/styles.ts
@@ -22,7 +22,9 @@ export const TabListWrapper = styled.div`
 `;
 
 export const Tab = styled( Ariakit.Tab )`
-	&& {
+	& {
+		display: inline-flex;
+		align-items: center;
 		position: relative;
 		border-radius: 0;
 		height: ${ space( 12 ) };
@@ -39,6 +41,10 @@ export const Tab = styled( Ariakit.Tab )`
 			opacity: 0.3;
 		}
 
+		&:hover {
+			color: ${ COLORS.theme.accent };
+		}
+
 		&:focus:not( :disabled ) {
 			position: relative;
 			box-shadow: none;
