diff --git a/packages/block-library/src/navigation/edit/navigation-menu-selector.js b/packages/block-library/src/navigation/edit/navigation-menu-selector.js
index d895de770fdafe..14b6e1706ffe71 100644
--- a/packages/block-library/src/navigation/edit/navigation-menu-selector.js
+++ b/packages/block-library/src/navigation/edit/navigation-menu-selector.js
@@ -131,7 +131,7 @@ function NavigationMenuSelector( {
 		<DropdownMenu
 			label={ selectorLabel }
 			icon={ moreVertical }
-			toggleProps={ { isSmall: true } }
+			toggleProps={ { size: 'small' } }
 		>
 			{ ( { onClose } ) => (
 				<>
diff --git a/packages/components/src/button/test/index.tsx b/packages/components/src/button/test/index.tsx
index 9b719f23a923f1..699db1b75429f0 100644
--- a/packages/components/src/button/test/index.tsx
+++ b/packages/components/src/button/test/index.tsx
@@ -593,6 +593,11 @@ describe( 'Button', () => {
 			expect( screen.getByRole( 'button' ) ).toHaveClass( 'is-small' );
 		} );
 
+		it( 'should have the is-small class when small class prop is passed', () => {
+			render( <Button size="small" /> );
+			expect( screen.getByRole( 'button' ) ).toHaveClass( 'is-small' );
+		} );
+
 		it( 'should prioritize the `size` prop over `isSmall`', () => {
 			render( <Button size="compact" isSmall /> );
 			expect( screen.getByRole( 'button' ) ).not.toHaveClass(
