diff --git a/packages/components/src/focal-point-picker/index.tsx b/packages/components/src/focal-point-picker/index.tsx
index aa6c5484948cf9..946db9f12e0cac 100644
--- a/packages/components/src/focal-point-picker/index.tsx
+++ b/packages/components/src/focal-point-picker/index.tsx
@@ -27,7 +27,6 @@ import {
 	MediaContainer,
 } from './styles/focal-point-picker-style';
 import { INITIAL_BOUNDS } from './utils';
-import { useUpdateEffect } from '../utils/hooks';
 import type { WordPressComponentProps } from '../context/wordpress-component';
 import type {
 	FocalPoint as FocalPointType,
@@ -35,8 +34,6 @@ import type {
 } from './types';
 import type { KeyboardEventHandler } from 'react';
 
-const GRID_OVERLAY_TIMEOUT = 600;
-
 /**
  * Focal Point Picker is a component which creates a UI for identifying the most important visual point of an image.
  *
@@ -116,6 +113,7 @@ export function FocalPointPicker( {
 			}
 
 			onDragStart?.( value, event );
+			setShowGridOverlay( true );
 			setPoint( value );
 		},
 		onDragMove: ( event ) => {
@@ -129,6 +127,7 @@ export function FocalPointPicker( {
 			setPoint( value );
 		},
 		onDragEnd: () => {
+			setShowGridOverlay( false );
 			onDragEnd?.();
 			onChange?.( point );
 		},
@@ -237,15 +236,6 @@ export function FocalPointPicker( {
 	const instanceId = useInstanceId( FocalPointPicker );
 	const id = `inspector-focal-point-picker-control-${ instanceId }`;
 
-	useUpdateEffect( () => {
-		setShowGridOverlay( true );
-		const timeout = window.setTimeout( () => {
-			setShowGridOverlay( false );
-		}, GRID_OVERLAY_TIMEOUT );
-
-		return () => window.clearTimeout( timeout );
-	}, [ x, y ] );
-
 	return (
 		<BaseControl
 			{ ...restProps }
