diff --git a/packages/components/src/custom-select-control-v2/custom-select.tsx b/packages/components/src/custom-select-control-v2/custom-select.tsx
index e925038812fa73..570dfcbf9c48bb 100644
--- a/packages/components/src/custom-select-control-v2/custom-select.tsx
+++ b/packages/components/src/custom-select-control-v2/custom-select.tsx
@@ -8,6 +8,7 @@ import * as Ariakit from '@ariakit/react';
  */
 import { createContext, useCallback, useMemo } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
+import { useInstanceId } from '@wordpress/compose';
 
 /**
  * Internal dependencies
@@ -95,6 +96,7 @@ function _CustomSelect(
 		store,
 		className,
 		isLegacy = false,
+		describedBy,
 		...restProps
 	} = props;
 
@@ -109,6 +111,8 @@ function _CustomSelect(
 		);
 
 	const contextValue = useMemo( () => ( { store, size } ), [ store, size ] );
+	const instanceId = useInstanceId( _CustomSelect );
+	const describedByID = `custom-select-described-by-${ instanceId }`;
 
 	return (
 		// Where should `restProps` be forwarded to?
@@ -138,7 +142,13 @@ function _CustomSelect(
 					store={ store }
 					// Match legacy behavior (move selection rather than open the popover)
 					showOnKeyDown={ ! isLegacy }
+					aria-describedby={ describedBy ? describedByID : undefined }
 				/>
+				{ describedBy && (
+					<div id={ describedByID } hidden>
+						{ describedBy }
+					</div>
+				) }
 				<Styled.SelectPopover
 					gutter={ 12 }
 					store={ store }
diff --git a/packages/components/src/custom-select-control-v2/types.ts b/packages/components/src/custom-select-control-v2/types.ts
index a31598c1db3ead..f5d127fadbf0ec 100644
--- a/packages/components/src/custom-select-control-v2/types.ts
+++ b/packages/components/src/custom-select-control-v2/types.ts
@@ -77,6 +77,10 @@ export type _CustomSelectProps = CustomSelectButtonProps & {
 	 * Accessible label for the control.
 	 */
 	label: string;
+	/**
+	 * Described by.
+	 */
+	describedBy?: string;
 };
 
 export type CustomSelectProps = _CustomSelectProps & CustomSelectSize;
diff --git a/packages/components/src/custom-select-control/index.tsx b/packages/components/src/custom-select-control/index.tsx
index 5bce6afc4d92b6..59396ea1944136 100644
--- a/packages/components/src/custom-select-control/index.tsx
+++ b/packages/components/src/custom-select-control/index.tsx
@@ -154,7 +154,6 @@ function CustomSelectControl( props: CustomSelectProps ) {
 
 	return (
 		<_CustomSelect
-			aria-describedby={ describedBy }
 			renderSelectedValue={
 				showSelectedHint ? renderSelectedValueHint : undefined
 			}
@@ -166,6 +165,7 @@ function CustomSelectControl( props: CustomSelectProps ) {
 				classNameProp
 			) }
 			isLegacy
+			describedBy={ describedBy }
 			{ ...restProps }
 		>
 			{ children }
