diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index db8b0f749217f4..91f681339bf7ca 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -13,6 +13,7 @@
 -   `InputControl`: Deprecate 36px default size ([#66897](https://github.com/WordPress/gutenberg/pull/66897)).
 -   `RadioGroup`: Log deprecation warning ([#68067](https://github.com/WordPress/gutenberg/pull/68067)).
 -   Soft deprecate `ButtonGroup` component. Use `ToggleGroupControl` instead ([#65429](https://github.com/WordPress/gutenberg/pull/65429)).
+-   `Navigation`: Log deprecation warning for removal in WP 7.1. Use `Navigator` instead ([#68158](https://github.com/WordPress/gutenberg/pull/68158)).
 
 ### Bug Fixes
 
diff --git a/packages/components/src/navigation/index.tsx b/packages/components/src/navigation/index.tsx
index 92f431dfb22fc7..ef37caf2f52140 100644
--- a/packages/components/src/navigation/index.tsx
+++ b/packages/components/src/navigation/index.tsx
@@ -6,6 +6,7 @@ import clsx from 'clsx';
 /**
  * WordPress dependencies
  */
+import deprecated from '@wordpress/deprecated';
 import { useEffect, useRef, useState } from '@wordpress/element';
 import { isRTL } from '@wordpress/i18n';
 
@@ -79,6 +80,12 @@ export function Navigation( {
 	const navigationTree = useCreateNavigationTree();
 	const defaultSlideOrigin = isRTL() ? 'right' : 'left';
 
+	deprecated( 'wp.components.Navigation (and all subcomponents)', {
+		since: '6.8',
+		version: '7.1',
+		alternative: 'wp.components.Navigator',
+	} );
+
 	const setActiveMenu: NavigationContextType[ 'setActiveMenu' ] = (
 		menuId,
 		slideInOrigin = defaultSlideOrigin
diff --git a/packages/components/src/navigation/test/index.tsx b/packages/components/src/navigation/test/index.tsx
index 20646a6c809bfc..fed939068c0bfd 100644
--- a/packages/components/src/navigation/test/index.tsx
+++ b/packages/components/src/navigation/test/index.tsx
@@ -176,6 +176,10 @@ describe( 'Navigation', () => {
 
 		const menuItems = screen.getAllByRole( 'listitem' );
 
+		expect( console ).toHaveWarnedWith(
+			'wp.components.Navigation (and all subcomponents) is deprecated since version 6.8 and will be removed in version 7.1. Please use wp.components.Navigator instead.'
+		);
+
 		expect( menuItems ).toHaveLength( 4 );
 		expect( menuItems[ 0 ] ).toHaveTextContent( 'Item 1' );
 		expect( menuItems[ 1 ] ).toHaveTextContent( 'Item 2' );
