diff --git a/packages/components/src/tools-panel/types.ts b/packages/components/src/tools-panel/types.ts
index e8e2f950de9a30..0569ff97655d3f 100644
--- a/packages/components/src/tools-panel/types.ts
+++ b/packages/components/src/tools-panel/types.ts
@@ -77,7 +77,10 @@ export type ToolsPanelHeaderProps = {
 	/**
 	 * The dropdown menu props to configure the panel's `DropdownMenu`.
 	 */
-	dropdownMenuProps?: React.ComponentProps< typeof DropdownMenu >;
+	dropdownMenuProps?: Omit<
+		React.ComponentProps< typeof DropdownMenu >,
+		'label'
+	>;
 	/**
 	 * The heading level of the panel's header.
 	 *
