diff --git a/packages/components/src/form-token-field/stories/index.story.tsx b/packages/components/src/form-token-field/stories/index.story.tsx
index 52daabe5608b0a..c8de30abbe7701 100644
--- a/packages/components/src/form-token-field/stories/index.story.tsx
+++ b/packages/components/src/form-token-field/stories/index.story.tsx
@@ -115,28 +115,36 @@ DropdownSelector.args = {
 };
 
 /**
- * The rendered output of each suggestion can be customized by passing a
- * render function to the `__experimentalRenderItem` prop. (This is still an experimental feature
- * and is subject to change.)
+ * The rendered content of each token can be customized by passing a
+ * render function to the `displayTransform` prop.
+ *
+ * Similarly, each suggestion can be customized by passing a
+ * render function to the `__experimentalRenderItem` prop. (This is still an
+ * experimental feature and is subject to change.)
  */
-export const WithCustomRenderItem: StoryFn< typeof FormTokenField > =
+export const WithCustomRenderedItems: StoryFn< typeof FormTokenField > =
 	DefaultTemplate.bind( {} );
-WithCustomRenderItem.args = {
+WithCustomRenderedItems.args = {
 	...Default.args,
+	displayTransform: ( token ) => `📍 ${ token }`,
 	__experimentalRenderItem: ( { item } ) => (
 		<div>{ `${ item } — a nice place to visit` }</div>
 	),
+	__experimentalExpandOnFocus: true,
 };
 
 /**
  * Only values for which the `__experimentalValidateInput` function returns
  * `true` will be tokenized. (This is still an experimental feature and is
  * subject to change.)
+ *
+ * In this example, the user can only add tokens that are already in the list.
  */
-export const WithValidatedInput: StoryFn< typeof FormTokenField > =
+export const ValidateNewTokens: StoryFn< typeof FormTokenField > =
 	DefaultTemplate.bind( {} );
-WithValidatedInput.args = {
+ValidateNewTokens.args = {
 	...Default.args,
 	__experimentalValidateInput: ( input: string ) =>
 		continents.includes( input ),
+	__experimentalExpandOnFocus: true,
 };
