diff --git a/packages/components/src/validated-form-controls/components/stories/overview.story.tsx b/packages/components/src/validated-form-controls/components/stories/overview.story.tsx
index 687967aed50786..0d8bab64c240ba 100644
--- a/packages/components/src/validated-form-controls/components/stories/overview.story.tsx
+++ b/packages/components/src/validated-form-controls/components/stories/overview.story.tsx
@@ -17,12 +17,18 @@ import { debounce } from '@wordpress/compose';
 import { ValidatedInputControl } from '..';
 import { formDecorator } from './story-utils';
 import type { ControlWithError } from '../../control-with-error';
+import Dropdown from '../../../dropdown';
 import { Button } from '../../../button';
+import Modal from '../../../modal';
+import { HStack } from '../../../h-stack';
 import { VStack } from '../../../v-stack';
 
 const meta: Meta< typeof ControlWithError > = {
 	title: 'Components/Selection & Input/Validated Form Controls/Overview',
 	id: 'components-validated-form-controls-overview',
+	parameters: {
+		controls: { disable: true },
+	},
 };
 export default meta;
 
@@ -357,3 +363,157 @@ export const ShowingErrorsAtArbitraryTimes: StoryObj<
 		);
 	},
 };
+
+/**
+ * A `form` wrapper and `type="submit"` button can be used to force validation when
+ * the user tries to commit their changes, while still allowing the modal to be closed by canceling.
+ * Optionally, the `shouldCloseOnClickOutside`, `isDismissible`, and `shouldCloseOnEsc` props
+ * on `Modal` can be disabled to force users to more explicitly signal whether they are trying to
+ * "submit close" or "cancel close" the dialog, as well as preventing data loss on accidental closures.
+ */
+export const ValidateInModal: StoryObj< typeof ValidatedInputControl > = {
+	render: function Template( { ...args } ) {
+		const [ isOpen, setIsOpen ] = useState( false );
+		const [ text, setText ] = useState< string | undefined >( '' );
+
+		return (
+			<>
+				<Button
+					variant="secondary"
+					__next40pxDefaultSize
+					onClick={ () => setIsOpen( true ) }
+				>
+					Open in modal
+				</Button>
+				{ isOpen && (
+					<Modal
+						title="Dialog title"
+						onRequestClose={ () => setIsOpen( false ) }
+						shouldCloseOnClickOutside={ false }
+						shouldCloseOnEsc={ false }
+						isDismissible={ false }
+					>
+						<form
+							onSubmit={ ( event ) => {
+								event.preventDefault();
+								setIsOpen( false );
+							} }
+						>
+							<VStack spacing={ 2 }>
+								<ValidatedInputControl
+									{ ...args }
+									value={ text }
+									onChange={ setText }
+									customValidity={
+										text === 'error'
+											? {
+													type: 'invalid',
+													message:
+														'The word "error" is not allowed.',
+											  }
+											: undefined
+									}
+								/>
+
+								<HStack justify="flex-end" spacing={ 2 }>
+									<Button
+										variant="tertiary"
+										__next40pxDefaultSize
+										onClick={ () => setIsOpen( false ) }
+									>
+										Cancel
+									</Button>
+									<Button
+										variant="primary"
+										__next40pxDefaultSize
+										type="submit"
+									>
+										Save
+									</Button>
+								</HStack>
+							</VStack>
+						</form>
+					</Modal>
+				) }
+			</>
+		);
+	},
+	args: {
+		label: 'Text',
+		required: true,
+		help: 'The word "error" will trigger an error.',
+	},
+};
+
+/**
+ * [Form methods](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement#instance_methods) like
+ * `reportValidity()` can be used to validate the fields when a popover is about to be closed,
+ * and prevent the closing of the popover when invalid.
+ */
+export const ValidateOnPopoverClose: StoryObj< typeof ValidatedInputControl > =
+	{
+		render: function Template( { ...args } ) {
+			const [ isOpen, setIsOpen ] = useState( false );
+			const formRef = useRef< HTMLFormElement >( null );
+			const [ text, setText ] = useState< string | undefined >( '' );
+
+			return (
+				<Dropdown
+					popoverProps={ { placement: 'bottom-start' } }
+					open={ isOpen }
+					onToggle={ ( willOpen ) => {
+						if ( ! willOpen ) {
+							const isValid = formRef.current?.reportValidity();
+							setIsOpen( ! isValid );
+						} else {
+							setIsOpen( true );
+						}
+					} }
+					renderContent={ () => (
+						<form
+							ref={ formRef }
+							onSubmit={ ( event ) => {
+								event.preventDefault();
+								setIsOpen( false );
+							} }
+						>
+							<ValidatedInputControl
+								{ ...args }
+								value={ text }
+								onChange={ setText }
+								customValidity={
+									text === 'error'
+										? {
+												type: 'invalid',
+												message:
+													'The word "error" is not allowed.',
+										  }
+										: undefined
+								}
+							/>
+						</form>
+					) }
+					renderToggle={ () => {
+						return (
+							<Button
+								__next40pxDefaultSize
+								variant="secondary"
+								onClick={ () => setIsOpen( ! isOpen ) }
+								aria-expanded={ isOpen }
+							>
+								Open in popover
+							</Button>
+						);
+					} }
+				/>
+			);
+		},
+		args: {
+			label: 'Text',
+			help: 'The word "error" will trigger an error.',
+			required: true,
+			style: {
+				width: '200px',
+			},
+		},
+	};
