diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index 01a7ae865b10dd..11fd97fe6c20a4 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -6,6 +6,7 @@
 
 -   `TextareaControl`: Add `min-height` to the textarea element ([#72867](https://github.com/WordPress/gutenberg/pull/72867)).
 -   `Card`, `CardHeader`, `CardBody` and `CardFooter`: Add support for directional padding through object-based size prop, allowing independent control of padding for each side ([#72511](https://github.com/WordPress/gutenberg/pull/72511)).
+-   `CustomSelectControl`: Fix the options with the same name are shown as the selected option ([#72189](https://github.com/WordPress/gutenberg/pull/72189)).
 
 ## 30.7.0 (2025-10-29)
 
diff --git a/packages/components/src/custom-select-control/index.tsx b/packages/components/src/custom-select-control/index.tsx
index e014e4bc642eef..d9c74bce79fcee 100644
--- a/packages/components/src/custom-select-control/index.tsx
+++ b/packages/components/src/custom-select-control/index.tsx
@@ -43,13 +43,13 @@ function applyOptionDeprecations( {
 	};
 }
 
-function getDescribedBy( currentValue: string, describedBy?: string ) {
+function getDescribedBy( currentName: string, describedBy?: string ) {
 	if ( describedBy ) {
 		return describedBy;
 	}
 
 	// translators: %s: The selected option.
-	return sprintf( __( 'Currently selected: %s' ), currentValue );
+	return sprintf( __( 'Currently selected: %s' ), currentName );
 }
 
 function CustomSelectControl< T extends CustomSelectOption >(
@@ -84,7 +84,7 @@ function CustomSelectControl< T extends CustomSelectOption >(
 	const store = Ariakit.useSelectStore< string >( {
 		async setValue( nextValue ) {
 			const nextOption = options.find(
-				( item ) => item.name === nextValue
+				( item ) => item.key === nextValue
 			);
 
 			if ( ! onChange || ! nextOption ) {
@@ -108,12 +108,12 @@ function CustomSelectControl< T extends CustomSelectOption >(
 			};
 			onChange( changeObject );
 		},
-		value: value?.name,
+		value: value?.key,
 		// Setting the first option as a default value when no value is provided
 		// is already done natively by the underlying Ariakit component,
 		// but doing this explicitly avoids the `onChange` callback from firing
 		// on initial render, thus making this implementation closer to the v1.
-		defaultValue: options[ 0 ]?.name,
+		defaultValue: options[ 0 ]?.key,
 	} );
 
 	const children = options
@@ -134,7 +134,7 @@ function CustomSelectControl< T extends CustomSelectOption >(
 			return (
 				<CustomSelectItem
 					key={ key }
-					value={ name }
+					value={ key }
 					children={ hint ? withHint : name }
 					style={ style }
 					className={ clsx(
@@ -151,22 +151,25 @@ function CustomSelectControl< T extends CustomSelectOption >(
 
 	const currentValue = Ariakit.useStoreState( store, 'value' );
 
-	const renderSelectedValueHint = () => {
-		const selectedOptionHint = options
+	const selectedOption =
+		options
 			?.map( applyOptionDeprecations )
-			?.find( ( { name } ) => currentValue === name )?.hint;
+			?.find( ( { key } ) => currentValue === key ) ?? options[ 0 ];
+
+	const renderSelectedValue = () => {
+		if ( ! showSelectedHint || ! selectedOption.hint ) {
+			return selectedOption?.name;
+		}
 
 		return (
 			<Styled.SelectedExperimentalHintWrapper>
-				{ currentValue }
-				{ selectedOptionHint && (
-					<Styled.SelectedExperimentalHintItem
-						// Keeping the classname for legacy reasons
-						className="components-custom-select-control__hint"
-					>
-						{ selectedOptionHint }
-					</Styled.SelectedExperimentalHintItem>
-				) }
+				{ selectedOption?.name }
+				<Styled.SelectedExperimentalHintItem
+					// Keeping the classname for legacy reasons
+					className="components-custom-select-control__hint"
+				>
+					{ selectedOption?.hint }
+				</Styled.SelectedExperimentalHintItem>
 			</Styled.SelectedExperimentalHintWrapper>
 		);
 	};
@@ -188,9 +191,7 @@ function CustomSelectControl< T extends CustomSelectOption >(
 		<>
 			<_CustomSelect
 				aria-describedby={ descriptionId }
-				renderSelectedValue={
-					showSelectedHint ? renderSelectedValueHint : undefined
-				}
+				renderSelectedValue={ renderSelectedValue }
 				size={ translatedSize }
 				store={ store }
 				className={ clsx(
@@ -205,7 +206,7 @@ function CustomSelectControl< T extends CustomSelectOption >(
 			</_CustomSelect>
 			<VisuallyHidden>
 				<span id={ descriptionId }>
-					{ getDescribedBy( currentValue, describedBy ) }
+					{ getDescribedBy( selectedOption?.name, describedBy ) }
 				</span>
 			</VisuallyHidden>
 		</>
