diff --git a/packages/components/CHANGELOG.md b/packages/components/CHANGELOG.md
index 6ac170d77ba647..95b780ebf244fd 100644
--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -8,6 +8,7 @@
 -   `TextareaControl`: Add `min-height` to the textarea element ([#72867](https://github.com/WordPress/gutenberg/pull/72867)).
 -   `Card`, `CardHeader`, `CardBody` and `CardFooter`: Add support for directional padding through object-based size prop, allowing independent control of padding for each side ([#72511](https://github.com/WordPress/gutenberg/pull/72511)).
 -   `CustomSelectControl`: Fix the options with the same name are shown as the selected option ([#72189](https://github.com/WordPress/gutenberg/pull/72189)).
+-   `NumberControl`: Fix crash when min prop is string and step prop contains decimal ([#73107](https://github.com/WordPress/gutenberg/pull/73107)).
 
 ## 30.7.0 (2025-10-29)
 
diff --git a/packages/components/src/utils/math.js b/packages/components/src/utils/math.js
index ae913a630fc848..1c7321c85a3d1f 100644
--- a/packages/components/src/utils/math.js
+++ b/packages/components/src/utils/math.js
@@ -81,12 +81,12 @@ export function clamp( value, min, max ) {
  */
 export function ensureValidStep( value, min, step ) {
 	const baseValue = getNumber( value );
+	const minValue = getNumber( min );
 	const stepValue = getNumber( step );
 	const precision = Math.max( getPrecision( step ), getPrecision( min ) );
-	const realMin = Math.abs( min ) === Infinity ? 0 : min;
 	// If the step is not a factor of the minimum then the step must be
 	// offset by the minimum.
-	const tare = realMin % stepValue ? realMin : 0;
+	const tare = minValue % stepValue ? minValue : 0;
 	const rounded = Math.round( ( baseValue - tare ) / stepValue ) * stepValue;
 	const fromMin = rounded + tare;
 	return precision ? getNumber( fromMin.toFixed( precision ) ) : fromMin;
