diff --git a/packages/components/src/badge/stories/e2e/index.story.tsx b/packages/components/src/badge/stories/e2e/index.story.tsx
new file mode 100644
index 00000000000000..5b30e68baafead
--- /dev/null
+++ b/packages/components/src/badge/stories/e2e/index.story.tsx
@@ -0,0 +1,21 @@
+import type { Meta, StoryObj } from '@storybook/react-vite';
+
+import Badge from '../..';
+
+const meta: Meta< typeof Badge > = {
+	component: Badge,
+	title: 'Components/Badge',
+};
+export default meta;
+
+type Story = StoryObj< typeof Badge >;
+
+export const TextOverflow: Story = {
+	args: {
+		children:
+			'This is an extremely long label thatshoulddemonstratetextoverflow behavior',
+	},
+	parameters: {
+		textOverflowContainers: true,
+	},
+};
